Replace deprecated bcopy call by using memmove instead.

Index: b/ttcp.c
===================================================================
--- a/ttcp.c
+++ b/ttcp.c
@@ -241,7 +241,7 @@
 			if ((addr=gethostbyname(host)) == NULL)
 				err("bad hostname");
 			sinhim.sin_family = addr->h_addrtype;
-			bcopy(addr->h_addr,(char*)&addr_tmp, addr->h_length);
+			memmove((char*)&addr_tmp, addr->h_addr, addr->h_length);
 #if defined(cray)
 			sinhim.sin_addr = addr_tmp;
 #else
