diff -ruN xf86-video-intel-2.4.2/configure.ac xf86-video-intel-2.4.2-patched/configure.ac
--- xf86-video-intel-2.4.2/configure.ac	2008-08-26 03:28:31.000000000 +0200
+++ xf86-video-intel-2.4.2-patched/configure.ac	2009-07-27 13:39:49.000000000 +0200
@@ -91,6 +91,7 @@
 # Checks for header files.
 AC_HEADER_STDC
 
+if test "$cross_compiling" = "no" ; then
 if test x$DRI != xno; then
         AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
@@ -111,6 +112,14 @@
                 DRI="no"
         fi
 fi
+else
+if test x$DRI != xno; then
+	have_dri_h="yes"
+	have_sarea_h="yes"
+	have_dristruct_h="yes"
+	DRI="yes"
+fi
+fi
 AC_MSG_RESULT([$DRI])
 
 save_CFLAGS="$CFLAGS"
