--- xf86-video-i810-1.6.5/configure.ac.orig	2007-06-10 20:59:27.000000000 +0200
+++ xf86-video-i810-1.6.5/configure.ac	2007-06-10 20:59:50.000000000 +0200
@@ -92,13 +92,14 @@ AC_HEADER_STDC
 
 AC_MSG_CHECKING([whether to include DRI support])
 if test x$DRI = xauto; then
+ 	if test "$cross_compiling" = "no" ; then
         AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
         AC_CHECK_FILE([${sdkdir}/sarea.h],
                       [have_sarea_h="yes"], [have_sarea_h="no"])
         AC_CHECK_FILE([${sdkdir}/dristruct.h],
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
-	AC_CHECK_FILE([${sdkdir}/damage.h],
+		AC_CHECK_FILE([${sdkdir}/damage.h],
                       [have_damage_h="yes"], [have_damage_h="no"])
 
         if test "$have_dri_h" = yes -a \
@@ -108,6 +109,12 @@ if test x$DRI = xauto; then
         else
                 DRI="no"
         fi
+ 	else
+ 		have_dri_h="yes"
+ 		have_sarea_h="yes"
+		have_dristruct_h="yes"
+		DRI="yes"
+ 	fi
 fi
 AC_MSG_RESULT([$DRI])
