--- a/configure.ac.orig	2010-01-04 22:20:07.000000000 +0000
+++ b/configure.ac	2010-02-19 12:12:50.000000000 +0000
@@ -103,7 +103,7 @@
 CFLAGS="$XORG_CFLAGS $DRI_CFLAGS $DRM_CFLAGS"
 CPPFLAGS="$XORG_CFLAGS $DRI_CFLAGS $DRM_CFLAGS"
 AC_MSG_CHECKING([whether to include DRI support])
-if test x$DRI != xno; then
+if test "$cross_compiling" = "no" -a test x$DRI != xno; then
         AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
         AC_CHECK_FILE([${sdkdir}/sarea.h],
