Index: cramfs-1.1/cramfsck.c
===================================================================
--- cramfs-1.1.orig/cramfsck.c	2011-09-09 15:10:06.000000000 +0200
+++ cramfs-1.1/cramfsck.c	2011-09-09 15:13:49.950897747 +0200
@@ -57,6 +57,10 @@
 #include <byteswap.h>
 #include <zlib.h>
 
+#ifdef __CYGWIN__
+#define _IO(type,nr)   _IOC(_IOC_NONE,(type),(nr),0)
+#endif /* __CYGWIN__ */
+
 /* Exit codes used by fsck-type programs */
 #define FSCK_OK          0	/* No errors */
 #define FSCK_NONDESTRUCT 1	/* File system errors corrected */
