From 4618fb3b788715c754872df5fed7bce0186f5089 Mon Sep 17 00:00:00 2001
From: Peter Seiderer <ps.report@gmx.net>
Date: Sun, 27 Apr 2014 13:23:41 +0200
Subject: [PATCH] configure.ac: add '--disable-python-bindings' option

Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 Makefile.am  |  2 +-
 configure.ac | 11 +++++++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index a83b35f..fb87c76 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = src tools python $(SUBDIR_TESTS)
+SUBDIRS = src tools $(SUBDIR_PYTHON_BINDINGS) $(SUBDIR_TESTS)
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = evemu.pc
diff --git a/configure.ac b/configure.ac
index c7e16b3..72c5fdf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -54,6 +54,17 @@ AS_IF([test "x$enable_tests" != "xno"], [
 
 AC_SUBST([SUBDIR_TESTS])
 
+AC_ARG_ENABLE([python-bindings],
+	AS_HELP_STRING([--disable-python-bindings],
+		 [Disable generation of python bindings]))
+
+AS_IF([test "x$enable_python_bindings" != "xno"], [
+	SUBDIR_PYTHON_BINDINGS=python
+])
+
+AC_SUBST([SUBDIR_PYTHON_BINDINGS])
+
+
 AC_SUBST(AM_CFLAGS,
          "-Wall -Wextra")
 
-- 
1.8.1.4

