diff -NurBabdH gtk+-1.2.10/gtk/testrgb.c gtk+-1.2.10-timezone-fix/gtk/testrgb.c
--- gtk+-1.2.10/gtk/testrgb.c	2000-08-18 23:36:34.000000000 +0200
+++ gtk+-1.2.10-timezone-fix/gtk/testrgb.c	2005-03-03 11:18:09.000000000 +0100
@@ -49,7 +49,10 @@
 get_time (void)
 {
   struct timeval tv;
-  struct timezone tz;
+   struct timezone {
+      int     tz_minuteswest; /* minutes W of Greenwich */
+      int     tz_dsttime;     /* type of dst correction */
+   } tz;
 
   gettimeofday (&tv, &tz);
 
