From 8287848f747bd37298a2d32b8fd06cc59eb28028 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Thu, 9 Feb 2017 21:40:43 +0100
Subject: [PATCH] Add --disable-tests to configure

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 Makefile.am  | 6 +++++-
 configure.ac | 6 ++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 1804945..cebcd31 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -5,8 +5,12 @@ nodist_include_HEADERS = include/maxminddb_config.h
 
 SUBDIRS = \
   src     \
-  bin     \
+  bin
+
+if TESTS
+SUBDIRS += \
   t
+endif
 
 EXTRA_DIST = doc t Changes.md LICENSE NOTICE README.md projects/VS12 projects/VS12-tests
 dist-hook:
diff --git a/configure.ac b/configure.ac
index 7916212..2a22fab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -119,6 +119,12 @@ AC_ARG_ENABLE(
         esac],[debug=false])
 AM_CONDITIONAL([DEBUG], [test x$debug = xtrue])
 
+AC_ARG_ENABLE([tests],
+        AS_HELP_STRING([--enable-tests], [Compilation of tests code]),
+        [enable_tests=${enableval}],
+        [enable_tests=yes])
+AM_CONDITIONAL([TESTS], [test "${enable_tests}" = "yes"])
+
 AC_CONFIG_FILES([Makefile
                  src/Makefile
                  bin/Makefile
-- 
2.5.0

