diff -urN xc-011010-0rig/include/Xfuncs.h xc-011010/include/Xfuncs.h
--- xc-011010-0rig/include/Xfuncs.h	2001-07-25 17:04:42.000000000 +0200
+++ xc-011010/include/Xfuncs.h	2007-08-13 13:08:39.000000000 +0200
@@ -45,6 +45,9 @@
 #include <strings.h>
 /* bcopy, bcmp, bzero declared */
 #define _XFUNCS_H_INCLUDED_STRING_H
+#elif defined(__UCLIBC__)
+#include <string.h>
+#define _XFUNCS_H_INCLUDED_STRING_H
 #else
 #include <string.h>
 #define _XFUNCS_H_INCLUDED_STRING_H
