diff -rdup at-3.1.10.orig/Makefile.in at-3.1.10/Makefile.in
--- at-3.1.10.orig/Makefile.in	2007-01-22 14:35:41.000000000 +0100
+++ at-3.1.10/Makefile.in	2007-01-22 14:41:41.000000000 +0100
@@ -62,6 +62,8 @@ MISC =  COPYING  Makefile.in configure a
 DIST = $(CSRCS) $(HEADERS) $(MISC) $(OTHERS)
 LIST = Filelist Filelist.asc
 
+IROOT = $(DESTDIR)
+
 .PHONY: all install clean dist distclean
 
 all: at atd atrun
@@ -78,7 +80,7 @@ atd: $(RUNOBJECTS)
 y.tab.c y.tab.h: parsetime.y
 	$(YACC) -d parsetime.y
 
-lex.yy.c: parsetime.l
+lex.yy.c: parsetime.l y.tab.h
 	$(LEX) -i parsetime.l
 
 atrun: atrun.in
diff -rdup at-3.1.10.orig/config.h.in at-3.1.10/config.h.in
--- at-3.1.10.orig/config.h.in	2007-01-22 14:35:41.000000000 +0100
+++ at-3.1.10/config.h.in	2007-01-22 14:35:35.000000000 +0100
@@ -181,3 +181,6 @@
 
 #undef HAVE_ATTRIBUTE_NORETURN
 #undef HAVE_PAM
+
+#undef NEED_YYWRAP
+
diff -rdup at-3.1.10.orig/configure at-3.1.10/configure
--- at-3.1.10.orig/configure	2007-01-22 14:35:41.000000000 +0100
+++ at-3.1.10/configure	2007-01-22 14:35:35.000000000 +0100
@@ -1037,7 +1037,7 @@ esac
 echo $ac_n "checking Trying to compile a trivial ANSI C program""... $ac_c" 1>&6
 echo "configure:1039: checking Trying to compile a trivial ANSI C program" >&5
 if test "$cross_compiling" = yes; then
-  { echo "configure: error: Could not compile and run even a trivial ANSI C program - check CC." 1>&2; exit 1; }
+  echo "$ac_t""assuming it works" 1>&6
 else
   cat > conftest.$ac_ext <<EOF
 #line 1044 "configure"
diff -rdup at-3.1.10.orig/getloadavg.c at-3.1.10/getloadavg.c
--- at-3.1.10.orig/getloadavg.c	2007-01-22 14:35:41.000000000 +0100
+++ at-3.1.10/getloadavg.c	2007-01-22 14:35:35.000000000 +0100
@@ -66,11 +66,12 @@ Boston, MA  02110-1301  USA */
 
 /* This should always be first.  */
 #ifdef HAVE_CONFIG_H
-#include <config.h>
+#include "config.h"
 #endif
 
-#include "lisp.h"
-#include "sysfile.h" /* for encapsulated open, close, read, write */
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
 
 #ifndef HAVE_GETLOADAVG
 
diff -rdup at-3.1.10.orig/parsetime.h at-3.1.10/parsetime.h
--- at-3.1.10.orig/parsetime.h	2005-08-05 05:16:01.000000000 +0200
+++ at-3.1.10/parsetime.h	2007-01-22 14:45:41.000000000 +0100
@@ -17,6 +17,7 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
 
+#include "panic.h"
 time_t parsetime(int argc, char **argv);
 
 extern char *last_token;
diff -rdup at-3.1.10.orig/parsetime.l at-3.1.10/parsetime.l
--- at-3.1.10.orig/parsetime.l	2007-01-22 14:35:41.000000000 +0100
+++ at-3.1.10/parsetime.l	2007-01-22 14:35:35.000000000 +0100
@@ -1,5 +1,6 @@
 %{
 
+#include "config.h"
 #include <string.h>
 #include <time.h>
 #include "y.tab.h"
