From 649140f09d8a8805125830cd7b2327d35447ffe8 Mon Sep 17 00:00:00 2001
From: Eric Le Bihan <eric.le.bihan.dev@free.fr>
Date: Sat, 14 Jul 2018 11:18:45 +0200
Subject: [PATCH] Only fix RPATH if install_rpath is not empty

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---
 mesonbuild/minstall.py | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/mesonbuild/minstall.py b/mesonbuild/minstall.py
index 1d721795..e04e1f6a 100644
--- a/mesonbuild/minstall.py
+++ b/mesonbuild/minstall.py
@@ -495,6 +495,14 @@ class Installer:
                 try:
                     depfixer.fix_rpath(outname, install_rpath, final_path,
                                        install_name_mappings, verbose=False)
+                    # Buildroot check-host-rpath script expects RPATH
+                    # But if install_rpath is empty, it will stripped.
+                    # So, preserve it in this case
+                    if install_rpath:
+                        depfixer.fix_rpath(outname, install_rpath, final_path,
+                                           install_name_mappings, verbose=False)
+                    else:
+                        print("Skipping RPATH fixing")
                 except SystemExit as e:
                     if isinstance(e.code, int) and e.code == 0:
                         pass
-- 
2.14.4

