From https://bugs.webkit.org/show_bug.cgi?id=130837

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura webkitgtk-2.4.8/Source/WebKit2/Platform/IPC/Connection.h webkitgtk-2.4.8-ppc/Source/WebKit2/Platform/IPC/Connection.h
--- webkitgtk-2.4.8/Source/WebKit2/Platform/IPC/Connection.h	2015-01-07 06:45:43.000000000 -0300
+++ webkitgtk-2.4.8-ppc/Source/WebKit2/Platform/IPC/Connection.h	2015-04-20 01:55:41.554547510 -0300
@@ -216,7 +216,11 @@
 
     Client* m_client;
     bool m_isServer;
+#if CPU(PPC)
+    uint64_t m_syncRequestID;
+#else
     std::atomic<uint64_t> m_syncRequestID;
+#endif
 
     bool m_onlySendMessagesAsDispatchWhenWaitingForSyncReplyWhenProcessingSuchAMessage;
     bool m_shouldExitOnSyncMessageSendFailure;
diff -Nura webkitgtk-2.4.8/Source/WebKit2/UIProcess/StatisticsRequest.cpp webkitgtk-2.4.8-ppc/Source/WebKit2/UIProcess/StatisticsRequest.cpp
--- webkitgtk-2.4.8/Source/WebKit2/UIProcess/StatisticsRequest.cpp	2015-01-07 06:45:43.000000000 -0300
+++ webkitgtk-2.4.8-ppc/Source/WebKit2/UIProcess/StatisticsRequest.cpp	2015-04-20 01:55:41.555547544 -0300
@@ -44,7 +44,11 @@
 
 uint64_t StatisticsRequest::addOutstandingRequest()
 {
+#if CPU(PPC)
+    static int64_t uniqueRequestID;
+#else
     static std::atomic<int64_t> uniqueRequestID;
+#endif
 
     uint64_t requestID = ++uniqueRequestID;
     m_outstandingRequests.add(requestID);
