--- mtd-utils-e6088d987c545d60a86e1f44836ab8ba072fffd9/ubi-utils/new-utils/src/ubicrc32.c.orig	2008-03-26 16:08:13.000000000 +1100
+++ mtd-utils-e6088d987c545d60a86e1f44836ab8ba072fffd9/ubi-utils/new-utils/src/ubicrc32.c	2008-03-26 16:08:20.000000000 +1100
@@ -26,7 +26,6 @@
 #include <stdint.h>
 #include <stdlib.h>
 #include <getopt.h>
-#include <argp.h>
 #include <unistd.h>
 #include <mtd/ubi-header.h>
 
