Add RUN

Signed-off-by: Francois Perrad <francois.perrad@gadz.org>

Index: b/mkppport
===================================================================
--- a/mkppport
+++ b/mkppport
@@ -1,6 +1,7 @@
 use strict;
 use warnings;
 
+use Config;
 use Getopt::Long;
 use File::Spec;
 use File::Compare qw( compare );
@@ -136,14 +137,18 @@
 #----------------------------------------------
 sub run
 {
+  my @run = $Config{run};
+  @run = () if not defined $run[0] or $run[0] eq '';
   my @args = ("-I" . File::Spec->catdir((File::Spec->updir) x 2, 'lib'), @_);
   my $run = $perl =~ m/\s/ ? qq("$perl") : $perl;
   for (@args) {
     $_ = qq("$_") if $^O eq 'VMS' && /^[^"]/;
     $run .= " $_";
   }
-  print "running $run\n";
-  system $run and die "$run failed: $?\n";
+#  print "running $run\n";
+#  system $run and die "$run failed: $?\n";
+  print "running ", join(' ', @run, $run), "\n";
+  system join(' ', @run, $run) and die "@run $run failed: $?\n";
 }
 
 __END__
