
Signed-off-by: Francois Perrad <francois.perrad@gadz.org>

Index: b/cpan/Module-Build/lib/Module/Build/Base.pm
===================================================================
--- a/cpan/Module-Build/lib/Module/Build/Base.pm
+++ b/cpan/Module-Build/lib/Module/Build/Base.pm
@@ -460,7 +460,7 @@
   my $proto = shift;
   my $c     = ref($proto) ? $proto->{config} : 'Module::Build::Config';
 
-  my $perl  = $^X;
+  my $perl  = $ENV{RUN_PERL} || $^X;
   my $perl_basename = File::Basename::basename($perl);
 
   my @potential_perls;
