Don't use RUN with make (only for perl)

Signed-off-by: Francois Perrad <francois.perrad@gadz.org>

Index: b/make_ext.pl
===================================================================
--- a/make_ext.pl
+++ b/make_ext.pl
@@ -458,11 +458,13 @@
 	# Give makefile an opportunity to rewrite itself.
 	# reassure users that life goes on...
 	my @args = ('config', @$pass_through);
-	system(@run, @make, @args) and print "@run @make @args failed, continuing anyway...\n";
+#	system(@run, @make, @args) and print "@run @make @args failed, continuing anyway...\n";
+	system(@make, @args) and print "@make @args failed, continuing anyway...\n";
     }
     my @targ = ($target, @$pass_through);
     print "Making $target in $ext_dir\n@run @make @targ\n";
-    my $code = system(@run, @make, @targ);
+#    my $code = system(@run, @make, @targ);
+    my $code = system(@make, @targ);
     die "Unsuccessful make($ext_dir): code=$code" if $code != 0;
 
     chdir $return_dir || die "Cannot cd to $return_dir: $!";
