--- binutils/gas/config/tc-nios2.c	2007/01/04 16:54:43	194
+++ binutils/gas/config/tc-nios2.c	2007/01/04 16:58:16	195
@@ -1661,7 +1661,7 @@
 
 
 	  /* apply the rightshift */
-	  (signed) fixup >>= howto->rightshift;
+	  fixup >>= howto->rightshift;
 
 	  /* truncate the fixup to right size */
 	  switch (fixP->fx_r_type)
@@ -1706,7 +1706,7 @@
 nios2_check_overflow (valueT fixup, reloc_howto_type * howto)
 {
   /* apply the rightshift before checking for overflow */
-  (signed) fixup >>= howto->rightshift;
+  fixup >>= howto->rightshift;
 
   /* check for overflow - return TRUE if overflow, FALSE if not */
   switch (howto->complain_on_overflow)
