From c060e6f06a81ffa400266a6d0bfc58511265a16f Mon Sep 17 00:00:00 2001
From: Olivier Singla <olivier.singla@gmail.com>
Date: Thu, 4 Aug 2016 21:23:41 +0200
Subject: [PATCH] Makefile: disable always building statically

Signed-off-by: Olivier Singla <olivier.singla@gmail.com>
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 7b8e7ab..5c4c71e 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -126,7 +126,7 @@ BUILT_SOURCES        = shellinabox/beep.h                                     \
 
 shellinaboxd_LDADD   = liblogging.la                                          \
                        libhttp.la
-shellinaboxd_LDFLAGS = -static
+shellinaboxd_LDFLAGS =
 ## Added this for compatibility with older versions of autoconf/automake
 docdir               = ${datadir}/doc/${PACKAGE}
 
-- 
2.7.4

