Fix various compilation errors when compiling against must libc

Signed-off-by: Dagg Stompler <daggs@gmx.com>

diff -Nupr a/src/init.c b/src/init.c
--- a/src/init.c	2017-08-13 18:44:27.000000000 +0300
+++ b/src/init.c	2017-08-13 18:34:45.000000000 +0300
@@ -37,6 +37,7 @@
 #include <sys/kd.h>
 #endif
 #include <sys/resource.h>
+#include <sys/ttydefaults.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <errno.h>
diff -Nupr a/src/init.h b/src/init.h
--- a/src/init.h	2017-08-13 19:35:46.000000000 +0300
+++ b/src/init.h	2017-08-13 19:36:01.000000000 +0300
@@ -22,6 +22,8 @@
  *
  */
 
+#include <time.h>
+
 /* Standard configuration */
 #define CHANGE_WAIT 0			/* Change runlevel while
 					   waiting for a process to exit? */
diff -Nupr a/src/mountpoint.c b/src/mountpoint.c
--- a/src/mountpoint.c	2017-08-13 20:28:26.000000000 +0300
+++ b/src/mountpoint.c	2017-08-13 20:28:40.000000000 +0300
@@ -24,6 +24,7 @@
  */
 
 #include <sys/stat.h>
+#include <sys/types.h>
 #include <unistd.h>
 #include <stdlib.h>
 #include <string.h>
