diff -Naupr blackbox-0.70.1.orig/lib/Image.cc blackbox-0.70.1/lib/Image.cc
--- blackbox-0.70.1.orig/lib/Image.cc	2005-04-08 17:41:09.000000000 +0200
+++ blackbox-0.70.1/lib/Image.cc	2008-02-24 08:31:28.000000000 +0100
@@ -42,8 +42,9 @@
 
 #include <assert.h>
 #include <math.h>
-#include <stdio.h>
-#include <stdlib.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 // #define COLORTABLE_DEBUG
 // #define MITSHM_DEBUG
diff -Naupr blackbox-0.70.1.orig/lib/Resource.cc blackbox-0.70.1/lib/Resource.cc
--- blackbox-0.70.1.orig/lib/Resource.cc	2005-04-06 23:16:50.000000000 +0200
+++ blackbox-0.70.1/lib/Resource.cc	2008-02-24 08:33:11.000000000 +0100
@@ -28,7 +28,8 @@
 #include <X11/Xlib.h>
 #include <X11/Xresource.h>
 
-#include <stdio.h>
+#include <cstdio>
+#include <cstring>
 
 
 bt::Resource::Resource(void)
diff -Naupr blackbox-0.70.1.orig/lib/XDG.cc blackbox-0.70.1/lib/XDG.cc
--- blackbox-0.70.1.orig/lib/XDG.cc	2005-04-06 16:04:38.000000000 +0200
+++ blackbox-0.70.1/lib/XDG.cc	2008-02-24 08:34:11.000000000 +0100
@@ -25,7 +25,8 @@
 #include "Util.hh"
 #include "XDG.hh"
 
-#include <stdlib.h>
+#include <cstdlib>
+#include <algorithm>
 
 
 // make sure directory names end with a slash
diff -Naupr blackbox-0.70.1.orig/src/BlackboxResource.cc blackbox-0.70.1/src/BlackboxResource.cc
--- blackbox-0.70.1.orig/src/BlackboxResource.cc	2005-10-18 09:34:46.000000000 +0200
+++ blackbox-0.70.1/src/BlackboxResource.cc	2008-02-24 08:34:49.000000000 +0100
@@ -33,6 +33,8 @@
 #include <X11/Xutil.h>
 #include <X11/cursorfont.h>
 
+#include <cstring>
+
 
 BlackboxResource::BlackboxResource(const std::string& rc): rc_file(rc) {
   screen_resources = 0;
diff -Naupr blackbox-0.70.1.orig/src/main.cc blackbox-0.70.1/src/main.cc
--- blackbox-0.70.1.orig/src/main.cc	2005-01-03 10:42:57.000000000 +0100
+++ blackbox-0.70.1/src/main.cc	2008-02-24 08:37:16.000000000 +0100
@@ -34,7 +34,8 @@
 #include "blackbox.hh"
 #include "../version.h"
 
-#include <stdio.h>
+#include <cstdio>
+#include <cstring>
 
 
 static void showHelp(int exitval) {
diff -Naupr blackbox-0.70.1.orig/src/Screen.cc blackbox-0.70.1/src/Screen.cc
--- blackbox-0.70.1.orig/src/Screen.cc	2005-10-18 10:07:22.000000000 +0200
+++ blackbox-0.70.1/src/Screen.cc	2008-02-24 08:35:46.000000000 +0100
@@ -45,8 +45,9 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <assert.h>
-#include <ctype.h>
 #include <dirent.h>
+#include <cctype>
+#include <cstring>
 
 
 static bool running = true;
diff -Naupr blackbox-0.70.1.orig/src/ScreenResource.cc blackbox-0.70.1/src/ScreenResource.cc
--- blackbox-0.70.1.orig/src/ScreenResource.cc	2005-04-13 07:54:08.000000000 +0200
+++ blackbox-0.70.1/src/ScreenResource.cc	2008-02-24 08:36:31.000000000 +0100
@@ -33,6 +33,8 @@
 
 #include <assert.h>
 
+#include <cstring>
+
 
 static const int iconify_width  = 9;
 static const int iconify_height = 9;
diff -Naupr blackbox-0.70.1.orig/util/bsetroot.cc blackbox-0.70.1/util/bsetroot.cc
--- blackbox-0.70.1.orig/util/bsetroot.cc	2005-03-15 08:01:37.000000000 +0100
+++ blackbox-0.70.1/util/bsetroot.cc	2008-02-24 08:38:41.000000000 +0100
@@ -30,7 +30,9 @@
 #include <cctype>
 
 #include <X11/Xatom.h>
-#include <stdio.h>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
 
 
 // ignore all X errors
