--- a/configure	2005-11-03 03:27:16.000000000 -0500
+++ b/configure	2008-02-02 23:25:25.000000000 -0500
@@ -1980,12 +1980,6 @@ am__tar='${AMTAR} chof - "$$tardir"'; am
 
 test "x$prefix" = "xNONE" && prefix="$ac_default_prefix"
 
-CPPFLAGS="$CPPFLAGS -I$prefix/include"
-LDFLAGS="$LDFLAGS -L$prefix/lib"
-if test "x$prefix" != "x/usr/local"; then
-  CPPFLAGS="$CPPFLAGS -I/usr/local/include"
-  LDFLAGS="$LDFLAGS -L/usr/local/lib"
-fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
