Fix nios2 detection.

Apache treats nios2 as OS/2.

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr httpd-2.4.12.org/configure.in httpd-2.4.12/configure.in
--- httpd-2.4.12.org/configure.in	2015-01-22 18:33:07.000000000 +0100
+++ httpd-2.4.12/configure.in	2015-04-02 22:01:32.851102219 +0200
@@ -268,7 +268,7 @@
 AC_MSG_NOTICE([])
 
 case $host in
-  *os2*)
+  *-os2*)
       # Use a custom made libtool replacement
       echo "using aplibtool"
       LIBTOOL="$abs_srcdir/srclib/apr/build/aplibtool"
