diff --git a/motion.c b/motion.c
index 805c490..3b2418a 100644
--- a/motion.c
+++ b/motion.c
@@ -322,7 +322,20 @@ static void sig_handler(int signo)
         if (cnt_list) {
             i = -1;
             while (cnt_list[++i])
-                cnt_list[i]->makemovie = 1;
+                cnt_list[i]->conf.emulate_motion = 1;
+                //cnt_list[i]->makemovie = 1;
+        }
+        break;
+    case SIGUSR2:
+        /*
+         * Ouch! We have been hit from the outside! Someone wants us to
+         * stop making a movie!
+         */
+        if (cnt_list) {
+            i = -1;
+            while (cnt_list[++i])
+                cnt_list[i]->conf.emulate_motion = 0;
+                //cnt_list[i]->makemovie = 1;
         }
         break;
     case SIGHUP:
@@ -2557,6 +2570,7 @@ static void setup_signals(struct sigaction *sig_handler_action, struct sigaction
     sigaction(SIGQUIT, sig_handler_action, NULL);
     sigaction(SIGTERM, sig_handler_action, NULL);
     sigaction(SIGUSR1, sig_handler_action, NULL);
+    sigaction(SIGUSR2, sig_handler_action, NULL);
 
     /* use SIGVTALRM as a way to break out of the ioctl, don't restart */
     sig_handler_action->sa_flags = 0;
