commit 0cd12918f93fc6559dad56d176a60ddc3f0844d6
Author: Joo Aun Saw <jasaw@dius.com.au>
Date:   Thu Sep 5 14:20:07 2019 +1000

    ffmpeg: ensure first frame PTS is zero

diff --git a/ffmpeg.c b/ffmpeg.c
index 7a42d24..665aa00 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -512,7 +512,11 @@ static int ffmpeg_set_pts(struct ffmpeg *ffmpeg, const struct timeval *tv1){
             ffmpeg_reset_movie_start_time(ffmpeg, tv1);
             pts_interval = 0;
         }
-        ffmpeg->picture->pts = av_rescale_q(pts_interval,(AVRational){1, 1000000L},ffmpeg->video_st->time_base) + ffmpeg->base_pts;
+        if (ffmpeg->last_pts < 0) {
+            // This is the very first frame, ensure PTS is zero
+            ffmpeg->picture->pts = 0;
+        } else
+            ffmpeg->picture->pts = av_rescale_q(pts_interval,(AVRational){1, 1000000L},ffmpeg->video_st->time_base) + ffmpeg->base_pts;
 
         if (ffmpeg->test_mode == TRUE){
             MOTION_LOG(INF, TYPE_ENCODER, NO_ERRNO
