diff --git a/ffmpeg.c b/ffmpeg.c
index 3f48db8..1819e35 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -419,6 +419,8 @@ struct ffmpeg *ffmpeg_open(const char *ffmpeg_video_codec, char *filename,
     c->codec_id   = ffmpeg->oc->oformat->video_codec;
     c->codec_type = AVMEDIA_TYPE_VIDEO;
     c->bit_rate   = bps;
+    if (ffmpeg->oc->oformat->video_codec == AV_CODEC_ID_H264)
+        av_opt_set(c->priv_data, "crf", "1", AV_OPT_SEARCH_CHILDREN);
     c->width      = width;
     c->height     = height;
     c->time_base.num = 1;
diff --git a/ffmpeg.h b/ffmpeg.h
index ccdc0f1..7067569 100644
--- a/ffmpeg.h
+++ b/ffmpeg.h
@@ -13,6 +13,7 @@
 #include <libavformat/avformat.h>
 #include <libavutil/imgutils.h>
 #include <libavutil/mathematics.h>
+#include <libavutil/opt.h>
 
 #if (LIBAVFORMAT_VERSION_MAJOR >= 56)
 
