diff --git a/ffmpeg.c b/ffmpeg.c
index 739bcb3..aa17ddc 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -349,8 +349,6 @@ static int ffmpeg_encode_video(struct ffmpeg *ffmpeg){
         //Packet is freed upon failure of encoding
         return -1;
     }
-    if (ffmpeg->picture->key_frame == 1)
-      ffmpeg->pkt.flags |= AV_PKT_FLAG_KEY;
 
     return 0;
 
@@ -373,9 +371,6 @@ static int ffmpeg_encode_video(struct ffmpeg *ffmpeg){
         return -2;
     }
 
-    if (ffmpeg->picture->key_frame == 1)
-      ffmpeg->pkt.flags |= AV_PKT_FLAG_KEY;
-
     return 0;
 
 #else
@@ -399,12 +394,16 @@ static int ffmpeg_encode_video(struct ffmpeg *ffmpeg){
         return -2;
     }
 
+    // Encoder did not provide metadata, set it up manually
     ffmpeg->pkt.size = retcd;
     ffmpeg->pkt.data = video_outbuf;
 
     if (ffmpeg->picture->key_frame == 1)
       ffmpeg->pkt.flags |= AV_PKT_FLAG_KEY;
 
+    ffmpeg->pkt.pts = ffmpeg->picture->pts;
+    ffmpeg->pkt.dts = ffmpeg->pkt.pts;
+
     free(video_outbuf);
 
     return 0;
