commit c20e58b27aaa24b1a800e3cf73de1d3311d8987f
Author: Joo Aun Saw <jasaw@dius.com.au>
Date:   Mon Aug 19 16:38:59 2019 +1000

    enable h264_omx encoder

diff --git a/ffmpeg.c b/ffmpeg.c
index a3a8701..b9b58de 100644
--- a/ffmpeg.c
+++ b/ffmpeg.c
@@ -597,7 +597,7 @@ static int ffmpeg_codec_is_blacklisted(const char *codec_name){
          * - remove the "h264_omx" from this blacklist.
          * More information: https://github.com/Motion-Project/motion/issues/433
          */
-        "h264_omx",
+        //"h264_omx",
 #if (LIBAVCODEC_VERSION_MAJOR < 57)
         /* h264_v4l2m2m does not exist on old versions of ffmpeg. */
         "h264_v4l2m2m",
