Add dynamic linker support for uClibc

Signed-off-by: Waldemar Brodkorb <wbx@uclibc-ng.org>

diff -Nur gcc-4.9.3.orig/gcc/config/microblaze/linux.h gcc-4.9.3/gcc/config/microblaze/linux.h
--- gcc-4.9.3.orig/gcc/config/microblaze/linux.h	2016-06-04 21:21:09.430646655 +0200
+++ gcc-4.9.3/gcc/config/microblaze/linux.h	2016-06-04 21:21:44.596003509 +0200
@@ -36,10 +36,13 @@
 
 #undef MUSL_DYNAMIC_LINKER
 #define MUSL_DYNAMIC_LINKER "/lib/ld-musl-microblaze" MUSL_DYNAMIC_LINKER_E ".so.1"
+#define UCLIBC_DYNAMIC_LINKER "/lib/ld-uClibc.so.0"
 #define GLIBC_DYNAMIC_LINKER "/lib/ld.so.1"
 
 #if DEFAULT_LIBC == LIBC_MUSL
 #define DYNAMIC_LINKER MUSL_DYNAMIC_LINKER
+#elif DEFAULT_LIBC == LIBC_UCLIBC
+#define DYNAMIC_LINKER UCLIBC_DYNAMIC_LINKER
 #else
 #define DYNAMIC_LINKER GLIBC_DYNAMIC_LINKER
 #endif
