--- busybox-1.20.0/findutils/find.c
+++ busybox-1.20.0-find/findutils/find.c
@@ -831,6 +831,11 @@ static action*** parse_params(char **arg
 	                        PARM_name      ,
 	                        PARM_iname     ,
 	IF_FEATURE_FIND_PATH(   PARM_path      ,)
+#if ENABLE_DESKTOP
+	/* -wholename is a synonym for -path */
+	/* We support it because Linux kernel's "make tags" uses it */
+	IF_FEATURE_FIND_PATH(   PARM_wholename ,)
+#endif
 	IF_FEATURE_FIND_PATH(   PARM_ipath     ,)
 	IF_FEATURE_FIND_REGEX(  PARM_regex     ,)
 	IF_FEATURE_FIND_TYPE(   PARM_type      ,)
@@ -869,6 +874,9 @@ static action*** parse_params(char **arg
 	                         "-name\0"
 	                         "-iname\0"
 	IF_FEATURE_FIND_PATH(   "-path\0"   )
+#if ENABLE_DESKTOP
+	IF_FEATURE_FIND_PATH(   "-wholename\0")
+#endif
 	IF_FEATURE_FIND_PATH(   "-ipath\0"  )
 	IF_FEATURE_FIND_REGEX(  "-regex\0"  )
 	IF_FEATURE_FIND_TYPE(   "-type\0"   )
@@ -1076,7 +1084,7 @@ static action*** parse_params(char **arg
 			ap->iname = (parm == PARM_iname);
 		}
 #if ENABLE_FEATURE_FIND_PATH
-		else if (parm == PARM_path || parm == PARM_ipath) {
+		else if (parm == PARM_path IF_DESKTOP(|| parm == PARM_wholename) || parm == PARM_ipath) {
 			action_path *ap;
 			dbg("%d", __LINE__);
 			ap = ALLOC_ACTION(path);
