--- linux/arch/mips/Makefile.orig	2006-01-02 20:21:10.000000000 -0700
+++ linux/arch/mips/Makefile	2006-03-19 17:25:29.000000000 -0700
@@ -22,15 +22,15 @@
 # Select the object file format to substitute into the linker script.
 #
 ifdef CONFIG_CPU_LITTLE_ENDIAN
-32bit-tool-prefix	= mipsel-linux-
-64bit-tool-prefix	= mips64el-linux-
+#32bit-tool-prefix	= mipsel-linux-
+#64bit-tool-prefix	= mips64el-linux-
 32bit-bfd		= elf32-tradlittlemips
 64bit-bfd		= elf64-tradlittlemips
 32bit-emul		= elf32ltsmip
 64bit-emul		= elf64ltsmip
 else
-32bit-tool-prefix	= mips-linux-
-64bit-tool-prefix	= mips64-linux-
+#32bit-tool-prefix	= mips-linux-
+#64bit-tool-prefix	= mips64-linux-
 32bit-bfd		= elf32-tradbigmips
 64bit-bfd		= elf64-tradbigmips
 32bit-emul		= elf32btsmip
@@ -39,18 +39,18 @@
 
 ifdef CONFIG_32BIT
 gcc-abi			= 32
-tool-prefix		= $(32bit-tool-prefix)
+#tool-prefix		= $(32bit-tool-prefix)
 UTS_MACHINE		:= mips
 endif
 ifdef CONFIG_64BIT
 gcc-abi			= 64
-tool-prefix		= $(64bit-tool-prefix)
+#tool-prefix		= $(64bit-tool-prefix)
 UTS_MACHINE		:= mips64
 endif
 
-ifdef CONFIG_CROSSCOMPILE
-CROSS_COMPILE		:= $(tool-prefix)
-endif
+#ifdef CONFIG_CROSSCOMPILE
+#CROSS_COMPILE		:= $(tool-prefix)
+#endif
 
 CHECKFLAGS-y				+= -D__linux__ -D__mips__ \
 					   -D_ABIO32=1 \
