--- orig/drivers/ieee1394/nodemgr.c	2006-03-21 01:41:51.000000000 -0700
+++ linux-2.6.16/drivers/ieee1394/nodemgr.c	2006-03-21 01:41:51.000000000 -0700
@@ -1487,7 +1487,7 @@
 	/* If there is no bus manager then we should set the root node's
 	 * force_root bit to promote bus stability per the 1394
 	 * spec. (8.4.2.6) */
-	if (host->busmgr_id == 0xffff && host->node_count > 1)
+	if (host->busmgr_id == 0x3f && host->node_count > 1)
 	{
 		u16 root_node = host->node_count - 1;
 
