--- orig/drivers/md/md.c	2006-03-21 01:43:49.000000000 -0700
+++ linux-2.6.16/drivers/md/md.c	2006-03-21 01:43:49.000000000 -0700
@@ -4860,7 +4860,7 @@
 
 	if ((code == SYS_DOWN) || (code == SYS_HALT) || (code == SYS_POWER_OFF)) {
 
-		printk(KERN_INFO "md: stopping all md devices.\n");
+		//printk(KERN_INFO "md: stopping all md devices.\n");
 
 		ITERATE_MDDEV(mddev,tmp)
 			if (mddev_trylock(mddev)==0)
