--- orig/drivers/scsi/scsi_ioctl.c	2005-12-03 02:10:27.000000000 -0700
+++ linux-2.6.14/drivers/scsi/scsi_ioctl.c	2005-12-03 02:10:27.000000000 -0700
@@ -374,6 +374,7 @@
 	if (!scsi_block_when_processing_errors(sdev))
 		return -ENODEV;
 
+#if 0
 	/* Check for deprecated ioctls ... all the ioctls which don't
 	 * follow the new unique numbering scheme are deprecated */
 	switch (cmd) {
@@ -389,6 +390,7 @@
 	default:
 		break;
 	}
+#endif
 
 	switch (cmd) {
 	case SCSI_IOCTL_GET_IDLUN:
