Drop ldconfig call, it's useless for cross-compiling.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura crda-3.18.orig/Makefile crda-3.18/Makefile
--- crda-3.18.orig/Makefile	2015-02-18 10:34:23.841259401 -0300
+++ crda-3.18/Makefile	2015-02-18 10:35:10.524201452 -0300
@@ -127,7 +127,6 @@
 	$(NQ) '  INSTALL  libreg'
 	$(Q)mkdir -p $(DESTDIR)/$(LIBDIR)
 	$(Q)cp $(LIBREG) $(DESTDIR)/$(LIBDIR)/
-	$(Q)ldconfig
 
 %.o: %.c regdb.h $(LIBREG)
 	$(NQ) '  CC  ' $@
