From 7982cccbb6ad58f3802bb266467290200d130a48 Mon Sep 17 00:00:00 2001
From: Ville Syrjala <syrjala@sci.fi>
Date: Sat, 27 Nov 2010 01:14:25 +0200
Subject: [PATCH] Too many '\' characters in the .pc file

---
 lite.pc.in |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/lite.pc.in b/lite.pc.in
index 48eb969..3a21eaa 100644
--- a/lite.pc.in
+++ b/lite.pc.in
@@ -9,4 +9,4 @@ Description: "LiTE" is a Toolkit Engine
 Requires: directfb >= @DFB_REQUIRED_VERSION@
 Version: @VERSION@
 Libs: -L${libdir} -llite
-Cflags: -I${includedir} -DLITEFONTDIR=\\\"@LITEFONTDIR@\\\"
+Cflags: -I${includedir} -DLITEFONTDIR=\"@LITEFONTDIR@\"
--
1.7.1

