--- package/lite/no-examples-no-tests.patch	(revision 0)
+++ package/lite/no-examples-no-tests.patch	(revision 0)
@@ -0,0 +1,11 @@
--- Makefile.in.orig	2008-07-07 13:07:12.000000000 +0200
+++ Makefile.in	2008-07-07 13:07:05.000000000 +0200
@@ -202,7 +202,7 @@
 target_vendor = @target_vendor@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = data fonts lite leck examples tests
+SUBDIRS = data fonts lite leck examples
 EXTRA_DIST = TODO lite.pc.in leck.pc.in
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = lite.pc leck.pc
