From b060873f444d79f3f5aa388427cb1ede6665834a Mon Sep 17 00:00:00 2001
From: Kent Yoder <key@linux.vnet.ibm.com>
Date: Mon, 5 Nov 2012 14:58:31 -0600
Subject: [PATCH] configure.in: remove -Werror

Signed-off-by: Kent Yoder <key@linux.vnet.ibm.com>
Signed-off-by: Noe Rubinstein <nrubinstein@aldebaran.com>
---
 configure.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.in b/configure.in
index da170d0..0f175be 100644
--- a/configure.in
+++ b/configure.in
@@ -138,7 +138,7 @@ AC_TYPE_SIZE_T
 AC_TYPE_SIGNAL
 AC_TYPE_UID_T
 
-CFLAGS="$CFLAGS -Wall -Werror -Wreturn-type -Wsign-compare"
+CFLAGS="$CFLAGS -Wall -Wreturn-type -Wsign-compare"
 
 AC_CONFIG_FILES(./Makefile		\
 		po/Makefile.in		\
-- 
2.1.4

