autotools: fix install directories

Signed-off-by: Francois Perrad <francois.perrad@gadz.org>

Index: b/Makefile.am
===================================================================
--- a/Makefile.am
+++ b/Makefile.am
@@ -3,7 +3,7 @@
 SUBDIRS = src \
 	  doc
 
-pkgconfigdir = $(libdir)/pkgconfig
+pkgconfigdir = $(prefix)/lib/pkgconfig
 pkgconfig_DATA = luacrypto.pc
 
 test: all
Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -31,7 +31,7 @@
 PKG_CHECK_MODULES([LUA], [lua])
 
 # lua libdir
-LUALIBDIR="`$PKGCONFIG --variable=libdir lua`"
+LUALIBDIR="`$PKGCONFIG --variable=INSTALL_CMOD lua`"
 
 # dest of headers
 CRYPTOINC="${includedir}/${PACKAGE_NAME}"
