diff -ur hal-0.5.8/hald/linux/classdev.c hal-0.5.8-patched/hald/linux/classdev.c
--- hal-0.5.8/hald/linux/classdev.c	2006-09-11 18:32:56.000000000 -0500
+++ hal-0.5.8-patched/hald/linux/classdev.c	2007-01-13 22:00:46.896769548 -0600
@@ -32,7 +32,10 @@
 #include <ctype.h>
 #include <limits.h>
 #include <linux/types.h>
-#include <net/if_arp.h> /* for ARPHRD_... */
+#include <linux/if_arp.h> /* for ARPHRD_... */
+#ifndef ARPHRD_IEEE80211_RADIOTAP
+# define ARPHRD_IEEE80211_RADIOTAP 803
+#endif
 #include <stdlib.h>
 #include <string.h>
 #include <stdint.h>
