Disable addition of RPATH to the generated libraries

Signed-off-by: Simon Dawson <spdawson@gmail.com>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

diff -Nurp a/SConstruct b/SConstruct
--- a/SConstruct	2014-08-23 18:41:09.000000000 +0100
+++ b/SConstruct	2014-08-25 09:46:53.754364989 +0100
@@ -300,9 +300,6 @@ if env["sysroot"]:
 
 # Don't hack RPATH unless libdir points somewhere that is not on the
 # minimum default load path.
-if env["shared"]:
-    if env["libdir"] not in ["/usr/lib", "/lib"]:
-        env.Prepend(RPATH=[installdir('libdir')])
 
 # Give deheader a way to set compiler flags
 if 'MORECFLAGS' in os.environ:
