diff -urN mtd-utils-1.1.0-0rig/compr_lzo.c mtd-utils-1.1.0/compr_lzo.c
--- mtd-utils-1.1.0-0rig/compr_lzo.c	2007-08-03 15:55:51.000000000 +0200
+++ mtd-utils-1.1.0/compr_lzo.c	2007-09-27 08:18:05.000000000 +0200
@@ -26,7 +26,7 @@
 #include <string.h>
 #include <asm/types.h>
 #include <linux/jffs2.h>
-#include <lzo/lzo1x.h>
+#include <lzo1x.h>
 #include "compr.h"
 
 extern int page_size;
 $(BUILDDIR)/flash_eraseall: $(BUILDDIR)/crc32.o $(BUILDDIR)/flash_eraseall.o
 	$(CC) $(LDFLAGS) -o $@ $^
