diff -Nura mtd-utils-1.2.0/Makefile mtd-utils-1.2.0-noubi/Makefile
--- mtd-utils-1.2.0/Makefile	2008-06-27 13:21:28.000000000 -0300
+++ mtd-utils-1.2.0-noubi/Makefile	2009-12-23 10:26:25.000000000 -0300
@@ -45,7 +45,6 @@
 .SUFFIXES:
 
 all: $(TARGETS)
-	make -C $(BUILDDIR)/ubi-utils
 
 IGNORE=${wildcard $(BUILDDIR)/.*.c.dep}
 -include ${IGNORE}
@@ -53,7 +52,6 @@
 clean:
 	rm -f $(BUILDDIR)/*.o $(TARGETS) $(BUILDDIR)/.*.c.dep $(SYMLINKS)
 	if [ "$(BUILDDIR)x" != ".x" ]; then rm -rf $(BUILDDIR); fi
-	make -C $(BUILDDIR)/ubi-utils clean
 
 $(SYMLINKS):
 	ln -sf ../fs/jffs2/$@ $@
@@ -95,4 +93,3 @@
 	install -m0755 ${TARGETS} ${DESTDIR}/${SBINDIR}/
 	mkdir -p ${DESTDIR}/${MANDIR}/man1
 	gzip -9c mkfs.jffs2.1 > ${DESTDIR}/${MANDIR}/man1/mkfs.jffs2.1.gz
-	make -C $(BUILDDIR)/ubi-utils install
