From 7598eeaa1defa2884adaa890bb115c493d69cc35 Mon Sep 17 00:00:00 2001
From: Bernhard Reutner-Fischer <rep.dot.nop@gmail.com>
Date: Sat, 23 Feb 2013 06:39:41 +0000
Subject: test: cater for config

Signed-off-by: Bernhard Reutner-Fischer <rep.dot.nop@gmail.com>
---
(limited to 'test')

diff --git a/test/math/Makefile.in b/test/math/Makefile.in
index e76cbdb..beef650 100644
--- a/test/math/Makefile.in
+++ b/test/math/Makefile.in
@@ -1,14 +1,16 @@
 # uClibc math tests
 # Licensed under the LGPL v2.1, see the file COPYING.LIB in this tarball.
 
-TESTS := basic-test tst-definitions test-fpucw test-float test-ifloat test-double test-idouble \
-    rint signgam ilogb
 # gamma (removed from TESTS, need to add "small errors are ok" machinery there)
-ifeq ($(UCLIBC_HAS_LONG_DOUBLE_MATH),y)
-TESTS += test-ldouble test-ildoubl compile_test c99_test
+TESTS_DISABLED := gamma
+ifeq ($(UCLIBC_HAS_LONG_DOUBLE_MATH),)
+TESTS_DISABLED += test-ldouble test-ildoubl compile_test c99_test
 else
 CFLAGS_basic-test := -DNO_LONG_DOUBLE
 endif
+ifeq ($(DO_C99_MATH),)
+TESTS_DISABLED += test-float test-ifloat test-double test-idouble rint signgam ilogb
+endif
 
 DODIFF_rint     := 1
 DODIFF_signgam  := 1
diff --git a/test/misc/Makefile.in b/test/misc/Makefile.in
index 9b74d22..52a3e71 100644
--- a/test/misc/Makefile.in
+++ b/test/misc/Makefile.in
@@ -4,6 +4,7 @@
 TESTS_DISABLED := outb tst-fnmatch bug-glob1 tst-gnuglob
 ifeq  ($(UCLIBC_HAS_LFS),)
 TESTS_DISABLED += dirent64
+TESTS_DISABLED += tst-statfs # assuming host has LFS on
 endif
 CFLAGS_dirent64 := -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64
 
diff --git a/test/misc/tst-statfs.c b/test/misc/tst-statfs.c
index 44ab3aa..b8b4229 100644
--- a/test/misc/tst-statfs.c
+++ b/test/misc/tst-statfs.c
@@ -1,5 +1,3 @@
-#define _FILE_OFFSET_BITS 64
-
 #include <sys/vfs.h>
 #include <errno.h>
 #include <stdio.h>
diff --git a/test/misc/tst-statvfs.c b/test/misc/tst-statvfs.c
index c1e8fde..4b67719 100644
--- a/test/misc/tst-statvfs.c
+++ b/test/misc/tst-statvfs.c
@@ -1,5 +1,3 @@
-#define _FILE_OFFSET_BITS 64
-
 #include <sys/statvfs.h>
 #include <errno.h>
 #include <stdio.h>
diff --git a/test/unistd/Makefile.in b/test/unistd/Makefile.in
index 24b9a37..cfef22e 100644
--- a/test/unistd/Makefile.in
+++ b/test/unistd/Makefile.in
@@ -5,7 +5,7 @@ ifeq ($(UCLIBC_HAS_LFS),)
 TESTS_DISABLED := tst-preadwrite64 tst-posix_fallocate64
 endif
 ifeq ($(UCLIBC_HAS_ADVANCED_REALTIME),)
-TESTS_DISABLED := tst-posix_fallocate
+TESTS_DISABLED += tst-posix_fallocate
 endif
 OPTS_getopt      := -abcXXX -9
 OPTS_getopt_long := --add XXX --delete YYY --verbose
--
cgit v0.9.1
