--- MPlayer-1.0rc1-orig/configure	2007-06-01 18:15:59.000000000 +0200
+++ MPlayer-1.0rc1/configure	2007-06-01 18:17:38.000000000 +0200
@@ -3803,7 +3803,7 @@ fi
 
 
 echocheck "X11 headers presence"
-  for I in `echo $_inc_extra | sed s/-I//g` /usr/X11/include /usr/X11R6/include /usr/include/X11R6 /usr/include /usr/openwin/include ; do
+  for I in `echo $_inc_extra | sed s/-I//g` ; do
     if test -f "$I/X11/Xlib.h" ; then
       _inc_x11="-I$I"
       _x11_headers="yes"
