From f4395ff932758884416753f88f5d98f33ec67479 Mon Sep 17 00:00:00 2001
From: Spenser Gilliland <spenser@gillilanding.com>
Date: Mon, 29 Dec 2014 13:54:54 +0100
Subject: [PATCH 3/5] Add stack direction for the Microblaze architecture

Signed-off-by: Spenser Gilliland <spenser@gillilanding.com>
Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 m4/stack-direction.m4 | 1 +
 1 file changed, 1 insertion(+)

diff --git a/m4/stack-direction.m4 b/m4/stack-direction.m4
index 3f70530..53f4216 100644
--- a/m4/stack-direction.m4
+++ b/m4/stack-direction.m4
@@ -32,6 +32,7 @@ AC_DEFUN([SV_STACK_DIRECTION],
       m68* | \
       m88k | \
       mcore | \
+      microblaze | \
       mips* | \
       mmix | \
       mn10200 | \
-- 
2.1.0

