diff -urN vpnc-0.4.0-0rig/Makefile vpnc-0.4.0/Makefile
--- vpnc-0.4.0-0rig/Makefile	2007-02-19 21:51:12.000000000 +0100
+++ vpnc-0.4.0/Makefile	2007-07-17 19:59:53.000000000 +0200
@@ -35,7 +35,7 @@
 RELEASE_VERSION := $(shell cat VERSION)
 
 CC=gcc
-CFLAGS += -W -Wall -O3 -Wmissing-declarations -Wwrite-strings -g
+CFLAGS += -W -Wall -O3 -Wmissing-declarations -Wwrite-strings -g -I$(INCLUDE)
 CPPFLAGS = -DVERSION=\"$(VERSION)\"
 LDFLAGS = -g $(shell libgcrypt-config --libs)
 CFLAGS +=  $(shell libgcrypt-config --cflags)
