diff -rdup vpnc-0.5.1.oorig/config.c vpnc-0.5.1/config.c
--- vpnc-0.5.1.oorig/config.c	2007-09-10 22:39:48.000000000 +0200
+++ vpnc-0.5.1/config.c	2007-09-19 14:19:46.000000000 +0200
@@ -538,7 +538,7 @@ static char *get_config_filename(const c
 {
 	char *realname;
 	
-	asprintf(&realname, "%s%s%s", index(name, '/') ? "" : "/etc/vpnc/", name, add_dot_conf ? ".conf" : "");
+	asprintf(&realname, "%s%s%s", strchr(name, '/') ? "" : "/etc/vpnc/", name, add_dot_conf ? ".conf" : "");
 	return realname;
 }
 
