Fix musl build

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr w_scan-20141122.org/descriptors.h w_scan-20141122/descriptors.h
--- w_scan-20141122.org/descriptors.h	2014-11-22 09:49:26.000000000 +0100
+++ w_scan-20141122/descriptors.h	2016-01-24 14:05:21.000000000 +0100
@@ -28,6 +28,7 @@
 #ifndef __DESCRIPTORS_H__
 #define __DESCRIPTORS_H__
 
+#include <time.h>
 #include "extended_frontend.h"
 
 /******************************************************************************
diff -uNr w_scan-20141122.org/version.h w_scan-20141122/version.h
--- w_scan-20141122.org/version.h	2014-11-22 11:59:56.000000000 +0100
+++ w_scan-20141122/version.h	2016-01-24 14:06:33.000000000 +0100
@@ -1,6 +1,6 @@
 /* AUTOMATICALLY GENERATED - DO NOT EDIT MANUALLY */
 #ifndef W_SCAN_VERSION_H
 #define W_SCAN_VERSION_H
-uint version=20141122;
+unsigned int version=20141122;
 #endif
 
