diff -upr dnsmasq-2.39/src/dnsmasq.h dnsmasq-2.39-orig/src/dnsmasq.h
--- dnsmasq-2.39/src/dnsmasq.h	2007-04-20 12:53:38.000000000 +0200
+++ dnsmasq-2.39-orig/src/dnsmasq.h	2007-06-12 11:12:19.000000000 +0200
@@ -79,7 +79,7 @@
 /* There doesn't seem to be a universally-available 
    userpace header for this. */
 extern int capset(cap_user_header_t header, cap_user_data_t data);
-#include <sys/prctl.h>
+#include <linux/prctl.h>
 #endif
 
 /* Min buffer size: we check after adding each record, so there must be 
