diff -urN linux-2.6.21.1-0rig/drivers/video/atmel_lcdfb.c linux-2.6.21.1/drivers/video/atmel_lcdfb.c
--- linux-2.6.21.1-0rig/drivers/video/atmel_lcdfb.c	2007-05-12 13:28:34.000000000 +0200
+++ linux-2.6.21.1/drivers/video/atmel_lcdfb.c	2007-05-12 19:41:45.000000000 +0200
@@ -540,7 +540,7 @@
 	info->fix = atmel_lcdfb_fix;
 
 	/* Enable LCDC Clocks */
-	if (cpu_is_at91sam9261() {
+	if (cpu_is_at91sam9261()) {
 		sinfo->bus_clk = clk_get(dev, "hck1");
 		if (IS_ERR(sinfo->bus_clk)) {
 			ret = PTR_ERR(sinfo->bus_clk);
diff -urN linux-2.6.21.1-0rig/include/asm-arm/arch-at91/cpu.h linux-2.6.21.1/include/asm-arm/arch-at91/cpu.h
--- linux-2.6.21.1-0rig/include/asm-arm/arch-at91/cpu.h	2007-05-12 13:28:34.000000000 +0200
+++ linux-2.6.21.1/include/asm-arm/arch-at91/cpu.h	2007-05-12 13:27:28.000000000 +0200
@@ -17,7 +17,7 @@
 #include <asm/arch/at91_dbgu.h>
 
 
-#define ARCH_ID_AT91RM9200	0x09290780
+#define ARCH_ID_AT91RM9200		0x09290780
 #define ARCH_ID_AT91SAM9260	0x019803a0
 #define ARCH_ID_AT91SAM9261	0x019703a0
 #define ARCH_ID_AT91SAM9263	0x019607a0
@@ -76,4 +76,11 @@
 #define cpu_is_at91sam9rl()	(0)
 #endif
 
+#define cpu_is_at32ap7000()	(0)
+#define cpu_is_at32ap7001()	(0)
+#define cpu_is_at32ap7002()	(0)
+#define cpu_is_at32ap7200()	(0)
+#define cpu_is_at32ap7010()	(0)
+#define cpu_is_at32ap7020()	(0)
+
 #endif
