From 44f80bce0914d4d9c5e7486c7cdb521ba637b324 Mon Sep 17 00:00:00 2001
From: Tzu-Jung Lee <tjlee@ambarella.com>
Date: Fri, 19 Jul 2013 22:45:18 +0800
Subject: [PATCH] build: get along with buildroot

Signed-off-by: Tzu-Jung Lee <tjlee@ambarella.com>
---
 Makefile | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/Makefile b/Makefile
index 2c3568e..98bb994 100644
--- a/Makefile
+++ b/Makefile
@@ -34,12 +34,6 @@ SHELL = /bin/sh
 .SUFFIXES:
 .SUFFIXES: .c .o
 
-ifdef CROSS_COMPILE
-CC = $(CROSS_COMPILE)gcc
-else
-CC = gcc
-endif
-
 # Use WARN=1 periodically to get too many warnings...
 ifdef WARN
 WARNING_FLAGS = -Wall -W -Wfloat-equal -Wundef -Wshadow -Wpointer-arith -Wcast-qual -Wconversion -Wmissing-prototypes -Wmissing-declarations -Wunreachable-code -Winline
@@ -452,3 +446,9 @@ test_lists:	$(BINDIR)/test_nal_unit_list  $(BINDIR)/test_es_unit_list
 	@echo +++ Testing ES unit lists
 	$(BINDIR)/test_es_unit_list
 	@echo +++ Test succeeded
+
+.PHONY: install
+install:
+	@for i in $(PROGS); do \
+		cp -f $$i $(DESTDIR)/usr/bin/; \
+	done
-- 
1.8.3.2

