From ed649ccb8ddc724546172d329fca39b23c49b804 Mon Sep 17 00:00:00 2001
From: "Arnout Vandecappelle (Essensium/Mind)" <arnout@mind.be>
Date: Mon, 17 Dec 2012 17:58:28 +0100
Subject: [PATCH] gplay: add missing include

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
---
 tools/gplay/gst_snapshot.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/gplay/gst_snapshot.c b/tools/gplay/gst_snapshot.c
index 40dbb6f..596f399 100755
--- a/tools/gplay/gst_snapshot.c
+++ b/tools/gplay/gst_snapshot.c
@@ -22,6 +22,7 @@
 
 #include <gst/gst.h>
 #include <string.h>
+#include <stdio.h>
 #include "gst_snapshot.h"
 
 static void
-- 
