From 53c7e8bc75f0f27752309ddae49e8d3b867a7681 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@gmail.com>
Date: Mon, 15 Aug 2016 18:20:47 +0200
Subject: [PATCH] Add syscall-number for sched_(gs)etattr() for SH4

Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 src/include/rt-sched.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/include/rt-sched.h b/src/include/rt-sched.h
index 679c4bd..83394c6 100644
--- a/src/include/rt-sched.h
+++ b/src/include/rt-sched.h
@@ -51,6 +51,16 @@
 #endif
 #endif
 
+/* SCHED_DEADLINE available on sh4 since kernel 4.8 */
+#ifdef __SH4__
+#ifndef __NR_sched_setattr
+#define __NR_sched_setattr		370
+#endif
+#ifndef __NR_sched_getattr
+#define __NR_sched_getattr		369
+#endif
+#endif
+
 #ifdef __tilegx__
 #define __NR_sched_setattr		274
 #define __NR_sched_getattr		275
-- 
2.5.5

