diff -rdup vsftpd-2.0.5.orig/sysutil.c vsftpd-2.0.5/sysutil.c
--- vsftpd-2.0.5.orig/sysutil.c	2006-07-03 18:26:40.000000000 +0200
+++ vsftpd-2.0.5/sysutil.c	2006-12-14 15:13:15.000000000 +0100
@@ -16,11 +16,15 @@
 #include "utility.h"
 #include "tunables.h"
 
+#include "builddefs.h"
+
+#if !defined __UCLIBC__ || (defined __UCLIBC__ && defined __UCLIBC_HAS_LFS__)
 /* Activate 64-bit file support on Linux/32bit plus others */
 #define _FILE_OFFSET_BITS 64
 #define _LARGEFILE_SOURCE 1
 #define _LARGEFILE64_SOURCE 1
 #define _LARGE_FILES 1
+#endif
 
 /* For Linux, this adds nothing :-) */
 #include "port/porting_junk.h"
