From 1831b618fbedf0bcf15b0465981d133d9e3c53ee Mon Sep 17 00:00:00 2001
From: Jean Delvare <jdelvare@suse.de>
Date: Wed, 6 Dec 2017 10:46:56 +0100
Subject: lib/Module.mk: Drop unused variable LIB_OBJECTS

Signed-off-by: Angelo Compagnucci <angelo@amarulasolutions.com>
---
 lib/Module.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/lib/Module.mk b/lib/Module.mk
index 44fa938..8a58f5b 100644
--- a/lib/Module.mk
+++ b/lib/Module.mk
@@ -28,7 +28,6 @@ LIB_SHLIBNAME	:= $(LIB_SHBASENAME).$(LIB_VER)
 LIB_STLIBNAME	:= libi2c.a
 
 LIB_LINKS	:= $(LIB_SHSONAME) $(LIB_SHBASENAME)
-LIB_OBJECTS	:= smbus.o
 
 LIB_TARGETS	:=
 ifeq ($(BUILD_DYNAMIC_LIB),1)
@@ -36,7 +35,6 @@ LIB_TARGETS	+= $(LIB_SHLIBNAME)
 endif
 ifeq ($(BUILD_STATIC_LIB),1)
 LIB_TARGETS	+= $(LIB_STLIBNAME)
-LIB_OBJECTS	+= smbus.ao
 endif
 
 #
-- 
2.7.4

