---
 src/df_knuckles/matrix.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: directfb-examples-1.2.0/src/df_knuckles/matrix.c
===================================================================
--- directfb-examples-1.2.0.orig/src/df_knuckles/matrix.c
+++ directfb-examples-1.2.0/src/df_knuckles/matrix.c
@@ -19,7 +19,7 @@
 static float Cosine[3600];
 static float Sine[3600];
 
-#define M_CLEAR(m) bzero(m, MATRIX_SIZE)
+#define M_CLEAR(m) memset(m, 0, MATRIX_SIZE)
 #define M_IDENTITY(m) memcpy(m, IdentityMatrix, MATRIX_SIZE)
 
 static void MultiplyMatrix(float *A, float *B)
