---
 Makefile |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: usb_modeswitch-1.0.7/Makefile
===================================================================
--- usb_modeswitch-1.0.7.orig/Makefile
+++ usb_modeswitch-1.0.7/Makefile
@@ -1,8 +1,8 @@
 PROG        = usb_modeswitch
 VERS        = 1.0.7
 STRIP	    = strip
-CC          = gcc
-CCFLAGS     = -l usb -Wall
+CC          ?= gcc
+CFLAGS      += -l usb -Wall
 RM          = /bin/rm -f
 OBJS        = usb_modeswitch.c
 PREFIX	    = $(DESTDIR)/usr
@@ -12,8 +12,7 @@
 .PHONY:     clean
 all:        $(PROG)
 $(PROG):    $(OBJS)
-	      $(CC) $(CCFLAGS) -o $(PROG) $(OBJS)
-	      $(STRIP) $(PROG)
+	      $(CC) $(CFLAGS) -o $(PROG) $(OBJS)
 
 clean:
 	      $(RM) ./usb_modeswitch
