--- xc/config/cf/host.def~	Thu May 12 15:28:01 MDT 2005
+++ xc/config/cf/host.def	Thu May 12 15:28:01 MDT 2005
@@ -0,0 +1,38 @@
+#define KDriveXServer		YES
+#define TinyXServer		YES
+#define XfbdevServer		YES
+#define CrossCompiling		YES
+#define ProjectRoot             /usr/X11R6
+#define NothingOutsideProjectRoot	YES
+
+#define HasFreetype2		NO
+#define HasFontconfig		NO
+#define HasLibpng		YES
+#define HasZlib			YES
+
+#define HasExpat		NO
+#define XprtServer		NO
+#define BuildXprintClients	NO
+#define TouchScreen		NO
+#define HasTsLib		NO
+#define BuildDocs		NO
+#define BuildXF86DRI		NO
+#define BuildGlxExt		NO
+#define BuildOSMesaLib		NO
+#define BuildLBX		NO
+#define BuildLowMem		NO
+#define BuildIPv6		NO
+#define HasPam			NO
+#define HasPamMisc		NO
+
+/*
+#define XF86CardDrivers		mga glint nv tga s3virge sis rendition \
+				neomagic i740 tdfx savage \
+				cirrus vmware tseng trident chips apm \
+				GlideDriver fbdev i128 \
+				ati AgpGartDrivers DevelDrivers ark \
+				cyrix siliconmotion vesa vga \
+				XF86OSCardDrivers XF86ExtraCardDrivers
+*/
+
+#define XInputDrivers		mouse keyboard
--- xc/config/cf/cross.def~	Thu May 12 15:28:01 MDT 2005
+++ xc/config/cf/cross.def	Thu May 12 15:28:01 MDT 2005
@@ -1,45 +1,45 @@
-/* $XFree86: xc/config/cf/cross.def,v 1.1 2000/12/08 22:09:34 keithp Exp $ */
+/* $XFree86: xc/config/cf/cross.def,v 1.2 2001/03/30 02:15:17 keithp Exp $ */
 /*
  * This file contains redefinitions of some symbols to enable
  * cross compilation: e.g. paths for include files and paths to
  * compiler images.  It will have to be edited to reflect these
  * given your local configuration.
  */
-#if 0
-#undef i386Architecture
+#define i386Architecture
-#define Arm32Architecture
+#undef Arm32Architecture
 
 #undef OptimizedCDebugFlags
-#define OptimizedCDebugFlags 	-O2
-#define ServerCDebugFlags	-O2
+#define OptimizedCDebugFlags 	-Os
+#define ServerCDebugFlags	-Os
 #undef StandardDefines
-#define StandardDefines 	-Dlinux -D__arm__ -D_POSIX_SOURCE \
+#define StandardDefines 	-Dlinux -D__REPLACE_ARCH__ -D_POSIX_SOURCE \
 				-D_BSD_SOURCE -D_GNU_SOURCE -DX_LOCALE
 #undef CcCmd
-#define StdIncDir /skiff/local/arm-linux/include
-#define PreIncDir 
+#define StdIncDir REPLACE_STAGING_DIR/include
+#define PreIncDir
 #undef PostIncDir
-#define PostIncDir /skiff/local/lib/gcc-lib/arm-linux/2.95.2/include
-#define CcCmd /skiff/local/bin/arm-linux-gcc
+#define PostIncDir REPLACE_STAGING_DIR/lib/gcc-lib/REPLACE_ARCH-linux-uclibc/3.3.3/include
+#define CcCmd REPLACE_STAGING_DIR/bin/REPLACE_ARCH-linux-gcc
 #undef CplusplusCmd
-#define HasCplusplus YES
-#define CplusplusCmd /skiff/local/bin/arm-linux-g++
+#define HasCplusplus NO
+#define CplusplusCmd NO
 #define DoRanlibCmd YES
-#define RanlibCmd /skiff/local/bin/arm-linux-ranlib
+#define RanlibCmd REPLACE_STAGING_DIR/bin/REPLACE_ARCH-linux-ranlib
 #undef ExtraLoadFlags
 #define ExtraLoadFlags
 #define FbNoPixelAddrCode
 #undef TermcapLibrary
-#define TermcapLibrary -ltermcap
+#define TermcapLibrary -lncurses
 
 #undef LdPostLib
-#define LdPostLib	-L/skiff/local/arm-linux/lib
+#define LdPostLib	-LREPLACE_STAGING_DIR/REPLACE_ARCH-linux-uclibc/lib
+#undef LdCmd
+#define LdCmd REPLACE_STAGING_DIR/bin/REPLACE_ARCH-linux-ld
 
 #undef ExtensionOSDefines
 #define ExtensionOSDefines
 
 #define ServerXdmcpDefines	/**/
 
 #define HostCcCmd  cc
-#endif
 #include <cross.rules>
