--- xc/programs/Xserver/fb/Imakefile.orig	2006-07-20 11:41:36.000000000 -0600
+++ xc/programs/Xserver/fb/Imakefile	2006-07-20 11:43:50.000000000 -0600
@@ -3,16 +3,6 @@
 XCOMM 
 XCOMM Id: Imakefile,v 1.1 1999/11/02 03:54:44 keithp Exp $
 
-#if defined(i386Architecture) && defined(HasGcc34) && HasGcc34
-MMXOPTIONS= -mmmx -Winline --param inline-unit-growth=10000 \
-	--param large-function-growth=10000 -DUSE_GCC34_MMX
-
-SpecialCObjectRule(fbmmx,fbmmx.c,$(MMXOPTIONS))
-SpecialCObjectRule(fbpict,fbpict.c,$(MMXOPTIONS))
-SpecialCObjectRule(fbfill,fbfill.c,$(MMXOPTIONS))
-
-#endif
-
 #if DoLoadableServer
 #if !BuildModuleInSubdir
 #define IHaveModules
