--- xc/programs/xterm/Imakefile.orig	2004-08-13 14:57:19.000000000 +0200
+++ xc/programs/xterm/Imakefile	2004-09-20 22:04:49.505263720 +0200
@@ -93,7 +93,7 @@
 #endif
 
 #ifdef UTF8support
-    UTF8_OPTION = -DOPT_WIDE_CHARS -DOPT_LUIT_PROG
+    UTF8_OPTION = -DOPT_WIDE_CHARS -DOPT_LUIT_PROG -DOPT_256_COLORS
         UTF8SRC = charclass.c precompose.c wcwidth.c xutf8.c
         UTF8OBJ = charclass.o precompose.o wcwidth.o xutf8.o
 #endif
