Prefer getty to agetty in console setup systemd units

Signed-off-by: Maxime Ripard <maxime.ripard@free-electrons.com>
Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---

diff -aburN systemd-212.orig/units/getty@.service.m4 systemd-212/units/getty@.service.m4
--- systemd-212.orig/units/getty@.service.m4	2014-01-28 11:08:51.000000000 +0100
+++ systemd-212/units/getty@.service.m4	2014-03-26 11:06:27.000000000 +0100
@@ -27,7 +27,7 @@
 
 [Service]
 # the VT is cleared by TTYVTDisallocate
-ExecStart=-/sbin/agetty --noclear %I $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 RestartSec=0
diff -aburN systemd-212.orig/units/serial-getty@.service.m4 systemd-212/units/serial-getty@.service.m4
--- systemd-212.orig/units/serial-getty@.service.m4	2014-03-13 18:47:24.000000000 +0100
+++ systemd-212/units/serial-getty@.service.m4	2014-03-26 11:07:01.000000000 +0100
@@ -22,7 +22,7 @@
 IgnoreOnIsolate=yes
 
 [Service]
-ExecStart=-/sbin/agetty --keep-baud 115200,38400,9600 %I $TERM
+ExecStart=-/sbin/getty -L %I 115200 vt100
 Type=idle
 Restart=always
 RestartSec=0
