--- vtun-2.6/configure.dist	2004-03-11 10:39:10.000000000 -0600
+++ vtun-2.6/configure	2004-03-11 10:45:52.000000000 -0600
@@ -2112,7 +2112,7 @@
     echo $ac_n "checking "for blowfish.h"""... $ac_c" 1>&6
 echo "configure:2114: checking "for blowfish.h"" >&5 
     ac_hdr_found=no
-    for p in $BLOWFISH_HDR_DIR /usr/include/ssl /usr/include/openssl /usr/include /usr/local/include /usr/local/ssl/include  /usr/include/crypto; do
+    for p in $BLOWFISH_HDR_DIR $SSL_HDR_DIR /usr/include/ssl /usr/include/openssl /usr/include /usr/local/include /usr/local/ssl/include  /usr/include/crypto; do
 	if test -n "$p"; then
 	  dir="$p"
 	else

