Fix musl build

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr ptpd-ptpd-2.3.1.org/src/dep/net.c ptpd-ptpd-2.3.1/src/dep/net.c
--- ptpd-ptpd-2.3.1.org/src/dep/net.c	2015-06-29 17:13:29.000000000 +0200
+++ ptpd-ptpd-2.3.1/src/dep/net.c	2016-01-24 00:35:44.509860065 +0100
@@ -82,7 +82,6 @@
 #ifdef SO_TIMESTAMPING
 #include <linux/net_tstamp.h>
 #include <linux/sockios.h>
-#include <linux/ethtool.h>
 #endif /* SO_TIMESTAMPING */
 
 /**
