From 05ec5e4411fe3188f72305050b11513cb153c5ff Mon Sep 17 00:00:00 2001
From: Gary Bisson <gary.bisson@boundarydevices.com>
Date: Tue, 14 Feb 2017 09:39:02 +0100
Subject: [PATCH] NativeStateDRM: add imx-drm driver support

Signed-off-by: Gary Bisson <gary.bisson@boundarydevices.com>
---
 src/native-state-drm.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/native-state-drm.cpp b/src/native-state-drm.cpp
index 4df75ec..454a24d 100644
--- a/src/native-state-drm.cpp
+++ b/src/native-state-drm.cpp
@@ -201,6 +201,7 @@ NativeStateDRM::init()
     // driver (udev?).
     static const char* drm_modules[] = {
         "i915",
+        "imx-drm",
         "nouveau",
         "radeon",
         "vmgfx",
-- 
2.11.0

