From 6080c0b229c6e332d7dd609d9435ac9baeeea443 Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Thu, 30 Jan 2014 16:33:02 -0300
Subject: [PATCH] roken-h-process: use Getopt::Std, getopts.pl is deprecated

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 cf/roken-h-process.pl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/cf/roken-h-process.pl b/cf/roken-h-process.pl
index 153a35c..72d3bd3 100644
--- a/cf/roken-h-process.pl
+++ b/cf/roken-h-process.pl
@@ -1,10 +1,10 @@
 #!/usr/bin/perl
 
-require 'getopts.pl';
+use Getopt::Std;
 
 my $debug = 0;
 
-Getopts('dc:p:o:') || die "foo";
+getopts('dc:p:o:') || die "foo";
 
 if ($opt_d) {
     $debug = 1;
-- 
1.8.3.2

