From 4d315359d15e1221f3463b77a960a60093aac893 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@gmail.com>
Date: Thu, 22 Jun 2017 00:19:56 +0200
Subject: [PATCH] cmake: remove Werror

Signed-off-by: Romain Naour <romain.naour@gmail.com>
---
 cmake/AddCompilationFlags.cmake | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/cmake/AddCompilationFlags.cmake b/cmake/AddCompilationFlags.cmake
index 5159ff9..a506033 100644
--- a/cmake/AddCompilationFlags.cmake
+++ b/cmake/AddCompilationFlags.cmake
@@ -24,9 +24,6 @@ endif()
 # Be less pedantic in release builds for users.
 set(CMAKE_CXX_FLAGS_RELEASE "-Wno-error -Wall -Wextra -Wno-unknown-pragmas -Wno-fatal-errors ${CMAKE_CXX_FLAGS_RELEASE}")
 
-# Be more pedantic in debug mode for developers.
-set(CMAKE_CXX_FLAGS_DEBUG "-Werror -Wall -Wextra -pedantic ${CMAKE_CXX_FLAGS_DEBUG}")
-
 # Platform-specific flags.
 if(WIN32)
   # MinGW: disable the console by default.
-- 
2.9.4

