--- a/xrandr.c.orig	2008-02-02 17:09:45.000000000 -0500
+++ b/xrandr.c	2008-02-02 17:04:51.000000000 -0500
@@ -189,7 +189,7 @@ reflection_name (Rotation rotation)
 
 #if HAS_RANDR_1_2
 typedef enum _policy {
-    clone, extend
+    clone1, extend
 } policy_t;
 
 typedef enum _relation {
@@ -1602,7 +1602,7 @@ main (int argc, char **argv)
     int		ret = 0;
 #if HAS_RANDR_1_2
     output_t	*output = NULL;
-    policy_t	policy = clone;
+    policy_t	policy = clone1;
     Bool    	setit_1_2 = False;
     Bool    	query_1_2 = False;
     Bool	modeit = False;
@@ -1863,7 +1863,7 @@ main (int argc, char **argv)
 	    continue;
 	}
 	if (!strcmp ("--clone", argv[i])) {
-	    policy = clone;
+	    policy = clone1;
 	    setit_1_2 = True;
 	    continue;
 	}
