diff --git a/configure.ac b/configure.ac
index 06a0ac9..b8d1640 100644
--- a/configure.ac
+++ b/configure.ac
@@ -158,7 +158,10 @@ if test x$HAVE_GCC_ASM = xyes -a x$HAVE_POWERPC = xyes ; then
     true)
 fi
 
-if test x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes ; then
+AC_ARG_ENABLE(vfp,
+  AC_HELP_STRING([--enable-vfp],[compile with Vector Floating-point unit support]),
+  enable_vfp=$enableval,enable_vfp=yes)
+if test "x$enable-vfp" = xyes -a x$HAVE_GCC_ASM = xyes -a x$HAVE_ARM = xyes; then
   AS_COMPILER_FLAG(["-Wa,-mfpu=vfp"],
     [LIBOIL_CFLAGS="$LIBOIL_CFLAGS -mfpu=vfp"],
     true)
