configure: don't require GNU-specific files when running automake

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN libid3tag-0.15.1b.orig/configure.ac libid3tag-0.15.1b/configure.ac
--- libid3tag-0.15.1b.orig/configure.ac	2004-01-24 00:22:46.000000000 +0100
+++ libid3tag-0.15.1b/configure.ac	2018-11-25 15:31:04.184342212 +0100
@@ -26,7 +26,7 @@
 
 AC_CONFIG_SRCDIR([id3tag.h])
 
-AM_INIT_AUTOMAKE
+AM_INIT_AUTOMAKE([foreign])
 
 AM_CONFIG_HEADER([config.h])
 
