From 7fa541f6c0a31e3ddcab6ea85040b531bbaa4651 Mon Sep 17 00:00:00 2001
From: Martin Bark <martin@barkynet.com>
Date: Tue, 12 Jul 2016 19:03:28 +0100
Subject: [PATCH] inspector: don't build when ssl support is disabled

Signed-off-by: Martin Bark <martin@barkynet.com>
---
 node.gyp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/node.gyp b/node.gyp
index 8a8bd00..fdccd60 100644
--- a/node.gyp
+++ b/node.gyp
@@ -302,7 +302,7 @@
             'deps/v8/src/third_party/vtune/v8vtune.gyp:v8_vtune'
           ],
         }],
-        [ 'v8_inspector=="true"', {
+        [ 'v8_inspector=="true" and node_use_openssl =="true"', {
           'defines': [
             'HAVE_INSPECTOR=1',
             'V8_INSPECTOR_USE_STL=1',
@@ -828,7 +828,7 @@
       ],
 
       'conditions': [
-        ['v8_inspector=="true"', {
+        ['v8_inspector=="true" and node_use_openssl =="true"', {
           'sources': [
             'src/inspector_socket.cc',
             'test/cctest/test_inspector_socket.cc'
-- 
2.7.4

