diff -urN haserl-0.9.20-0rig/src/haserl.c haserl-0.9.20/src/haserl.c
--- haserl-0.9.20-0rig/src/haserl.c	2007-10-30 22:58:00.000000000 +0100
+++ haserl-0.9.20/src/haserl.c	2007-10-30 23:00:10.000000000 +0100
@@ -210,7 +210,7 @@
     }
   memcpy ((char *) (entry + strlen (prefix)), str, strlen (str));
 
-  keylen = (size_t) (index (entry, '=') - entry);
+  keylen = (size_t) (strchr (entry, '=') - entry);
 
   if (keylen <= 0)
     {
