From 0dcd2368c3e23a5cc1cc4a979d9c241ed9302236 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@openwide.fr>
Date: Sun, 4 Jan 2015 16:06:07 +0100
Subject: [PATCH] Makefile: install static library and headers separately

Signed-off-by: Romain Naour <romain.naour@openwide.fr>
---
 Makefile | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index e27dcbd..ecf52d1 100644
--- a/Makefile
+++ b/Makefile
@@ -45,12 +45,10 @@ perl-lib:	static
 	cd LockDev && make OPTIMIZE="-O2 -g -Wall"
 	cd LockDev && make test
 
-.PHONY: install install_dev install_dbg install_doc install_run 
-install:	install_dev install_dbg install_doc install_run
+.PHONY: install install_dev install_dbg install_doc install_run install_static
+install:	install_dev install_dbg install_doc install_run install_static
 
-install_dev:	${static} src/lockdev.h
-	install -m755 -d	${libdir}
-	install -m644 ${static}	${libdir}
+install_dev:	src/lockdev.h
 	install -m755 -d	${incdir}
 	install -m644 src/lockdev.h	${incdir}
 	install -m644 src/ttylock.h	${incdir}
@@ -69,6 +67,10 @@ install_doc:	docs/lockdev.3
 	install -m755 -d	${mandir}/man3
 	install -m644 docs/lockdev.3	${mandir}/man3
 
+install_static:	${static}
+	install -m755 -d	${libdir}
+	install -m644 ${static}	${libdir}
+
 install_run:	${shared}
 	install -m755 -d	${libdir}
 	install -m644 ${shared}	${libdir}
-- 
1.9.3

