--- a/src/Makefile.in.orig	2008-01-30 11:01:59.000000000 -0500
+++ b/src/Makefile.in	2008-01-30 11:03:59.000000000 -0500
@@ -76,8 +76,11 @@ LTCOMPILE = $(LIBTOOL) --tag=CC --mode=c
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
 	$(AM_CFLAGS) $(CFLAGS)
 CCLD = $(CC)
+HOST_CC = gcc
 LINK = $(LIBTOOL) --tag=CC --mode=link $(CCLD) $(AM_CFLAGS) $(CFLAGS) \
 	$(AM_LDFLAGS) $(LDFLAGS) -o $@
+mkg3states_LINK = $(HOST_CC) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) \
+	$(AM_LDFLAGS) $(LDFLAGS_FOR_BUILD) -o $@
 SOURCES = $(liblbxutil_la_SOURCES) $(nodist_liblbxutil_la_SOURCES) \
 	$(mkg3states_SOURCES)
 DIST_SOURCES = $(liblbxutil_la_SOURCES) $(mkg3states_SOURCES)
@@ -282,7 +285,7 @@ clean-noinstPROGRAMS:
 	done
 mkg3states$(EXEEXT): $(mkg3states_OBJECTS) $(mkg3states_DEPENDENCIES) 
 	@rm -f mkg3states$(EXEEXT)
-	$(LINK) $(mkg3states_LDFLAGS) $(mkg3states_OBJECTS) $(mkg3states_LDADD) $(LIBS)
+	$(mkg3states_LINK) $(mkg3states_LDFLAGS) $(mkg3states_OBJECTS) $(mkg3states_LDADD) $(LIBS)
 
 mostlyclean-compile:
 	-rm -f *.$(OBJEXT)
@@ -386,18 +389,18 @@ lbxdelta.lo: $(srcdir)/delta/lbxdelta.c
 @am__fastdepCC_FALSE@	$(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o lbxdelta.lo `test -f '$(srcdir)/delta/lbxdelta.c' || echo '$(srcdir)/'`$(srcdir)/delta/lbxdelta.c
 
 mkg3states.o: $(srcdir)/image/mkg3states.c
-@am__fastdepCC_TRUE@	if $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT mkg3states.o -MD -MP -MF "$(DEPDIR)/mkg3states.Tpo" -c -o mkg3states.o `test -f '$(srcdir)/image/mkg3states.c' || echo '$(srcdir)/'`$(srcdir)/image/mkg3states.c; \
+@am__fastdepCC_TRUE@	if $(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) -MT mkg3states.o -MD -MP -MF "$(DEPDIR)/mkg3states.Tpo" -c -o mkg3states.o `test -f '$(srcdir)/image/mkg3states.c' || echo '$(srcdir)/'`$(srcdir)/image/mkg3states.c; \
 @am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/mkg3states.Tpo" "$(DEPDIR)/mkg3states.Po"; else rm -f "$(DEPDIR)/mkg3states.Tpo"; exit 1; fi
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$(srcdir)/image/mkg3states.c' object='mkg3states.o' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o mkg3states.o `test -f '$(srcdir)/image/mkg3states.c' || echo '$(srcdir)/'`$(srcdir)/image/mkg3states.c
+@am__fastdepCC_FALSE@	$(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) -c -o mkg3states.o `test -f '$(srcdir)/image/mkg3states.c' || echo '$(srcdir)/'`$(srcdir)/image/mkg3states.c
 
 mkg3states.obj: $(srcdir)/image/mkg3states.c
-@am__fastdepCC_TRUE@	if $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -MT mkg3states.obj -MD -MP -MF "$(DEPDIR)/mkg3states.Tpo" -c -o mkg3states.obj `if test -f '$(srcdir)/image/mkg3states.c'; then $(CYGPATH_W) '$(srcdir)/image/mkg3states.c'; else $(CYGPATH_W) '$(srcdir)/$(srcdir)/image/mkg3states.c'; fi`; \
+@am__fastdepCC_TRUE@	if $(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) -MT mkg3states.obj -MD -MP -MF "$(DEPDIR)/mkg3states.Tpo" -c -o mkg3states.obj `if test -f '$(srcdir)/image/mkg3states.c'; then $(CYGPATH_W) '$(srcdir)/image/mkg3states.c'; else $(CYGPATH_W) '$(srcdir)/$(srcdir)/image/mkg3states.c'; fi`; \
 @am__fastdepCC_TRUE@	then mv -f "$(DEPDIR)/mkg3states.Tpo" "$(DEPDIR)/mkg3states.Po"; else rm -f "$(DEPDIR)/mkg3states.Tpo"; exit 1; fi
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	source='$(srcdir)/image/mkg3states.c' object='mkg3states.obj' libtool=no @AMDEPBACKSLASH@
 @AMDEP_TRUE@@am__fastdepCC_FALSE@	DEPDIR=$(DEPDIR) $(CCDEPMODE) $(depcomp) @AMDEPBACKSLASH@
-@am__fastdepCC_FALSE@	$(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS) -c -o mkg3states.obj `if test -f '$(srcdir)/image/mkg3states.c'; then $(CYGPATH_W) '$(srcdir)/image/mkg3states.c'; else $(CYGPATH_W) '$(srcdir)/$(srcdir)/image/mkg3states.c'; fi`
+@am__fastdepCC_FALSE@	$(HOST_CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) -c -o mkg3states.obj `if test -f '$(srcdir)/image/mkg3states.c'; then $(CYGPATH_W) '$(srcdir)/image/mkg3states.c'; else $(CYGPATH_W) '$(srcdir)/$(srcdir)/image/mkg3states.c'; fi`
 
 mostlyclean-libtool:
 	-rm -f *.lo
