--- gcc-3.4.6/gcc/collect2.c.orig	2009-05-11 13:00:31.000000000 -0600
+++ gcc-3.4.6/gcc/collect2.c	2009-05-11 13:00:26.000000000 -0600
@@ -1534,7 +1534,7 @@
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, O_RDWR);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
