--- a/configure.ac.orig	2007-06-04 12:56:16.000000000 +0200
+++ a/configure.ac	2007-06-04 12:55:58.000000000 +0200
@@ -50,7 +50,7 @@
 
 # Find the xcb-proto protocol descriptions
 AC_MSG_CHECKING(XCBPROTO_XCBINCLUDEDIR)
-XCBPROTO_XCBINCLUDEDIR=`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
+XCBPROTO_XCBINCLUDEDIR=$STAGING_DIR`$PKG_CONFIG --variable=xcbincludedir xcb-proto`
 AC_MSG_RESULT($XCBPROTO_XCBINCLUDEDIR)
 AC_SUBST(XCBPROTO_XCBINCLUDEDIR)
 
