From 0677e6cff00506de56d11bfa230b6c366e74f6ed Mon Sep 17 00:00:00 2001
From: Erik Larsson <erik.larsson@combitech.se>
Date: Thu, 8 Mar 2018 19:04:37 +0100
Subject: [PATCH] Add support for overriding BL32 and BL33 not only BL31

Signed-off-by: Erik Larsson <karl.erik.larsson@gmail.com>
Signed-off-by: Christopher Dahlberg <crille.dahlberg@gmail.com>
Signed-off-by: Marcus Folkesson <marcus.folkesson@gmail.com>
---
 iMX8M/mkimage_fit_atf.sh | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/iMX8M/mkimage_fit_atf.sh b/iMX8M/mkimage_fit_atf.sh
index 3a3bd10..727f746 100755
--- a/iMX8M/mkimage_fit_atf.sh
+++ b/iMX8M/mkimage_fit_atf.sh
@@ -11,29 +11,29 @@ if [ ! -f $BL31 ]; then
 	echo "ERROR: BL31 file $BL31 NOT found" >&2
 	exit 0
 else
-	echo "bl31.bin size: " >&2
-	ls -lct bl31.bin | awk '{print $5}' >&2
+	echo "$BL31 size: " >&2
+	ls -lct $BL31 | awk '{print $5}' >&2
 fi
 
-BL32="tee.bin"
+[ -z "$BL32" ] && BL32="tee.bin"
 
 if [ ! -f $BL32 ]; then
 	BL32=/dev/null
 else
-	echo "Building with TEE support, make sure your bl31 is compiled with spd. If you do not want tee, please delete tee.bin" >&2
-	echo "tee.bin size: " >&2
-	ls -lct tee.bin | awk '{print $5}' >&2
+	echo "Building with TEE support, make sure your bl31 is compiled with spd. If you do not want tee, please delete $BL32" >&2
+	echo "$BL32 size: " >&2
+	ls -lct $BL32 | awk '{print $5}' >&2
 fi
 
-BL33="u-boot-nodtb.bin"
+[ -z "$BL33" ] && BL33="u-boot-nodtb.bin"
 
 if [ ! -f $BL33 ]; then
 	echo "ERROR: $BL33 file NOT found" >&2
 	exit 0
 else
 	
-	echo "u-boot-nodtb.bin size: " >&2
-	ls -lct u-boot-nodtb.bin | awk '{print $5}' >&2
+	echo "$BL33: " >&2
+	ls -lct $BL33 | awk '{print $5}' >&2
 fi
 
 for dtname in $*
-- 
2.7.4

