From a73f20fbe921c7ecff3efda23bc506fdd935ee4d Mon Sep 17 00:00:00 2001
From: Erik Larsson <erik.larsson@combitech.se>
Date: Tue, 6 Mar 2018 12:28:39 +0100
Subject: [PATCH] Add LDFLAGS to link step

Signed-off-by: Erik Larsson <karl.erik.larsson@gmail.com>
---
 iMX8M/soc.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index 8346769..38a14bb 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -15,7 +15,7 @@ FW_DIR = imx-boot/imx-boot-tools/imx8mq
 
 $(MKIMG): mkimage_imx8.c
 	@echo "Compiling mkimage_imx8"
-	$(CC) $(CFLAGS) mkimage_imx8.c -o $(MKIMG) -lz
+	$(CC) $(CFLAGS) mkimage_imx8.c -o $(MKIMG) $(LDFLAGS) -lz
 
 $(DCD_CFG): $(DCD_CFG_SRC)
 	@echo "Converting iMX8M DCD file" 
-- 
2.7.4

