From c2e96bdc453334f10faf8db0eb06e41ab3ca4fe9 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sat, 23 Jun 2012 21:58:07 +0200
Subject: [PATCH] uClibc without RPC support does not install rpcent.h

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 tirpc/rpc/rpcent.h |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/tirpc/rpc/rpcent.h b/tirpc/rpc/rpcent.h
index 7fd93d9..8ce05ed 100644
--- a/tirpc/rpc/rpcent.h
+++ b/tirpc/rpc/rpcent.h
@@ -46,8 +46,9 @@
 
 __BEGIN_DECLS
 
-/* These are defined in /usr/include/rpc/netdb.h */
-#if 0
+/* These are defined in /usr/include/rpc/netdb.h, unless we are using
+   the C library without RPC support. */
+#if defined(__UCLIBC__) && !defined(__UCLIBC_HAS_RPC__)
 struct rpcent {
 	char	*r_name;	/* name of server for this rpc program */
 	char	**r_aliases;	/* alias list */
-- 
1.7.9.5

