From fbe5fccf0e3bdaae3defc66a288176797fd12a17 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sun, 24 Jun 2012 21:40:21 +0200
Subject: [PATCH] Make IPv6 support optional

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index b5db263..f086255 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -7,7 +7,7 @@
 
 noinst_HEADERS = rpc_com.h
 
-AM_CPPFLAGS = -I$(top_srcdir)/tirpc -DPORTMAP -DINET6 \
+AM_CPPFLAGS = -I$(top_srcdir)/tirpc -DPORTMAP \
 		-D_GNU_SOURCE -Wall -pipe  
 
 lib_LTLIBRARIES = libtirpc.la
-- 
2.0.0

