Fix deprecation from bison 3.0+
From http://comments.gmane.org/gmane.linux.lfs.beyond.devel/24620

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura gstreamer-0.10.36.orig/gst/parse/grammar.y gstreamer-0.10.36/gst/parse/grammar.y
--- gstreamer-0.10.36.orig/gst/parse/grammar.y	2013-10-04 09:52:21.685282055 -0300
+++ gstreamer-0.10.36/gst/parse/grammar.y	2013-10-04 10:21:54.800211238 -0300
@@ -26,7 +26,6 @@
  */

 #define YYERROR_VERBOSE 1
-#define YYLEX_PARAM scanner

 #define YYENABLE_NLS 0

@@ -648,6 +647,7 @@
 %right '.'
 %left '!' '='

+%lex-param { void *scanner }
 %parse-param { void *scanner }
 %parse-param { graph_t *graph }
 %pure-parser
