Link to libncurses instead of the legacy libcurses.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura statserial-1.1.orig/Makefile statserial-1.1/Makefile
--- statserial-1.1.orig/Makefile	2014-07-04 12:47:41.279440415 -0300
+++ statserial-1.1/Makefile	2014-07-04 12:47:47.948665740 -0300
@@ -10,7 +10,7 @@
 LDFLAGS = -s -N
 
 statserial:	statserial.o
-	$(LD) $(LDFLAGS) -o statserial statserial.o -lcurses
+	$(LD) $(LDFLAGS) -o statserial statserial.o -lncurses
 
 statserial.o: statserial.c
 	$(CC) $(CFLAGS) -c statserial.c
