From 2b4a43a94fb988e344f2d7edc74d6a4a5f64571d Mon Sep 17 00:00:00 2001
From: Sagaert Johan <sagaert.johan@proximus.be>
Date: Wed, 15 Apr 2015 00:40:46 +0200
Subject: [PATCH] don't threat warnings as errors

Signed-off-by: Sagaert Johan <sagaert.johan@proximus.be>
---
 Makefile.am.inc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am.inc b/Makefile.am.inc
index fec591b..5eb461e 100644
--- a/Makefile.am.inc
+++ b/Makefile.am.inc
@@ -1,2 +1,2 @@
-AM_CFLAGS = -Wall -Werror -Wno-error=deprecated-declarations -Wextra -Wwrite-strings -Wno-unused-parameter -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
+AM_CFLAGS = -Wall -Wno-error=deprecated-declarations -Wextra -Wwrite-strings -std=gnu99 -D_GNU_SOURCE -D_REENTRANT
 
-- 
2.1.0

