Link libfcgi against the math library.

Signed-off-by: Thomas Claveirole <thomas.claveirole@green-communications.fr>

Index: b/libfcgi/Makefile.am
===================================================================
--- a/libfcgi/Makefile.am
+++ b/libfcgi/Makefile.am
@@ -18,6 +18,7 @@
                      os_@SYSTEM@.c
 libfcgi_la_CC      = @PTHREAD_CC@
 libfcgi_la_CFLAGS  = @PTHREAD_CFLAGS@
+libfcgi_la_LIBADD  = -lm
 
 libfcgi___la_SOURCES = $(INCLUDE_FILES)       \
                        $(INCLUDEDIR)/fcgio.h  \
