Support for libpng 1.5+ shamelessly taken from Gentoo.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

--- a/libmb/mbpixbuf.c
+++ b/libmb/mbpixbuf.c
@@ -247,7 +247,7 @@
     return NULL;
   }
 
-  if ( setjmp( png_ptr->jmpbuf ) ) {
+  if ( setjmp( png_jmpbuf(png_ptr) ) ) {
     png_destroy_read_struct( &png_ptr, &info_ptr, NULL);
     fclose(fd);
     return NULL;
@@ -269,8 +269,8 @@
       ( color_type == PNG_COLOR_TYPE_GRAY_ALPHA ))
     png_set_gray_to_rgb(png_ptr);
  
-  if ( info_ptr->color_type == PNG_COLOR_TYPE_RGB_ALPHA 
-       || info_ptr->color_type == PNG_COLOR_TYPE_GRAY_ALPHA
+  if ( color_type == PNG_COLOR_TYPE_RGB_ALPHA 
+       || color_type == PNG_COLOR_TYPE_GRAY_ALPHA
        )
     *has_alpha = 1;
   else
