diff -urpN fbv-1.0b.orig/configure fbv-1.0b/configure
--- fbv-1.0b.orig/configure	2004-09-07 13:29:27.000000000 +0200
+++ fbv-1.0b/configure	2008-04-24 10:52:37.000000000 +0200
@@ -80,6 +80,7 @@ while true ; do
 	esac
 done
 
+[ -z "$CC" ] && CC=cc
 [ -z "$prefix" ] && prefix="/usr/local"
 [ -z "$bindir" ] && bindir="${prefix}/bin"
 [ -z "$mandir" ] && mandir="${prefix}/man"
@@ -106,12 +107,12 @@ xdir="/usr/X11R6"
 ungif="no"
 echo "libungif check" >>./config.log
 echo "  1st:" >>./config.log
-cc 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lungif $libs
+$CC 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lungif $libs
 if [ -e \$\$~test ]; then
 	libs="-lungif $libs" ; ungif="yes"
 else
 	echo "  2nd: -lX11 -L$xdir/lib" >>./config.log
-	cc 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lungif -lX11 -L$xdir/lib $libs
+	$CC 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lungif -lX11 -L$xdir/lib $libs
 	if [ -e \$\$~test ]; then
 	libs="-lungif -lX11 -L$xdir/lib $libs" ; ungif="yes"
 	fi
@@ -124,7 +125,7 @@ echo "libungif: $ungif" >> ./config.log
 echo -n "checking for libjpeg presence... "
 if [ "$jpeg" != "disabled" ]; then
 jpeg="no"
-cc 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -ljpeg $libs
+$CC 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -ljpeg $libs
 if [ -e \$\$~test ]; then
 	libs="-ljpeg $libs" ; jpeg="yes"
 fi
@@ -135,7 +136,7 @@ echo "libjpeg: $jpeg" >> ./config.log
 echo -n "checking for libpng presence... "
 if [ "$png" != "disabled" ]; then
 png="no"
-cc 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lpng $libs
+$CC 2>>./config.log >>./config.log -o \$\$~test \$\$~test.c -lpng $libs
 if [ -e \$\$~test ]; then
 	libs="-lpng $libs" ; png="yes"
 fi
