Support for libpng 1.5+ shamelessly taken from Gentoo.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

--- a/png.c
+++ b/png.c
@@ -69,7 +69,7 @@
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
@@ -161,7 +161,7 @@
         fclose(fh); return(FH_ERROR_FORMAT);
     }
     rp=0;
-    if (setjmp(png_ptr->jmpbuf))
+    if (setjmp(png_jmpbuf(png_ptr)))
     {
 	png_destroy_read_struct(&png_ptr, &info_ptr, (png_infopp)NULL);
         if(rp) free(rp);
