From 845478644ba54a6947e9b224f2e5cd342e8257a9 Mon Sep 17 00:00:00 2001
From: Andrew Hsieh <andrewhsieh@google.com>
Date: Wed, 25 Jun 2014 22:13:48 -0700
Subject: [PATCH] Fix a typo in vmlaq_lane_s32

BUG=15526898

Change-Id: I4e35a764d369d378808dab29beefe34d1f93249b
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---

diff --git a/gcc/config/aarch64/arm_neon.h b/gcc/config/aarch64/arm_neon.h
index 73c7e7d..92b0119 100644
--- a/gcc/config/aarch64/arm_neon.h
+++ b/gcc/config/aarch64/arm_neon.h
@@ -9984,7 +9984,7 @@
 #define vmlaq_lane_s32(a, b, c, d)                                      \
   __extension__                                                         \
     ({                                                                  \
-       int32x4_t c_ = (c);                                              \
+       int32x2_t c_ = (c);                                              \
        int32x4_t b_ = (b);                                              \
        int32x4_t a_ = (a);                                              \
        int32x4_t result;                                                \
