--- busybox/Config.h	6 Apr 2002 04:22:15 -0000	1.116
+++ busybox/Config.h	30 May 2002 05:10:32 -0000
@@ -7,8 +7,8 @@
 //
 //
 // BusyBox Applications
-//#define BB_ADJTIMEX
-//#define BB_AR
+#define BB_ADJTIMEX
+#define BB_AR
 #define BB_ASH
 #define BB_BASENAME
 #define BB_CAT
@@ -37,12 +37,12 @@
 //#define BB_DUMPKMAP
 #define BB_ECHO
 #define BB_ENV
-//#define BB_EXPR
+#define BB_EXPR
 //#define BB_FBSET
 //#define BB_FDFLUSH
 #define BB_FIND
 #define BB_FREE
-//#define BB_FREERAMDISK
+#define BB_FREERAMDISK
 //#define BB_FSCK_MINIX
 //#define BB_GETOPT
 #define BB_GREP
@@ -51,12 +51,12 @@
 #define BB_HALT
 #define BB_HEAD
 //#define BB_HOSTID
-//#define BB_HOSTNAME
+#define BB_HOSTNAME
 //#define BB_HUSH
 #define BB_ID
-//#define BB_IFCONFIG
+#define BB_IFCONFIG
 #define BB_INIT
-//#define BB_INSMOD
+#define BB_INSMOD
 #define BB_KILL
 #define BB_KILLALL
 #define BB_KLOGD
@@ -68,11 +68,11 @@
 //#define BB_LOADKMAP
 #define BB_LOGGER
 //#define BB_LOGNAME
-//#define BB_LOSETUP
+#define BB_LOSETUP
 #define BB_LS
 #define BB_LSMOD
 //#define BB_MAKEDEVS
-//#define BB_MD5SUM
+#define BB_MD5SUM
 #define BB_MKDIR
 //#define BB_MKFIFO
 //#define BB_MKFS_MINIX
@@ -85,11 +85,11 @@
 //#define BB_MSH
 //#define BB_MT
 #define BB_MV
-//#define BB_NC
-//#define BB_NSLOOKUP
+#define BB_NC
+#define BB_NSLOOKUP
 #define BB_PIDOF
-//#define BB_PING
-//#define BB_PIVOT_ROOT
+#define BB_PING
+#define BB_PIVOT_ROOT
 #define BB_POWEROFF
 //#define BB_PRINTF
 #define BB_PS
@@ -101,8 +101,8 @@
 #define BB_RESET
 #define BB_RM
 #define BB_RMDIR
-//#define BB_RMMOD
-//#define BB_ROUTE
+#define BB_RMMOD
+#define BB_ROUTE
 //#define BB_RPM2CPIO
 #define BB_SED
 //#define BB_SETKEYCODES
@@ -115,13 +115,13 @@
 #define BB_TAIL
 #define BB_TAR
 //#define BB_TEE
-//#define BB_TEST
+#define BB_TEST
 //#define BB_TELNET
 //#define BB_TFTP
-//#define BB_TIME
+#define BB_TIME
 #define BB_TOUCH
 //#define BB_TR
-//#define BB_TRACEROUTE
+#define BB_TRACEROUTE
 #define BB_TRUE_FALSE
 #define BB_TTY
 //#define BB_UNIX2DOS
@@ -133,10 +133,10 @@
 //#define BB_UPDATE
 #define BB_UPTIME
 //#define BB_USLEEP
-//#define BB_VI
+#define BB_VI
 //#define BB_WATCHDOG
 #define BB_WC
-//#define BB_WGET
+#define BB_WGET
 #define BB_WHICH
 #define BB_WHOAMI
 #define BB_XARGS
@@ -217,7 +217,7 @@
 #define BB_FEATURE_USE_INITTAB
 //
 //Enable init being called as /linuxrc
-#define BB_FEATURE_LINUXRC
+//#define BB_FEATURE_LINUXRC
 //
 //Have init enable core dumping for child processes (for debugging only) 
 //#define BB_FEATURE_INIT_COREDUMPS
@@ -226,7 +226,7 @@
 //#define BB_FEATURE_EXTRA_QUIET
 //
 // enable syslogd -R remotehost
-#define BB_FEATURE_REMOTE_LOG
+//#define BB_FEATURE_REMOTE_LOG
 //
 // enable syslogd -C
 //#define BB_FEATURE_IPC_SYSLOG
@@ -277,7 +277,7 @@
 #define BB_FEATURE_COMMAND_TAB_COMPLETION
 //
 // Attempts to match usernames in a ~-prefixed path
-//#define BB_FEATURE_COMMAND_USERNAME_COMPLETION
+#define BB_FEATURE_COMMAND_USERNAME_COMPLETION
 //
 //Allow the shell to invoke all the compiled in BusyBox applets as if they
 //were shell builtins.  Nice for staticly linking an emergency rescue shell,
@@ -298,7 +298,7 @@
 // current username and hostname.  On systems that don't have usernames
 // or hostnames, this can look hideous.
 // Only relevant if a shell is enabled.
-//#define BB_FEATURE_SH_FANCY_PROMPT
+#define BB_FEATURE_SH_FANCY_PROMPT
 //
 // Uncomment this option to disable job control.  Job control lets you 
 // run jobs in the background (which completely useless for is all you 
@@ -314,7 +314,7 @@
 //#define BB_FEATURE_FBSET_READMODE
 //
 // Support insmod/lsmod/rmmod for post 2.1 kernels
-//#define BB_FEATURE_NEW_MODULE_INTERFACE
+#define BB_FEATURE_NEW_MODULE_INTERFACE
 //
 // Support insmod/lsmod/rmmod for pre 2.1 kernels
 //#define BB_FEATURE_OLD_MODULE_INTERFACE
@@ -332,16 +332,16 @@
 //#define BB_FEATURE_MINIX2
 //
 // Enable ifconfig status reporting output -- this feature adds 7k.
-//#define BB_FEATURE_IFCONFIG_STATUS
+#define BB_FEATURE_IFCONFIG_STATUS
 //
 // Enable ifconfig slip-specific options "keepalive" and "outfill"
-//#define BB_FEATURE_IFCONFIG_SLIP
+#define BB_FEATURE_IFCONFIG_SLIP
 //
 // Enable ifconfig options "mem_start", "io_addr", and "irq".
-//#define BB_FEATURE_IFCONFIG_MEMSTART_IOADDR_IRQ
+#define BB_FEATURE_IFCONFIG_MEMSTART_IOADDR_IRQ
 //
 // Enable ifconfig option "hw".  Currently works for only with "ether".
-//#define BB_FEATURE_IFCONFIG_HW
+#define BB_FEATURE_IFCONFIG_HW
 //
 // Allows "broadcast +" to set broadcast automatically based on hostaddr
 // and netmask, at a cost of about 100 bytes of code (i386).
@@ -379,7 +379,7 @@
 #define BB_FEATURE_FIND_NEWER
 //
 // Support for the -A -B and -C context flags in grep
-//#define BB_FEATURE_GREP_CONTEXT
+#define BB_FEATURE_GREP_CONTEXT
 //
 // Support for the EGREP applet (alias to the grep applet)
 //#define BB_FEATURE_GREP_EGREP_ALIAS
