--- busybox/Config.h	17 Sep 2002 22:04:28 -0000	1.118
+++ busybox/Config.h	17 Sep 2002 23:43:54 -0000
@@ -8,7 +8,7 @@
 //
 // BusyBox Applications
 //#define BB_ADJTIMEX
-//#define BB_AR
+#define BB_AR
 #define BB_ASH
 #define BB_BASENAME
 #define BB_CAT
@@ -18,14 +18,14 @@
 #define BB_CHROOT
 #define BB_CHVT
 #define BB_CLEAR
-//#define BB_CMP
+#define BB_CMP
 #define BB_CP
-//#define BB_CPIO
+#define BB_CPIO
 #define BB_CUT
 #define BB_DATE
-//#define BB_DC
+#define BB_DC
 #define BB_DD
-//#define BB_DEALLOCVT
+#define BB_DEALLOCVT
 #define BB_DF
 #define BB_DIRNAME
 #define BB_DMESG
@@ -37,12 +37,12 @@
 //#define BB_DUMPKMAP
 #define BB_ECHO
 #define BB_ENV
-//#define BB_EXPR
-//#define BB_FBSET
+#define BB_EXPR
+#define BB_FBSET
 //#define BB_FDFLUSH
 #define BB_FIND
 #define BB_FREE
-//#define BB_FREERAMDISK
+#define BB_FREERAMDISK
 //#define BB_FSCK_MINIX
 //#define BB_GETOPT
 #define BB_GREP
@@ -50,13 +50,13 @@
 #define BB_GZIP
 #define BB_HALT
 #define BB_HEAD
-//#define BB_HOSTID
-//#define BB_HOSTNAME
+#define BB_HOSTID
+#define BB_HOSTNAME
 //#define BB_HUSH
 #define BB_ID
-//#define BB_IFCONFIG
+#define BB_IFCONFIG
 #define BB_INIT
-//#define BB_INSMOD
+#define BB_INSMOD
 #define BB_KILL
 #define BB_KILLALL
 #define BB_KLOGD
@@ -67,77 +67,77 @@
 //#define BB_LOADFONT
 //#define BB_LOADKMAP
 #define BB_LOGGER
-//#define BB_LOGNAME
-//#define BB_LOSETUP
+#define BB_LOGNAME
+#define BB_LOSETUP
 #define BB_LS
 #define BB_LSMOD
 //#define BB_MAKEDEVS
-//#define BB_MD5SUM
+#define BB_MD5SUM
 #define BB_MKDIR
 //#define BB_MKFIFO
 //#define BB_MKFS_MINIX
 #define BB_MKNOD
 #define BB_MKSWAP
-//#define BB_MKTEMP
+#define BB_MKTEMP
 #define BB_MODPROBE
 #define BB_MORE
 #define BB_MOUNT
 //#define BB_MSH
 //#define BB_MT
 #define BB_MV
-//#define BB_NC
-//#define BB_NSLOOKUP
+#define BB_NC
+#define BB_NSLOOKUP
 #define BB_PIDOF
-//#define BB_PING
-//#define BB_PIVOT_ROOT
+#define BB_PING
+#define BB_PIVOT_ROOT
 #define BB_POWEROFF
-//#define BB_PRINTF
+#define BB_PRINTF
 #define BB_PS
 #define BB_PWD
-//#define BB_RDATE
-//#define BB_READLINK
+#define BB_RDATE
+#define BB_READLINK
 #define BB_REBOOT
 //#define BB_RENICE
 #define BB_RESET
 #define BB_RM
 #define BB_RMDIR
-//#define BB_RMMOD
-//#define BB_ROUTE
+#define BB_RMMOD
+#define BB_ROUTE
 //#define BB_RPM2CPIO
 #define BB_SED
 //#define BB_SETKEYCODES
 #define BB_SLEEP
 #define BB_SORT
-//#define BB_STTY
+#define BB_STTY
 #define BB_SWAPONOFF
 #define BB_SYNC
 #define BB_SYSLOGD
 #define BB_TAIL
 #define BB_TAR
-//#define BB_TEE
-//#define BB_TEST
-//#define BB_TELNET
+#define BB_TEE
+#define BB_TEST
+#define BB_TELNET
 //#define BB_TFTP
-//#define BB_TIME
-//#define BB_TOP
+#define BB_TIME
+#define BB_TOP
 #define BB_TOUCH
-//#define BB_TR
-//#define BB_TRACEROUTE
+#define BB_TR
+#define BB_TRACEROUTE
 #define BB_TRUE_FALSE
 #define BB_TTY
 //#define BB_UNIX2DOS
-//#define BB_UUENCODE
-//#define BB_UUDECODE
+#define BB_UUENCODE
+#define BB_UUDECODE
 #define BB_UMOUNT
 #define BB_UNIQ
 #define BB_UNAME
 //#define BB_UPDATE
 #define BB_UPTIME
-//#define BB_USLEEP
-//#define BB_VI
+#define BB_USLEEP
+#define BB_VI
 //#define BB_WATCHDOG
 #define BB_WC
-//#define BB_WGET
+#define BB_WGET
 #define BB_WHICH
 #define BB_WHOAMI
 #define BB_XARGS
@@ -185,7 +185,7 @@
 #define BB_FEATURE_VERBOSE_USAGE
 //
 // Use termios to manipulate the screen ('more' is prettier with this on)
-//#define BB_FEATURE_USE_TERMIOS
+#define BB_FEATURE_USE_TERMIOS
 //
 // calculate terminal & column widths (for more, ls, and telnet)
 #define BB_FEATURE_AUTOWIDTH
@@ -224,10 +224,10 @@
 //#define BB_FEATURE_INIT_COREDUMPS
 //
 //Make sure nothing is printed to the console on boot
-//#define BB_FEATURE_EXTRA_QUIET
+#define BB_FEATURE_EXTRA_QUIET
 //
 // enable syslogd -R remotehost
-#define BB_FEATURE_REMOTE_LOG
+//#define BB_FEATURE_REMOTE_LOG
 //
 // enable syslogd -C
 //#define BB_FEATURE_IPC_SYSLOG
@@ -248,7 +248,7 @@
 //
 // If you are using uClibc, be sure that you've already compiled
 // uClibc with INCLUDE_RPC=true (contained in the Config file)
-//#define BB_FEATURE_NFSMOUNT
+#define BB_FEATURE_NFSMOUNT
 //
 // Enable support forced filesystem unmounting 
 // (i.e., in case of an unreachable NFS system).
@@ -278,7 +278,7 @@
 #define BB_FEATURE_COMMAND_TAB_COMPLETION
 //
 // Attempts to match usernames in a ~-prefixed path
-//#define BB_FEATURE_COMMAND_USERNAME_COMPLETION
+#define BB_FEATURE_COMMAND_USERNAME_COMPLETION
 //
 //Allow the shell to invoke all the compiled in BusyBox applets as if they
 //were shell builtins.  Nice for staticly linking an emergency rescue shell,
@@ -299,7 +299,7 @@
 // current username and hostname.  On systems that don't have usernames
 // or hostnames, this can look hideous.
 // Only relevant if a shell is enabled.
-//#define BB_FEATURE_SH_FANCY_PROMPT
+#define BB_FEATURE_SH_FANCY_PROMPT
 //
 // Uncomment this option to disable job control.  Job control lets you 
 // run jobs in the background (which completely useless for is all you 
@@ -333,16 +333,16 @@
 //#define BB_FEATURE_MINIX2
 //
 // Enable ifconfig status reporting output -- this feature adds 7k.
-//#define BB_FEATURE_IFCONFIG_STATUS
+#define BB_FEATURE_IFCONFIG_STATUS
 //
 // Enable ifconfig slip-specific options "keepalive" and "outfill"
 //#define BB_FEATURE_IFCONFIG_SLIP
 //
 // Enable ifconfig options "mem_start", "io_addr", and "irq".
-//#define BB_FEATURE_IFCONFIG_MEMSTART_IOADDR_IRQ
+#define BB_FEATURE_IFCONFIG_MEMSTART_IOADDR_IRQ
 //
 // Enable ifconfig option "hw".  Currently works for only with "ether".
-//#define BB_FEATURE_IFCONFIG_HW
+#define BB_FEATURE_IFCONFIG_HW
 //
 // Allows "broadcast +" to set broadcast automatically based on hostaddr
 // and netmask, at a cost of about 100 bytes of code (i386).
@@ -380,7 +380,7 @@
 #define BB_FEATURE_FIND_NEWER
 //
 // Support for the -A -B and -C context flags in grep
-//#define BB_FEATURE_GREP_CONTEXT
+#define BB_FEATURE_GREP_CONTEXT
 //
 // Support for the EGREP applet (alias to the grep applet)
 //#define BB_FEATURE_GREP_EGREP_ALIAS
