--- gcc-3.3.2/libstdc++-v3/testsuite/Makefile.in.dist	2003-10-26 00:07:43.000000000 -0500
+++ gcc-3.3.2/libstdc++-v3/testsuite/Makefile.in	2003-10-26 00:06:35.000000000 -0500
@@ -204,9 +204,9 @@
 PROGRAMS =  $(noinst_PROGRAMS)
 
 abi_check_OBJECTS =  abi_check.$(OBJEXT)
-abi_check_LDADD = $(LDADD)
+abi_check_LDADD = -lstdc++ $(LDADD)
 abi_check_DEPENDENCIES = 
-abi_check_LDFLAGS = 
+abi_check_LDFLAGS = -L../src
 CXXFLAGS = @CXXFLAGS@
 CXXCOMPILE = $(CXX) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
 LTCXXCOMPILE = $(LIBTOOL) --mode=compile $(CXX) $(DEFS) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) $(AM_CXXFLAGS) $(CXXFLAGS)
