--- gcc-3.3/libstdc++-v3/configure	2003-05-01 20:37:25.000000000 -0600
+++ gcc-3.3-patched/libstdc++-v3/configure	2003-06-21 00:22:02.000000000 -0600
@@ -5163,10 +5163,6 @@
 EOF
 
       cat >> confdefs.h <<\EOF
-#define HAVE_FREXPF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
 #define HAVE_HYPOTF 1
 EOF
 
@@ -10293,103 +10289,6 @@
 #define HAVE_MMAP 1
 EOF
 
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_ACOSF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_ASINF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_ATAN2F 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_ATANF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_CEILF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_COPYSIGN 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_COPYSIGNF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_COSF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_COSHF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_EXPF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_FABSF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_FLOORF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_FMODF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_FREXPF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_LDEXPF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_LOG10F 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_LOGF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_MODFF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_POWF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_SINF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_SINHF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_SQRTF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_TANF 1
-EOF
-
-      cat >> confdefs.h <<\EOF
-#define HAVE_TANHF 1
-EOF
-
       ;;
   esac
 
