diff -urN ltp-full-20040506.orig/testcases/kernel/syscalls/fmtmsg/Makefile ltp-full-20040506/testcases/kernel/syscalls/fmtmsg/Makefile
--- ltp-full-20040506.orig/testcases/kernel/syscalls/fmtmsg/Makefile	2003-03-04 09:15:27.000000000 -0700
+++ ltp-full-20040506/testcases/kernel/syscalls/fmtmsg/Makefile	2004-05-12 14:03:56.000000000 -0600
@@ -24,7 +24,7 @@
 LOADLIBES+=	-L../../../../lib -lltp
 
 SRCS=$(wildcard *.c)
-TARGETS=$(patsubst %.c,%,$(SRCS))
+TARGETS=#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -urN ltp-full-20040506.orig/testcases/kernel/syscalls/profil/Makefile ltp-full-20040506/testcases/kernel/syscalls/profil/Makefile
--- ltp-full-20040506.orig/testcases/kernel/syscalls/profil/Makefile	2003-03-04 09:16:26.000000000 -0700
+++ ltp-full-20040506/testcases/kernel/syscalls/profil/Makefile	2004-05-12 14:03:56.000000000 -0600
@@ -23,7 +23,7 @@
 LOADLIBES+=	-L../../../../lib -lltp 
 
 SRCS=$(wildcard *.c)
-TARGETS=$(patsubst %.c,%,$(SRCS))
+TARGETS=#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -urN ltp-full-20040506.orig/testcases/kernel/syscalls/swapon/swapon02.c ltp-full-20040506/testcases/kernel/syscalls/swapon/swapon02.c
--- ltp-full-20040506.orig/testcases/kernel/syscalls/swapon/swapon02.c	2004-04-23 12:13:20.000000000 -0600
+++ ltp-full-20040506/testcases/kernel/syscalls/swapon/swapon02.c	2004-05-12 23:02:50.000000000 -0600
@@ -85,9 +85,7 @@
 #include <sys/swap.h>
 #include <asm/page.h>
 #include <asm/atomic.h>
-#ifdef OLDER_DISTRO_RELEASE
-#include <linux/swap.h>
-#endif
+#define MAX_SWAPFILES 32
 #include <fcntl.h>
 #include <pwd.h>
 #include <string.h>
@@ -96,10 +94,6 @@
 #include "test.h"
 #include "usctest.h"
 
-#ifndef OLDER_DISTRO_RELEASE
-#define MAX_SWAPFILES 32 
-#endif
-
 static void setup();
 static void cleanup();
 static int setup01();
