--- ltp-full-20031002/testcases/kernel/syscalls/fmtmsg/Makefile.orig	2003-10-02 19:18:10.000000000 -0600
+++ ltp-full-20031002/testcases/kernel/syscalls/fmtmsg/Makefile	2003-10-02 19:18:32.000000000 -0600
@@ -24,7 +24,7 @@
 LOADLIBES+=	-L../../../../lib -lltp
 
 SRCS=$(wildcard *.c)
-TARGETS=$(patsubst %.c,%,$(SRCS))
+TARGETS=#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
--- ltp-full-20031002/testcases/kernel/syscalls/profil/Makefile.orig	2003-10-02 19:27:53.000000000 -0600
+++ ltp-full-20031002/testcases/kernel/syscalls/profil/Makefile	2003-10-02 19:28:02.000000000 -0600
@@ -23,7 +23,7 @@
 LOADLIBES+=	-L../../../../lib -lltp 
 
 SRCS=$(wildcard *.c)
-TARGETS=$(patsubst %.c,%,$(SRCS))
+TARGETS=#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
--- ltp-full-20031002/testcases/network/rpc/rpc01/Makefile.orig	2003-10-02 19:36:51.000000000 -0600
+++ ltp-full-20031002/testcases/network/rpc/rpc01/Makefile	2003-10-02 19:37:03.000000000 -0600
@@ -2,7 +2,7 @@
 LDLIBS += 
 
 SRCS=$(wildcard *.c)
-TARGETS=$(patsubst %.c,%,$(SRCS))
+TARGETS=#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
--- ltp-full-20031002/tools/netpipe-2.4-ipv6/Makefile.orig	2003-11-07 16:41:39.000000000 -0700
+++ ltp-full-20031002/tools/netpipe-2.4-ipv6/Makefile	2003-11-07 16:42:41.000000000 -0700
@@ -10,7 +10,7 @@
 DRIV_OBJ   = netpipe.o
 INCLUDES   = netpipe.h
 # Default target is just TCP
-TARGETS    = NPtcp-ipv6
+TARGETS    = #NPtcp-ipv6
 # If you have TCP, MPI and PVM
 #TARGETS    = NPtcp NPmpi NPpvm
 CFLAGS		    += -O -Wall
@@ -28,12 +28,12 @@
 targets:    $(TARGETS)
 
 install:
-	@ln -f NPtcp-ipv6 ../../testcases/bin	
+	#@ln -f NPtcp-ipv6 ../../testcases/bin	
 #
 # This section of the Makefile is for compiling the binaries
 #
 
-TCP:	NPtcp-ipv6
+TCP:	#NPtcp-ipv6
 	@echo 'NPtcp has been built.'
 
 NPtcp-ipv6:	NPtcp.o TCP.o
--- ltp-full-20031106/testcases/kernel/syscalls/string/string01.orig.c	2004-01-01 23:54:37.000000000 -0700
+++ ltp-full-20031106/testcases/kernel/syscalls/string/string01.c	2004-01-01 23:55:48.000000000 -0700
@@ -413,10 +413,10 @@
 	//fprintf(temp, "\tStrncmp\n" );
 	i = 0;
 	while ( t_ncmp[i].s1 ) {
-	    if ((n = strncmp( t_ncmp[i].s1, t_ncmp[i].s2, t_ncmp[i].n ))
-							!= t_ncmp[i].e_res) {
+	    n = strncmp( t_ncmp[i].s1, t_ncmp[i].s2, t_ncmp[i].n );
+	    if (sign(n) != sign(t_ncmp[i].e_res)) {
 		  fprintf(temp, "(Strncmp) test %d: expected %d, got %d",
-			i, t_ncmp[i].e_res, n );
+			i, sign(t_ncmp[i].e_res), sign(n) );
 		  local_flag = FAILED;
 		}
 	    i++;
--- ltp-full-20040108/testcases/kernel/syscalls/pipe/pipe07.c.orig	2003-12-15 10:08:10.000000000 -0700
+++ ltp-full-20040108/testcases/kernel/syscalls/pipe/pipe07.c	2004-01-11 00:55:55.000000000 -0700
@@ -84,8 +84,9 @@
 	setup();
         /* Get the currently used number of file descriptors */
 	mypid=getpid();
-	cmdstring=malloc(sizeof(cmdstring));
-	sprintf(cmdstring,"ls -A -1 /proc/%d/fd | wc -l | awk {'print $1'}> pipe07.tmp",mypid);
+	cmdstring=malloc(BUFSIZ);
+	snprintf(cmdstring, BUFSIZ, "ls -A -1 /proc/%d/fd | "
+		"wc -l | awk {'print $1'}> pipe07.tmp", mypid);
 	if (system(cmdstring) == 0)
 	{
 		f = fopen("pipe07.tmp", "r");	
