--- net-snmp-5.0.7/configure.in.orig	2003-06-09 14:54:18.000000000 -0600
+++ net-snmp-5.0.7/configure.in	2003-06-09 14:54:42.000000000 -0600
@@ -1710,13 +1710,8 @@
 if test $cross_compiling = yes; then
   if test $with_endianness = "big"; then
     AC_DEFINE(WORDS_BIGENDIAN)
-  elif test -z $with_endianness; then
-    AC_MSG_ERROR([You are cross-compiling, but you have not specified the target's endianness])
   fi
 else
-  if test $with_endianness; then
-    AC_MSG_ERROR([Endianness has been specified, but you are not cross-compiling.])
-  fi
   AC_C_BIGENDIAN
 fi
 
