diff -uNr openssh-3.5p1.orig/Makefile.in openssh-3.5p1/Makefile.in
--- openssh-3.5p1.orig/Makefile.in	Sun Jul 14 19:02:21 2002
+++ openssh-3.5p1/Makefile.in	Mon Mar 17 14:54:20 2003
@@ -217,19 +217,19 @@
 	$(srcdir)/mkinstalldirs $(DESTDIR)$(mandir)/$(mansubdir)8
 	$(srcdir)/mkinstalldirs $(DESTDIR)$(libexecdir)
 	(umask 022 ; $(srcdir)/mkinstalldirs $(DESTDIR)$(PRIVSEP_PATH))
-	$(INSTALL) -m 0755 -s ssh $(DESTDIR)$(bindir)/ssh
-	$(INSTALL) -m 0755 -s scp $(DESTDIR)$(bindir)/scp
-	$(INSTALL) -m 0755 -s ssh-add $(DESTDIR)$(bindir)/ssh-add
-	$(INSTALL) -m 0755 -s ssh-agent $(DESTDIR)$(bindir)/ssh-agent
-	$(INSTALL) -m 0755 -s ssh-keygen $(DESTDIR)$(bindir)/ssh-keygen
-	$(INSTALL) -m 0755 -s ssh-keyscan $(DESTDIR)$(bindir)/ssh-keyscan
-	$(INSTALL) -m 0755 -s sshd $(DESTDIR)$(sbindir)/sshd
+	$(INSTALL) -m 0755 ssh $(DESTDIR)$(bindir)/ssh
+	$(INSTALL) -m 0755 scp $(DESTDIR)$(bindir)/scp
+	$(INSTALL) -m 0755 ssh-add $(DESTDIR)$(bindir)/ssh-add
+	$(INSTALL) -m 0755 ssh-agent $(DESTDIR)$(bindir)/ssh-agent
+	$(INSTALL) -m 0755 ssh-keygen $(DESTDIR)$(bindir)/ssh-keygen
+	$(INSTALL) -m 0755 ssh-keyscan $(DESTDIR)$(bindir)/ssh-keyscan
+	$(INSTALL) -m 0755 sshd $(DESTDIR)$(sbindir)/sshd
 	if test ! -z "$(INSTALL_SSH_RAND_HELPER)" ; then \
-		$(INSTALL) -m 0755 -s ssh-rand-helper $(DESTDIR)$(libexecdir)/ssh-rand-helper ; \
+		$(INSTALL) -m 0755 ssh-rand-helper $(DESTDIR)$(libexecdir)/ssh-rand-helper ; \
 	fi
-	$(INSTALL) -m 4711 -s ssh-keysign $(DESTDIR)$(SSH_KEYSIGN)
-	@NO_SFTP@$(INSTALL) -m 0755 -s sftp $(DESTDIR)$(bindir)/sftp
-	@NO_SFTP@$(INSTALL) -m 0755 -s sftp-server $(DESTDIR)$(SFTP_SERVER)
+	$(INSTALL) -m 4711 ssh-keysign $(DESTDIR)$(SSH_KEYSIGN)
+	@NO_SFTP@$(INSTALL) -m 0755 sftp $(DESTDIR)$(bindir)/sftp
+	@NO_SFTP@$(INSTALL) -m 0755 sftp-server $(DESTDIR)$(SFTP_SERVER)
 	$(INSTALL) -m 644 ssh.1.out $(DESTDIR)$(mandir)/$(mansubdir)1/ssh.1
 	$(INSTALL) -m 644 scp.1.out $(DESTDIR)$(mandir)/$(mansubdir)1/scp.1
 	$(INSTALL) -m 644 ssh-add.1.out $(DESTDIR)$(mandir)/$(mansubdir)1/ssh-add.1
diff -uNr openssh-3.5p1.orig/configure.ac openssh-3.5p1/configure.ac
--- openssh-3.5p1.orig/configure.ac	Thu Sep 26 02:38:47 2002
+++ openssh-3.5p1/configure.ac	Mon Mar 17 14:55:00 2003
@@ -504,6 +504,9 @@
 	[
 		AC_MSG_RESULT(no)
 		AC_DEFINE(BROKEN_ONE_BYTE_DIRENT_D_NAME)
+	],
+	[AC_MSG_RESULT(yes)
+	AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to yes.])
 	]
 )
 
@@ -535,6 +538,9 @@
 				[
 					AC_MSG_RESULT(no)
 					AC_MSG_ERROR([** Incomplete or missing s/key libraries.])
+				],
+				[AC_MSG_RESULT(yes)
+				AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to yes.])
 				])
 		fi
 	]
@@ -676,6 +682,9 @@
 			AC_MSG_RESULT(no)
 			AC_DEFINE(BROKEN_SNPRINTF)
 			AC_MSG_WARN([****** Your snprintf() function is broken, complain to your vendor])
+		],
+		[AC_MSG_RESULT(yes)
+		AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to yes.])
 		]
 	)
 fi
@@ -810,6 +819,10 @@
 	[
 		AC_MSG_RESULT(not found)
 		AC_MSG_ERROR(OpenSSL version header not found.)
+	],
+	[
+		ssl_header_ver="90700f (OpenSSL 0.9.7a 19 Feb 2003)"
+		AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to $ssl_header_ver])
 	]
 )
 
@@ -843,6 +856,10 @@
 	[
 		AC_MSG_RESULT(not found)
 		AC_MSG_ERROR(OpenSSL library not found.)
+	],
+	[
+		ssl_library_ver="90700f (OpenSSL 0.9.7a 19 Feb 2003)"
+		AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to $ssl_library_ver])
 	]
 )
 
@@ -860,6 +877,10 @@
 	[
 		AC_MSG_RESULT(no)
 		AC_MSG_ERROR(Your OpenSSL headers do not match your library)
+	],
+	[
+		AC_MSG_RESULT(yes)
+		AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to yes.])
 	]
 )
 
@@ -889,6 +910,11 @@
 		# Default to use of the rand helper if OpenSSL doesn't
 		# seed itself
 		USE_RAND_HELPER=yes
+	],
+	[
+		OPENSSL_SEEDS_ITSELF=yes
+		AC_MSG_RESULT(yes)
+		AC_MSG_WARN([Cannot run test when crosscompiling, defaulted to yes.])
 	]
 )
 
@@ -1475,7 +1501,8 @@
 #else
 main() { exit(0); }
 #endif
-		], [ true ], [ AC_DEFINE(BROKEN_SNPRINTF) ]
+		], [ true ], [ AC_DEFINE(BROKEN_SNPRINTF) ],
+		[ true ]
 	)
 fi
 AC_SUBST(NO_SFTP)
@@ -1596,6 +1623,7 @@
 }
 		],
 		[ ac_cv_have_accrights_in_msghdr="yes" ],
+		[ ac_cv_have_accrights_in_msghdr="no" ],
 		[ ac_cv_have_accrights_in_msghdr="no" ]
 	)
 ])
@@ -1620,7 +1648,8 @@
 }
 		],
 		[ ac_cv_have_control_in_msghdr="yes" ],
-		[ ac_cv_have_control_in_msghdr="no" ]
+		[ ac_cv_have_control_in_msghdr="no" ],
+		[ ac_cv_have_control_in_msghdr="yes" ]
 	)
 ])
 if test "x$ac_cv_have_control_in_msghdr" = "xyes" ; then
@@ -1931,13 +1960,14 @@
 		)
 	fi
 fi
+if test "$cross_compiling" != yes; then
 AC_CHECK_FILE("/dev/ptc", 
 	[
 		AC_DEFINE_UNQUOTED(HAVE_DEV_PTS_AND_PTC)
 		have_dev_ptc=1
 	]
 )
-
+fi
 # Options from here on. Some of these are preset by platform above
 AC_ARG_WITH(mantype,
 	[  --with-mantype=man|cat|doc  Set man page type],
diff -uNr openssh-3.5p1.orig/sshd_config openssh-3.5p1/sshd_config
--- openssh-3.5p1.orig/sshd_config	Fri Sep 27 05:21:58 2002
+++ openssh-3.5p1/sshd_config	Mon Mar 17 14:55:00 2003
@@ -89,5 +89,8 @@
 #Banner /some/path
 #VerifyReverseMapping no
 
+ClientAliveInterval 15
+ClientAliveCountMax 4
+
 # override default of no subsystems
-Subsystem	sftp	/usr/libexec/sftp-server
+Subsystem	sftp	/usr/sbin/sftp-server
