--- pcmcia-cs-3.1.34/etc/config.opts.orig	Tue Jul  9 09:36:23 2002
+++ pcmcia-cs-3.1.34/etc/config.opts	Tue Jul  9 09:37:47 2002
@@ -5,7 +5,7 @@
 
 # System resources available for PCMCIA devices
 
-include port 0x100-0x4ff, port 0x800-0x8ff, port 0xc00-0xcff
+include port 0x100-0x4ff, port 0xc00-0xcff
 include memory 0xc0000-0xfffff
 include memory 0xa0000000-0xa0ffffff, memory 0x60000000-0x60ffffff
 
diff -urN pcmcia-cs-3.2.3.orig/etc/shared pcmcia-cs-3.2.3/etc/shared
--- pcmcia-cs-3.2.3.orig/etc/shared	2002-08-18 21:21:11.000000000 -0600
+++ pcmcia-cs-3.2.3/etc/shared	2003-02-14 05:05:07.000000000 -0700
@@ -2,6 +2,8 @@
 # shared 1.31 2002/08/19 03:21:11 (David Hinds)
 #
 
+umask 022
+
 usage ()
 {
     echo "usage: $0 [action] [device name]"
--- pcmcia-cs-3.2.3/etc/rc.pcmcia~	2003-06-05 19:24:49.000000000 -0600
+++ pcmcia-cs-3.2.3/etc/rc.pcmcia	2003-06-05 19:25:04.000000000 -0600
@@ -48,7 +48,6 @@
 for N in PCMCIA PCIC PCIC_OPTS CORE_OPTS CARDMGR_OPTS SCHEME ; do
     V=`eval echo '$'ENV_$N` ; if [ "$V" ] ; then eval $N=\"$V\" ; fi
 done
-if [ "$PCMCIA" -a "$PCMCIA" != "yes" ] ; then exit 0 ; fi
 
 usage()
 {
