Drop check requirement, we don't do unit tests.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura radvd-2.6.orig/configure.ac radvd-2.6/configure.ac
--- radvd-2.6.orig/configure.ac	2014-09-04 15:54:41.735038724 -0300
+++ radvd-2.6/configure.ac	2014-09-04 15:54:53.106431815 -0300
@@ -50,10 +50,6 @@
 AC_PROG_CC_C99
 AC_PROG_RANLIB
 
-if test "$arch" = "linux" ; then
-PKG_CHECK_MODULES([CHECK], [check >= 0.9.4])
-fi
-
 dnl Determine of netlink is available
 AC_MSG_CHECKING(netlink)
 AC_TRY_COMPILE([
