---
 gconf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: config/gconf.c
===================================================================
--- config.orig/gconf.c
+++ config/gconf.c
@@ -1521,7 +1521,7 @@
 	/* Determine GUI path */
 	env = getenv(SRCTREE);
 	if (env)
-		glade_file = g_strconcat(env, "/scripts/kconfig/gconf.glade", NULL);
+		glade_file = g_strconcat(env, "/package/config/gconf.glade", NULL);
 	else if (av[0][0] == '/')
 		glade_file = g_strconcat(av[0], ".glade", NULL);
 	else
