---
 conf.c   |    4 ++++
 mconf.c  |    2 +-
 qconf.cc |    2 ++
 3 files changed, 7 insertions(+), 1 deletion(-)

Index: config.new/conf.c
===================================================================
--- config.new.orig/conf.c
+++ config.new/conf.c
@@ -641,6 +641,10 @@
 			fprintf(stderr, _("\n*** Error during writing of the Buildroot configuration.\n\n"));
 			exit(1);
 		}
+		if (conf_write_autoconf()) {
+			fprintf(stderr, _("\n*** Error during update of the Buildroot configuration.\n\n"));
+			return 1;
+		}
 	}
 	return 0;
 }
Index: config.new/mconf.c
===================================================================
--- config.new.orig/mconf.c
+++ config.new/mconf.c
@@ -862,6 +862,6 @@
 			"\n\n"));
 	}
 
-	return 0;
+	return conf_write_autoconf();
 }
 
Index: config.new/qconf.cc
===================================================================
--- config.new.orig/qconf.cc
+++ config.new/qconf.cc
@@ -1612,6 +1612,7 @@
 {
 	if (!conf_get_changed()) {
 		e->accept();
+		conf_write_autoconf();
 		return;
 	}
 	QMessageBox mb("qconf", _("Save configuration?"), QMessageBox::Warning,
@@ -1622,6 +1623,7 @@
 	switch (mb.exec()) {
 	case QMessageBox::Yes:
 		conf_write(NULL);
+		conf_write_autoconf();
 	case QMessageBox::No:
 		e->accept();
 		break;
