From 354a8217eb735c9678f42436d647571fcebc3ead Mon Sep 17 00:00:00 2001
From: Maxime Hadjinlian <maxime.hadjinlian@gmail.com>
Date: Wed, 15 Jan 2014 00:46:10 +0100
Subject: [PATCH 3/3] Remove prefix hack in configure

Signed-off-by: Maxime Hadjinlian <maxime.hadjinlian@gmail.com>
---
 configure.ac | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/configure.ac b/configure.ac
index 9db3b21..9b99df4 100644
--- a/configure.ac
+++ b/configure.ac
@@ -100,15 +100,6 @@ gtk_CHECK_GTK_DOC
 dnl Check for good random number sources
 AC_CHECK_FILES(/dev/random /dev/urandom /dev/srandom /dev/arandom)
 
-dnl Dirty path hack.  Helps some people with badly set up search paths.
-if test "$prefix" = "NONE"; then
-  LDFLAGS="$LDFLAGS -L$ac_default_prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$ac_default_prefix/include"
-else
-  LDFLAGS="$LDFLAGS -L$prefix/lib"
-  CPPFLAGS="$CPPFLAGS -I$prefix/include"
-fi
-
 dnl Checks for libraries.
 ye_CHECK_LIBM
 
-- 
1.8.5.2

