diff -urN linux-2.6.22.1-atmel-0rig/arch/arm/mach-at91/at91sam9261_devices.c linux-2.6.22.1-atmel/arch/arm/mach-at91/at91sam9261_devices.c
--- linux-2.6.22.1-atmel-0rig/arch/arm/mach-at91/at91sam9261_devices.c	2007-07-10 20:56:30.000000000 +0200
+++ linux-2.6.22.1-atmel/arch/arm/mach-at91/at91sam9261_devices.c	2007-09-04 16:09:54.000000000 +0200
@@ -20,7 +20,7 @@
 #include <asm/arch/at91sam9261.h>
 #include <asm/arch/at91sam9261_matrix.h>
 #include <asm/arch/at91sam926x_mc.h>
-
+#include <video/atmel_lcdc.h>
 #include "generic.h"
 
 
