Fix dependency issue to allow parallel build

Signed-off-by: Qais Yousef <qais.yousef@imgtec.com>

Index: bellagio-0.9.3/src/Makefile.am
===================================================================
--- bellagio-0.9.3.orig/src/Makefile.am
+++ bellagio-0.9.3/src/Makefile.am
@@ -8,6 +8,7 @@ omxregister_bellagio_SOURCES = omxregist
 omxregister_bellagio_CFLAGS = -DOMXILCOMPONENTSPATH=\"$(plugindir)/\" \
 			      -I$(top_srcdir)/include
 omxregister_bellagio_LDFLAGS = -lomxil-bellagio -L$(builddir)
+omxregister_bellagio_DEPENDENCIES = libomxil-bellagio.la
 
 lib_LTLIBRARIES = libomxil-bellagio.la
 libomxil_bellagio_la_SOURCES = component_loader.h \
