Remove -Werror from CXXFLAGS

Signed-off-by: Markos Chandras <markos.chandras@imgtec.com>

Index: synergy-1.3.1/configure.in
===================================================================
--- synergy-1.3.1.orig/configure.in
+++ synergy-1.3.1/configure.in
@@ -239,7 +239,6 @@ dnl checks for system services
 
 dnl enable maximum compiler warnings and warnings are errors.
 ACX_CXX_WARNINGS
-ACX_CXX_WARNINGS_ARE_ERRORS
 
 dnl adjust compiler and linker variables
 CXXFLAGS="$CXXFLAGS $SYNERGY_CXXFLAGS $ARCH_CFLAGS"
