diff -rup linux-2.6.24-orig/arch/avr32/mach-at32ap/at32ap700x.c linux-2.6.24/arch/avr32/mach-at32ap/at32ap700x.c
--- linux-2.6.24-orig/arch/avr32/mach-at32ap/at32ap700x.c	2008-02-04 09:51:43.000000000 -0500
+++ linux-2.6.24/arch/avr32/mach-at32ap/at32ap700x.c	2008-02-04 10:24:29.000000000 -0500
@@ -1102,11 +1102,16 @@ at32_add_device_mci(unsigned int id, str
 		goto fail;
 
 	select_peripheral(PA(10), PERIPH_A, 0);	/* CLK	 */
-	select_peripheral(PA(11), PERIPH_A, 0);	/* CMD	 */
+	select_peripheral(PA(11), PERIPH_A, 0);	/* CMD1	 */
 	select_peripheral(PA(12), PERIPH_A, 0);	/* DATA0 */
 	select_peripheral(PA(13), PERIPH_A, 0);	/* DATA1 */
 	select_peripheral(PA(14), PERIPH_A, 0);	/* DATA2 */
 	select_peripheral(PA(15), PERIPH_A, 0);	/* DATA3 */
+	select_peripheral(PB(6), PERIPH_B, 0);	/* CMD2 */
+	select_peripheral(PB(7), PERIPH_B, 0);	/* DATA4 */
+	select_peripheral(PB(8), PERIPH_B, 0);	/* DATA5 */
+	select_peripheral(PB(9), PERIPH_B, 0);	/* DATA6 */
+	select_peripheral(PB(10), PERIPH_B, 0);	/* DATA7 */
 
 	if (data) {
 		if (data->detect_pin != GPIO_PIN_NONE)
Only in linux-2.6.24/arch/avr32/mach-at32ap: at32ap700x.c~
