diff -ur libpng-1.2.25/pngconf.h libpng-1.2.25-patched/pngconf.h
--- libpng-1.2.25/pngconf.h	2008-02-18 16:31:05.000000000 -0600
+++ libpng-1.2.25-patched/pngconf.h	2008-03-05 07:30:22.900478085 -0600
@@ -319,13 +319,6 @@
 #      define PNG_SAVE_BSD_SOURCE
 #      undef _BSD_SOURCE
 #    endif
-#    ifdef _SETJMP_H
-     /* If you encounter a compiler error here, see the explanation
-      * near the end of INSTALL.
-      */
-         __pngconf.h__ already includes setjmp.h;
-         __dont__ include it again.;
-#    endif
 #  endif /* __linux__ */
 
    /* include setjmp.h for error handling */
