decoders/speex: fix constness of speex mode

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN sdl_sound-1.0.3.orig/decoders/speex.c sdl_sound-1.0.3/decoders/speex.c
--- sdl_sound-1.0.3.orig/decoders/speex.c	2014-11-17 19:40:09.543553556 +0100
+++ sdl_sound-1.0.3/decoders/speex.c	2014-11-17 19:40:37.119864580 +0100
@@ -124,7 +124,7 @@
 
 static int process_header(speex_t *speex, Sound_Sample *sample)
 {
-    SpeexMode *mode;
+    const SpeexMode * mode;
     SpeexHeader *hptr;
     SpeexHeader header;
     int enh_enabled = SPEEX_USE_PERCEPTUAL_ENHANCER;
