diff -ur e2fsprogs-1.39/lib/Makefile.elf-lib e2fsprogs-1.39.fixed/lib/Makefile.elf-lib
--- e2fsprogs-1.39/lib/Makefile.elf-lib	2006-03-11 10:37:57.000000000 +1100
+++ e2fsprogs-1.39.fixed/lib/Makefile.elf-lib	2007-10-31 17:37:43.000000000 +1100
@@ -45,7 +45,7 @@
 	@echo "	SYMLINK $(ELF_INSTALL_DIR)/$(ELF_SONAME)"
 	@$(LN_S) -f $(ELF_LIB) $(DESTDIR)$(ELF_INSTALL_DIR)/$(ELF_SONAME)
 	@echo "	SYMLINK $(libdir)/$(ELF_IMAGE).so"
-	@$(LN_S) -f $(ELF_INSTALL_DIR)/$(ELF_SONAME) \
+	@$(LN_S) -f $(ELF_SONAME) \
 		$(DESTDIR)$(libdir)/$(ELF_IMAGE).so
 	@echo "	LDCONFIG"
 	@-$(LDCONFIG)
