Disable sanitizer checks, they're busted.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura libmemcached-1.0.18.orig/m4/ax_harden_compiler_flags.m4 libmemcached-1.0.18/m4/ax_harden_compiler_flags.m4
--- libmemcached-1.0.18.orig/m4/ax_harden_compiler_flags.m4	2014-11-13 08:07:09.011614311 -0300
+++ libmemcached-1.0.18/m4/ax_harden_compiler_flags.m4	2014-11-13 08:07:52.406092440 -0300
@@ -222,8 +222,6 @@
 #         GCC 4.5 removed this.
 #         _APPEND_COMPILE_FLAGS_ERROR([-Wunreachable-code])
 
-          _SET_SANITIZE_FLAGS
-
           AS_IF([test "x$ax_enable_debug" = xno],
             [AS_IF([test "x$ac_cv_vcs_checkout" = xyes],
               [AS_IF([test "x${host_os}" != "xmingw"],
@@ -337,8 +335,6 @@
                 _APPEND_COMPILE_FLAGS_ERROR([-fstack-protector-all])
                 ])])])])
 
-          _SET_SANITIZE_FLAGS
-
           AS_IF([test "x$ac_cv_warnings_as_errors" = xyes],
                 [AX_APPEND_FLAG([-Werror])])
           AC_LANG_POP([C++])
