From 3e65bbfc1c556ac8bd1d30db66d6e63957066b75 Mon Sep 17 00:00:00 2001
From: Dave Beckett <dave@dajobe.org>
Date: Sat, 31 May 2014 12:56:22 -0700
Subject: [PATCH] Oauth services are on https www.flickr.com

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 src/common.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/common.c b/src/common.c
index 245db18..c697a7d 100644
--- a/src/common.c
+++ b/src/common.c
@@ -76,9 +76,9 @@ const char* const flickcurl_version_string = VERSION;
 const char* const flickcurl_flickr_service_uri =  "https://api.flickr.com/services/rest/";
 const char* const flickcurl_flickr_upload_service_uri =  "https://up.flickr.com/services/upload/";
 const char* const flickcurl_flickr_replace_service_uri =  "https://up.flickr.com/services/replace/";
-const char* const flickcurl_flickr_oauth_request_token_uri =  "https://api.flickr.com/services/oauth/request_token";
-const char* const flickcurl_flickr_oauth_authorize_uri =  "https://api.flickr.com/services/oauth/authorize";
-const char* const flickcurl_flickr_oauth_access_token_uri =  "https://api.flickr.com/services/oauth/access_token";
+const char* const flickcurl_flickr_oauth_request_token_uri =  "https://www.flickr.com/services/oauth/request_token";
+const char* const flickcurl_flickr_oauth_authorize_uri =  "https://www.flickr.com/services/oauth/authorize";
+const char* const flickcurl_flickr_oauth_access_token_uri =  "https://www.flickr.com/services/oauth/access_token";
 
 
 static void
-- 
2.0.0

