diff -urN kexec-tools-1.101-0rig/util/Makefile kexec-tools-1.101/util/Makefile
--- kexec-tools-1.101-0rig/util/Makefile	2004-12-16 22:08:21.000000000 +0100
+++ kexec-tools-1.101/util/Makefile	2007-07-24 16:32:48.000000000 +0200
@@ -1,4 +1,4 @@
 BIN_TO_HEX:= $(OBJDIR)/bin/bin-to-hex
 $(BIN_TO_HEX): util/bin-to-hex.c
 	mkdir -p $(@D)
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(HOSTCC) $(HOST_CFLAGS) $< -o $@
diff -urN kexec-tools-1.101-0rig/util/Makefile~ kexec-tools-1.101/util/Makefile~
--- kexec-tools-1.101-0rig/util/Makefile~	1970-01-01 01:00:00.000000000 +0100
+++ kexec-tools-1.101/util/Makefile~	2004-12-16 22:08:21.000000000 +0100
@@ -0,0 +1,4 @@
+BIN_TO_HEX:= $(OBJDIR)/bin/bin-to-hex
+$(BIN_TO_HEX): util/bin-to-hex.c
+	mkdir -p $(@D)
+	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
