From 5ae1754025315d85fac11cb4eb2474789ee6475e Mon Sep 17 00:00:00 2001
From: Sam Lancia <sam@gpsm.co.uk>
Date: Sat, 7 Sep 2019 20:54:29 +0100
Subject: [PATCH] Lrzip.h: add missing header for va_list on some platforms

Signed-off-by: Sam Lancia <sam@gpsm.co.uk>
---
 Lrzip.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Lrzip.h b/Lrzip.h
index 29bc2a9..8934c59 100644
--- a/Lrzip.h
+++ b/Lrzip.h
@@ -20,6 +20,7 @@
 #ifndef LIBLRZIP_H
 #define LIBLRZIP_H
 
+#include <stdarg.h>
 #include <stdbool.h>
 #include <stdio.h>
 #ifdef _WIN32
-- 
2.17.1


