diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/armbpabi.sc binutils/ld/scripttempl/armbpabi.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/armbpabi.sc	2005-11-13 18:16:37.000000000 +0100
+++ binutils/ld/scripttempl/armbpabi.sc	2005-12-06 13:46:28.000000000 +0100
@@ -80,7 +80,7 @@ CTOR=".ctors        ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elf.sc binutils/ld/scripttempl/elf.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elf.sc	2005-11-13 18:16:37.000000000 +0100
+++ binutils/ld/scripttempl/elf.sc	2005-12-06 13:47:36.000000000 +0100
@@ -205,7 +205,7 @@ CTOR=".ctors        ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elf32sh-symbian.sc binutils/ld/scripttempl/elf32sh-symbian.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elf32sh-symbian.sc	2005-11-13 18:16:37.000000000 +0100
+++ binutils/ld/scripttempl/elf32sh-symbian.sc	2005-12-06 13:48:10.000000000 +0100
@@ -99,7 +99,7 @@ CTOR=".ctors     ALIGN(4) : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elf_chaos.sc binutils/ld/scripttempl/elf_chaos.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elf_chaos.sc	2005-08-22 21:27:45.000000000 +0200
+++ binutils/ld/scripttempl/elf_chaos.sc	2005-12-06 13:48:16.000000000 +0100
@@ -118,7 +118,7 @@ CTOR="
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elfd10v.sc binutils/ld/scripttempl/elfd10v.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elfd10v.sc	2003-05-05 23:46:50.000000000 +0200
+++ binutils/ld/scripttempl/elfd10v.sc	2005-12-06 13:48:26.000000000 +0100
@@ -23,7 +23,7 @@ CTOR=".ctors ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elfd30v.sc binutils/ld/scripttempl/elfd30v.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elfd30v.sc	2003-05-05 23:46:50.000000000 +0200
+++ binutils/ld/scripttempl/elfd30v.sc	2005-12-06 13:48:36.000000000 +0100
@@ -15,7 +15,7 @@ CTOR=".ctors ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/elfxtensa.sc binutils/ld/scripttempl/elfxtensa.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/elfxtensa.sc	2005-11-13 18:16:37.000000000 +0100
+++ binutils/ld/scripttempl/elfxtensa.sc	2005-12-06 13:48:53.000000000 +0100
@@ -176,7 +176,7 @@ CTOR=".ctors        ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/iq2000.sc binutils/ld/scripttempl/iq2000.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/iq2000.sc	2005-08-22 21:27:45.000000000 +0200
+++ binutils/ld/scripttempl/iq2000.sc	2005-12-06 13:49:38.000000000 +0100
@@ -86,7 +86,7 @@ CTOR=".ctors ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
diff -ruNp binutils-2.16.91.0.4.oorig/ld/scripttempl/xstormy16.sc binutils/ld/scripttempl/xstormy16.sc
--- binutils-2.16.91.0.4.oorig/ld/scripttempl/xstormy16.sc	2005-08-22 21:27:45.000000000 +0200
+++ binutils/ld/scripttempl/xstormy16.sc	2005-12-06 13:50:32.000000000 +0100
@@ -70,7 +70,7 @@ CTOR=".ctors ${CONSTRUCTING-0} : 
     KEEP (*crtbegin*.o(.ctors))
 
     /* We don't want to include the .ctor section from
-       from the crtend.o file until after the sorted ctors.
+       the crtend.o file until after the sorted ctors.
        The .ctor section from the crtend file contains the
        end of ctors marker and it must be last */
 
