--- strace-4.5.14/net.c.oorig	2005-07-05 01:33:38.000000000 +0200
+++ strace-4.5.14/net.c	2006-10-09 16:35:02.000000000 +0200
@@ -800,8 +800,12 @@
 
 #ifdef SOL_PACKET
 static const struct xlat sockpacketoptions[] = {
+#if defined(PACKET_ADD_MEMBERSHIP)
 	{ PACKET_ADD_MEMBERSHIP,	"PACKET_ADD_MEMBERSHIP"	},
+#endif
+#if defined(PACKET_DROP_MEMBERSHIP)
 	{ PACKET_DROP_MEMBERSHIP,	"PACKET_DROP_MEMBERSHIP"},
+#endif
 #if defined(PACKET_RECV_OUTPUT)
 	{ PACKET_RECV_OUTPUT,		"PACKET_RECV_OUTPUT"	},
 #endif
