--- strace-4.5.15.orig/configure.ac	2007-01-11 12:37:55.000000000 +0100
+++ strace-4.5.15/configure.ac	2007-01-29 17:38:26.000000000 +0100
@@ -91,14 +91,14 @@ hppa*|parisc*)
 	arch=hppa
 	AC_DEFINE([HPPA], 1, [Define for the HPPA architecture.])
 	;;
-sh)
-	arch=sh
-	AC_DEFINE([SH], 1, [Define for the SH architecture.])
-	;;
 sh64)
 	arch=sh64
 	AC_DEFINE([SH64], 1, [Define for the SH64 architecture.])
 	;;
+sh*)
+	arch=sh
+	AC_DEFINE([SH], 1, [Define for the SH architecture.])
+	;;
 x86?64*)
 	arch=x86_64
 	AC_DEFINE([X86_64], 1, [Define for the AMD x86-64 architecture.])

