diff -ur strace-4.5.15/signal.c strace-4.5.15-patched/signal.c
--- strace-4.5.15/signal.c	2007-01-11 16:08:38.000000000 -0600
+++ strace-4.5.15-patched/signal.c	2007-02-06 20:49:34.714320249 -0600
@@ -1440,7 +1440,7 @@
 	  	tcp->u_rval = tcp->u_error = 0;
 		if(tcp->u_arg[0] == 0)
 		  	return 0;
-		tcp->auxstr = sprintsigmask("mask now ", tcp->u_arg[1]);
+		tcp->auxstr = sprintsigmask("mask now ", tcp->u_arg[1], 0);
 		return RVAL_NONE | RVAL_STR;
 	}
 	return 0;
