--- a/1.0.0.rc15/lib/device/scan.c.orig	2008-06-20 09:32:05.000000000 -0500
+++ b/1.0.0.rc15/lib/device/scan.c	2008-12-01 17:15:25.000000000 -0600
@@ -69,7 +69,7 @@
 	static char *ret = NULL, *sysfs_mp;
 
 	if (!(sysfs_mp = find_sysfs_mp(lc)))
-		LOG_ERR(lc, NULL, "finding sysfs mount point");
+		{ log_notice(lc, "finding sysfs mount point"); return NULL; }
 
 	if ((ret = dbg_malloc(strlen(sysfs_mp) + strlen(path) + 1)))
 		sprintf(ret, "%s%s", sysfs_mp, path);
@@ -315,7 +315,7 @@
 	} else {
 		sysfs = 0;
 		path = (char *) _PATH_DEV;
-		log_print(lc, "carrying on with %s", path);
+		log_notice(lc, "carrying on with %s", path);
 	}
 
 	if (!(d = opendir(path))) {
