diff -rdup mdadm-2.6.1.orig/mdadm.h mdadm-2.6.1/mdadm.h
--- mdadm-2.6.1.orig/mdadm.h	2007-02-22 04:59:28.000000000 +0100
+++ mdadm-2.6.1/mdadm.h	2007-04-16 20:53:53.000000000 +0200
@@ -382,7 +382,7 @@ struct stat64;
 #define HAVE_NFTW  we assume
 #define HAVE_FTW
 
-#ifdef UCLIBC
+#if defined __UCLIBC__ || /* older did use that? */ defined UCLIBC
 # include <features.h>
 # ifndef  __UCLIBC_HAS_FTW__
 #  undef HAVE_FTW

