diff -Naur mdadm-2.6.9/Makefile mdadm-2.6.9-patched/Makefile
--- mdadm-2.6.9/Makefile	2009-03-10 07:01:17.000000000 +0100
+++ mdadm-2.6.9-patched/Makefile	2009-10-10 15:53:13.329068128 +0200
@@ -161,7 +161,9 @@
 sha1.o : sha1.c sha1.h md5.h
 	$(CC) $(CFLAGS) -DHAVE_STDINT_H -o sha1.o -c sha1.c
 
-install : mdadm install-man
+install : install-mdadm install-man
+
+install-mdadm : mdadm
 	$(INSTALL) -D $(STRIP) -m 755 mdadm $(DESTDIR)$(BINDIR)/mdadm
 
 install-static : mdadm.static install-man

