diff -Nur linux-3.14.orig/drivers/net/ethernet/xilinx/xilinx_emaclite.c linux-3.14/drivers/net/ethernet/xilinx/xilinx_emaclite.c
--- linux-3.14.orig/drivers/net/ethernet/xilinx/xilinx_emaclite.c	2014-03-31 05:40:15.000000000 +0200
+++ linux-3.14/drivers/net/ethernet/xilinx/xilinx_emaclite.c	2014-04-11 22:53:42.000000000 +0200
@@ -1249,6 +1249,7 @@
 	{ .compatible = "xlnx,opb-ethernetlite-1.01.b", },
 	{ .compatible = "xlnx,xps-ethernetlite-1.00.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-2.00.a", },
+	{ .compatible = "xlnx,xps-ethernetlite-2.00.b", },
 	{ .compatible = "xlnx,xps-ethernetlite-2.01.a", },
 	{ .compatible = "xlnx,xps-ethernetlite-3.00.a", },
 	{ /* end of list */ },
