From ac3ffd15e1cf16059b4ccd90b7c816b078a621a5 Mon Sep 17 00:00:00 2001
From: Mark Simulacrum <mark.simulacrum@gmail.com>
Date: Wed, 20 Jun 2018 19:47:00 -0600
Subject: [PATCH] Fix warnings in library from error-chain

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---
 src/lib.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git rust-1.27.1.orig/src/tools/rust-installer/src/lib.rs rust-1.27.1/src/tools/rust-installer/src/lib.rs
index ff0495b..cc5e740 100644
--- rust-1.27.1.orig/src/tools/rust-installer/src/lib.rs
+++ rust-1.27.1/src/tools/rust-installer/src/lib.rs
@@ -22,6 +22,7 @@ extern crate winapi;
 #[macro_use]
 extern crate lazy_static;
 
+#[allow(warnings)]
 mod errors {
     error_chain!{
         foreign_links {
-- 
2.11.0

