From d90cef58439bcd14426d64b16599e060cd6464f9 Mon Sep 17 00:00:00 2001
From: Mark Simulacrum <mark.simulacrum@gmail.com>
Date: Wed, 20 Jun 2018 16:54:40 -0600
Subject: [PATCH] Permit warnings inside error-chain expansion

Signed-off-by: Eric Le Bihan <eric.le.bihan.dev@free.fr>
---
 src/main.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git rust-1.27.1.orig/src/tools/rust-installer/src/main.rs rust-1.27.1/src/tools/rust-installer/src/main.rs
index cabffb8..007ed71 100644
--- rust-1.27.1.orig/src/tools/rust-installer/src/main.rs
+++ rust-1.27.1/src/tools/rust-installer/src/main.rs
@@ -7,6 +7,7 @@ extern crate installer;
 use errors::*;
 use clap::{App, ArgMatches};
 
+#[allow(warnings)]
 mod errors {
     error_chain!{
         links {
-- 
2.11.0

