--- a/stream/stream_dvb.c
+++ b/stream/stream_dvb.c
@@ -37,9 +37,7 @@ Foundation, Inc., 675 Mass Ave, Cambridg
 #include <sys/poll.h>
 #include <unistd.h>
 #include <fcntl.h>
-#include <string.h>
 #include <errno.h>
-#include <fcntl.h>
 
 #include "stream.h"
 #include "libmpdemux/demuxer.h"
@@ -168,7 +166,7 @@ static dvb_channels_list *dvb_get_channe
 		if((line[0] == '#') || (strlen(line) == 0))
 			continue;
 
-		colon = index(line, ':');
+		colon = strchr(line, ':');
 		if(colon)
 		{
 			k = colon - line;
