Index: config.sub
===================================================================
RCS file: /cvsroot/config/config/config.sub,v
retrieving revision 1.336
diff -u -p -r1.336 config.sub
--- config.sub	22 Dec 2005 23:50:21 -0000	1.336
+++ config.sub	30 Dec 2005 23:37:29 -0000
@@ -298,6 +298,10 @@ case $basic_machine in
 	ms1)
 		basic_machine=mt-unknown
 		;;
+	nios2 | nios2-* | nios2 | nios2-*)
+		basic_machine=nios2-altera
+		os=-none
+		;;
 
 	# We use `pc' rather than `unknown'
 	# because (1) that's what they normally are, and
