--- openssh-3.6.1p1/Makefile.in.orig	2003-03-20 17:34:34.000000000 -0700
+++ openssh-3.6.1p1/Makefile.in	2003-04-25 17:09:00.000000000 -0600
@@ -27,7 +27,7 @@
 RAND_HELPER=$(libexecdir)/ssh-rand-helper
 PRIVSEP_PATH=@PRIVSEP_PATH@
 SSH_PRIVSEP_USER=@SSH_PRIVSEP_USER@
-STRIP_OPT=@STRIP_OPT@
+STRIP_OPT=
 
 PATHS= -DSSHDIR=\"$(sysconfdir)\" \
 	-D_PATH_SSH_PROGRAM=\"$(SSH_PROGRAM)\" \

--- openssh-4.6p1.oorig/sshd_config	2006-07-24 06:06:47.000000000 +0200
+++ openssh-4.6p1/sshd_config	2007-06-28 15:58:49.000000000 +0200
@@ -83,9 +83,11 @@
 #AllowTcpForwarding yes
 #GatewayPorts no
 #X11Forwarding no
+X11Forwarding yes
 #X11DisplayOffset 10
 #X11UseLocalhost yes
 #PrintMotd yes
+PrintMotd no
 #PrintLastLog yes
 #TCPKeepAlive yes
 #UseLogin no
@@ -102,8 +104,12 @@
 # no default banner path
 #Banner none
 
+ClientAliveInterval 15
+ClientAliveCountMax 4
+AcceptEnv LANG LC_*
+
 # override default of no subsystems
 Subsystem	sftp	/usr/libexec/sftp-server
 
