diff -ru openssh-6.6p1.org/channels.c openssh-6.6p1/channels.c
--- openssh-6.6p1.org/channels.c	2014-02-27 00:18:33.000000000 +0100
+++ openssh-6.6p1/channels.c	2014-09-12 05:33:43.791424676 +0200
@@ -41,6 +41,7 @@
 
 #include "includes.h"
 
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/un.h>
diff -ru openssh-6.6p1.org/openbsd-compat/bsd-poll.c openssh-6.6p1/openbsd-compat/bsd-poll.c
--- openssh-6.6p1.org/openbsd-compat/bsd-poll.c	2014-02-06 00:44:13.000000000 +0100
+++ openssh-6.6p1/openbsd-compat/bsd-poll.c	2014-09-12 05:39:35.580947451 +0200
@@ -19,6 +19,7 @@
 #include "includes.h"
 #if !defined(HAVE_POLL)
 
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/time.h>
 #ifdef HAVE_SYS_SELECT_H
diff -ru openssh-6.6p1.org/sshconnect.c openssh-6.6p1/sshconnect.c
--- openssh-6.6p1.org/sshconnect.c	2014-02-06 23:24:34.000000000 +0100
+++ openssh-6.6p1/sshconnect.c	2014-09-12 05:39:57.372236592 +0200
@@ -15,6 +15,7 @@
 
 #include "includes.h"
 
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <sys/stat.h>
diff -ru openssh-6.6p1.org/sshd.c openssh-6.6p1/sshd.c
--- openssh-6.6p1.org/sshd.c	2014-02-27 00:20:08.000000000 +0100
+++ openssh-6.6p1/sshd.c	2014-09-12 05:42:20.233576519 +0200
@@ -44,6 +44,7 @@
 
 #include "includes.h"
 
+#include <sys/param.h>
 #include <sys/types.h>
 #include <sys/ioctl.h>
 #include <sys/socket.h>
diff -ru openssh-6.6p1.org/ssh-keyscan.c openssh-6.6p1/ssh-keyscan.c
--- openssh-6.6p1.org/ssh-keyscan.c	2013-12-07 01:24:02.000000000 +0100
+++ openssh-6.6p1/ssh-keyscan.c	2014-09-12 05:41:03.102092462 +0200
@@ -10,6 +10,7 @@
 #include "includes.h"
  
 #include "openbsd-compat/sys-queue.h"
+#include <sys/param.h>
 #include <sys/resource.h>
 #ifdef HAVE_SYS_TIME_H
 # include <sys/time.h>
diff -ru openssh-6.6p1.org/ssh-pkcs11-helper.c openssh-6.6p1/ssh-pkcs11-helper.c
--- openssh-6.6p1.org/ssh-pkcs11-helper.c	2013-12-05 00:23:21.000000000 +0100
+++ openssh-6.6p1/ssh-pkcs11-helper.c	2014-09-12 05:40:20.488482522 +0200
@@ -17,6 +17,7 @@
 
 #include "includes.h"
 
+#include <sys/param.h>
 #include <sys/types.h>
 #ifdef HAVE_SYS_TIME_H
 # include <sys/time.h>
