--- a/configure.in	2008-01-21 21:12:10.000000000 -0500
+++ b/configure.in	2008-02-09 13:32:20.000000000 -0500
@@ -318,11 +318,6 @@ if test "$pango_os_win32" = "yes"; then
 fi
 
 #
-# Checks for ATSUI
-#
-AC_CHECK_HEADER(Carbon/Carbon.h, [have_atsui=true], [have_atsui=true])
-
-#
 # Checks for Cairo
 #
 have_cairo=false

--- a/configure.in	2008-02-09 13:37:25.000000000 -0500
+++ b/configure.in	2008-02-09 13:39:42.000000000 -0500
@@ -374,12 +374,6 @@ if $have_cairo ; then
     have_cairo=true
   fi
 
-  PKG_CHECK_EXISTS(cairo-atsui >= $cairo_required, have_cairo_atsui=true, :)
-  if $have_cairo_atsui && $have_atsui ; then
-    AC_DEFINE(HAVE_CAIRO_ATSUI, 1, [Whether Cairo uses ATSUI for fonts])
-    have_cairo=true
-  fi
-
   if $have_cairo ; then
     AC_DEFINE(HAVE_CAIRO, 1, [Have usable Cairo library and font backend])
   fi

--- a/configure	2008-02-09 13:25:54.000000000 -0500
+++ b/configure	2008-02-09 13:27:56.000000000 -0500
@@ -22516,7 +22516,7 @@ fi
   if $have_cairo_atsui && $have_atsui ; then
 
 cat >>confdefs.h <<\_ACEOF
-#define HAVE_CAIRO_ATSUI 1
+/* #undef HAVE_CAIRO_ATSUI */
 _ACEOF
 
     have_cairo=true
