--- a/Makefile.in	2008-01-21 22:28:49.000000000 -0500
+++ b/Makefile.in	2008-02-09 13:02:41.000000000 -0500
@@ -260,7 +260,7 @@ sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = pango modules pango-view examples docs tools tests
+SUBDIRS = pango modules pango-view examples docs tools 
 EXTRA_DIST = \
 	autogen.sh		\
 	pango.pc.in		\

--- a/Makefile.am	2008-01-17 16:37:49.000000000 -0500
+++ b/Makefile.am	2008-02-09 13:35:13.000000000 -0500
@@ -1,6 +1,6 @@
 ## Process this file with automake to create Makefile.in.
 
-SUBDIRS= pango modules pango-view examples docs tools tests
+SUBDIRS= pango modules pango-view examples docs tools 
 
 EXTRA_DIST = 			\
 	autogen.sh		\
