--- xf86-video-savage-2.1.2/configure.ac.orig	2007-06-10 11:51:23.000000000 +0200
+++ xf86-video-savage-2.1.2/configure.ac	2007-06-10 11:52:46.000000000 +0200
@@ -68,12 +68,18 @@
 AC_HEADER_STDC
 
 if test "$DRI" != no; then
+	if test "$cross_compiling" = no; then
         AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
         AC_CHECK_FILE([${sdkdir}/sarea.h],
                       [have_sarea_h="yes"], [have_sarea_h="no"])
         AC_CHECK_FILE([${sdkdir}/dristruct.h],
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
+	else
+		have_dri_h="yes"
+		have_sarea_h="yes"
+		have_dristruct_h="yes"
+	fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
