---
 Make.Rules      |   15 ++++++++-------
 libcap/Makefile |    2 +-
 2 files changed, 9 insertions(+), 8 deletions(-)

Index: libcap-2.19/Make.Rules
===================================================================
--- libcap-2.19.orig/Make.Rules
+++ libcap-2.19/Make.Rules
@@ -45,24 +45,25 @@
 KERNEL_HEADERS := $(topdir)/libcap/include
 IPATH += -I$(topdir)/libcap/include -I$(KERNEL_HEADERS)
 
-CC := gcc
-CFLAGS := -O2
-BUILD_CC := $(CC)
-BUILD_CFLAGS := $(CFLAGS) $(IPATH)
-AR := ar
-RANLIB := ranlib
+CC ?= gcc
+CFLAGS ?= -O2
+CC_FOR_BUILD ?= $(CC)
+CFLAGS_FOR_BUILD ?= $(CFLAGS)
+AR ?= ar
+RANLIB ?= ranlib
 DEBUG = -g #-DDEBUG
 WARNINGS=-fPIC -Wall -Wwrite-strings \
         -Wpointer-arith -Wcast-qual -Wcast-align \
         -Wstrict-prototypes -Wmissing-prototypes \
         -Wnested-externs -Winline -Wshadow
 LD=$(CC) -Wl,-x -shared
-LDFLAGS := #-g
+LDFLAGS ?= #-g
 
 SYSTEM_HEADERS = /usr/include
 INCS=$(topdir)/libcap/include/sys/capability.h
 LDFLAGS += -L$(topdir)/libcap
 CFLAGS += -Dlinux $(WARNINGS) $(DEBUG) $(IPATH)
+CFLAGS_FOR_BUILD += $(IPATH)
 PAM_CAP := $(shell if [ -f /usr/include/security/pam_modules.h ]; then echo yes ; else echo no ; fi)
 INDENT := $(shell if [ -n "$(which indent 2>/dev/null)" ]; then echo "| indent -kr" ; fi)
 DYNAMIC := $(shell if [ ! -d "$(topdir)/.git" ]; then echo yes; fi)
Index: libcap-2.19/libcap/Makefile
===================================================================
--- libcap-2.19.orig/libcap/Makefile
+++ libcap-2.19/libcap/Makefile
@@ -33,7 +33,7 @@
 endif
 
 _makenames: _makenames.c cap_names.list.h
-	$(BUILD_CC) $(BUILD_CFLAGS) $< -o $@
+	$(CC_FOR_BUILD) $(CFLAGS_FOR_BUILD) $< -o $@
 
 cap_names.h: _makenames
 	./_makenames > cap_names.h
