Fix compilation with Xorg xserver 1.16.

Signed-off-by: Jérôme Pouiller <jezz@sysmic.org>

--- a/EXA/src/vivante_fbdev/vivante_fbdev_driver.c	2014-07-25 15:38:24.000000000 +0200
+++ b/EXA/src/vivante_fbdev/vivante_fbdev_driver.c	2014-10-02 21:00:58.755409342 +0200
@@ -261,7 +261,7 @@
         setupDone = TRUE;
         xf86AddDriver(&FBDEV, module, HaveDriverFuncs);
         if(gVivFb)
-            LoadExtension(&VIVExt, FALSE);
+            LoadExtensionList(&VIVExt, 1, FALSE);
 		return (pointer)1;
     } else {
         if (errmaj) *errmaj = LDR_ONCEONLY;
