From 8cf5909c6f8e0a892155dea14a07adfad445762d Mon Sep 17 00:00:00 2001
From: Haavard Skinnemoen <hskinnemoen@atmel.com>
Date: Sat, 23 Jun 2007 18:34:09 +0200
Subject: [PATCH] Include usb/ch9.h instead of usb_ch9.h

Signed-off-by: Haavard Skinnemoen <hskinnemoen@atmel.com>
---
 usb.c       |    2 +-
 usbstring.c |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/usb.c b/usb.c
index ea6c7de..b161154 100644
--- a/usb.c
+++ b/usb.c
@@ -35,7 +35,7 @@
 
 #include <linux/types.h>
 #include <linux/usb_gadgetfs.h>
-#include <linux/usb_ch9.h>
+#include <linux/usb/ch9.h>
 
 #ifdef	AIO
 /* this aio code works with libaio-0.3.106 */
diff --git a/usbstring.c b/usbstring.c
index 4bf2965..993acc2 100644
--- a/usbstring.c
+++ b/usbstring.c
@@ -11,7 +11,7 @@
 #include <string.h>
 
 #include <linux/types.h>
-#include <linux/usb_ch9.h>
+#include <linux/usb/ch9.h>
 
 #include "usbstring.h"
 
-- 
1.5.2.2

