From 9dfac0d55a31be59a856bad9d3a5071d65176597 Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Sat, 11 Jan 2014 21:49:22 +0100
Subject: [PATCH 5/6] cmake: lapack: add library and header install rules

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 SRC/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/SRC/CMakeLists.txt b/SRC/CMakeLists.txt
index ac4cce3..aa60555 100644
--- a/SRC/CMakeLists.txt
+++ b/SRC/CMakeLists.txt
@@ -377,4 +377,5 @@ if(BUILD_COMPLEX16)
 endif()
 add_library(lapack ${ALLOBJ} ${ALLXOBJ})
 target_link_libraries(lapack blas)
-
+install(TARGETS lapack LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
+install(FILES "${CLAPACK_SOURCE_DIR}/INCLUDE/clapack.h" DESTINATION include)
-- 
1.8.5.3

