From http://redmine.lighttpd.net/issues/2515

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

--- a/src/mod_extforward.c	(revision 2909)
+++ b/src/mod_extforward.c	(working copy)
@@ -439,7 +439,6 @@
 #ifdef HAVE_IPV6
 		ipstr_to_sockaddr(srv, real_remote_addr, &sock);
 #else
-		UNUSED(addrs_left);
 		sock.ipv4.sin_addr.s_addr = inet_addr(real_remote_addr);
 		sock.plain.sa_family = (sock.ipv4.sin_addr.s_addr == 0xFFFFFFFF) ? AF_UNSPEC : AF_INET;
 #endif
