---
 configure.ac |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: libmpd-0.17.0/configure.ac
===================================================================
--- libmpd-0.17.0.orig/configure.ac
+++ libmpd-0.17.0/configure.ac
@@ -25,7 +25,7 @@
 AC_SUBST(EXTRA_CFLAGS)
 AC_SUBST(EXTRA_LDFLAGS)   
 
-PKG_CHECK_MODULES([glib], glib-2.0 >= 2.16)
+PKG_CHECK_MODULES([glib], glib-2.0 >= 2.15)
 AC_SUBST(glib_LIBS)
 AC_SUBST(glib_CFLAGS)
 
