Add necessary definitions for AArch64

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: xserver_xorg-server-1.9.4/include/servermd.h
===================================================================
--- xserver_xorg-server-1.9.4.orig/include/servermd.h	2009-11-04 20:31:46.000000000 +0100
+++ xserver_xorg-server-1.9.4/include/servermd.h	2012-12-28 18:16:55.000000000 +0100
@@ -68,6 +68,14 @@
  * GLYPHPADBYTES is used.
  */
 
+#ifdef __aarch64__
+
+# define IMAGE_BYTE_ORDER	LSBFirst
+# define BITMAP_BIT_ORDER       LSBFirst
+# define GLYPHPADBYTES		4
+
+#endif
+
 #ifdef __avr32__
 
 #define IMAGE_BYTE_ORDER        MSBFirst
