Correct the directory path for gc_hal_eglplatform_type.h

Signed-off-by: Henk Fijnvandraat <h.fijnvandraat@inter.nl.net>

diff -rup gpu-viv-bin-mx6q-1.1.0-orig/usr/include/gc_vdk_types.h gpu-viv-bin-mx6q-1.1.0/usr/include/gc_vdk_types.h
--- gpu-viv-bin-mx6q-1.1.0-orig/usr/include/gc_vdk_types.h	2012-12-18 10:35:55.000000000 +0100
+++ gpu-viv-bin-mx6q-1.1.0/usr/include/gc_vdk_types.h	2013-05-09 21:09:28.080138681 +0200
@@ -39,7 +39,7 @@ extern "C" {
 #endif
 
 #include <EGL/egl.h>
-#include "gc_hal_eglplatform_type.h"
+#include <HAL/gc_hal_eglplatform_type.h>
 
 
 /*******************************************************************************
