diff -urN coreutils-5.3.0/lib/regex.c coreutils-5.3.0-patched/lib/regex.c
--- coreutils-5.3.0/lib/regex.c	2004-10-05 01:30:48.000000000 -0500
+++ coreutils-5.3.0-patched/lib/regex.c	2005-12-08 00:40:30.500238000 -0600
@@ -8179,7 +8179,7 @@
       if (msg_size > errbuf_size)
         {
 #if defined HAVE_MEMPCPY || defined _LIBC
-	  *((char *) __mempcpy (errbuf, msg, errbuf_size - 1)) = '\0';
+	  *((char *) mempcpy (errbuf, msg, errbuf_size - 1)) = '\0';
 #else
           memcpy (errbuf, msg, errbuf_size - 1);
           errbuf[errbuf_size - 1] = 0;
