diff -Nura rp-pppoe-3.11.orig/src/configure rp-pppoe-3.11/src/configure
--- rp-pppoe-3.11.orig/src/configure	2012-08-21 10:01:10.536440032 -0300
+++ rp-pppoe-3.11/src/configure	2012-08-21 10:01:45.353442397 -0300
@@ -4517,7 +4517,7 @@
 modprobe pppoe > /dev/null 2>&1
 fi
 if test "$cross_compiling" = yes; then :
-  ac_cv_linux_kernel_pppoe=no; $ECHO "cross-compiling, default: "
+  ac_cv_linux_kernel_pppoe=yes; $ECHO "cross-compiling, default: "
 else
   cat confdefs.h - <<_ACEOF >conftest.$ac_ext
 /* end confdefs.h.  */
diff -Nura rp-pppoe-3.11.orig/src/pppoe-server.c rp-pppoe-3.11/src/pppoe-server.c
--- rp-pppoe-3.11.orig/src/pppoe-server.c	2012-08-21 10:01:10.535440032 -0300
+++ rp-pppoe-3.11/src/pppoe-server.c	2012-08-21 10:02:02.597443569 -0300
@@ -20,6 +20,9 @@
 
 #include "config.h"
 
+/* Patched hack to make this cross compile */
+#define HAVE_LINUX_KERNEL_PPPOE 1
+
 #if defined(HAVE_NETPACKET_PACKET_H) || defined(HAVE_LINUX_IF_PACKET_H)
 #define _POSIX_SOURCE 1 /* For sigaction defines */
 #endif
