--- rp-pppoe-3.10.orig/src/configure.in	2006-04-02 16:29:42.000000000 +0200
+++ rp-pppoe-3.10/src/configure.in	2007-08-24 11:51:04.000000000 +0200
@@ -231,6 +231,7 @@ esac
 
 dnl Figure out packing order of structures
 AC_MSG_CHECKING([packing order of bit fields])
+if test "x$rpppoe_cv_pack_bitfields" = "x" ; then
 AC_TRY_RUN([
 union foo {
     struct bar {
@@ -254,8 +255,8 @@ main(void)
 	return 2;
     }
 }], rpppoe_cv_pack_bitfields=normal, rpppoe_cv_pack_bitfields=rev,
-$ECHO "no defaults for cross-compiling"; exit 0)
-
+$ECHO "no defaults for cross-compiling")
+fi
 if test "$rpppoe_cv_pack_bitfields" = "rev" ; then
 	AC_MSG_RESULT(reversed)
 	AC_DEFINE(PACK_BITFIELDS_REVERSED)
--- rp-pppoe-3.10.orig/src/configure	2010-08-20 10:46:39.000000000 -0700
+++ rp-pppoe-3.10/src/configure	2010-08-20 11:02:16.000000000 -0700
@@ -6272,7 +6272,9 @@ esac
 { echo "$as_me:$LINENO: checking packing order of bit fields" >&5
 echo $ECHO_N "checking packing order of bit fields... $ECHO_C" >&6; }
 if test "$cross_compiling" = yes; then
-  $ECHO "no defaults for cross-compiling"; exit 0
+if test "x$rpppoe_cv_pack_bitfields" = "x" ; then
+  $ECHO "no defaults for cross-compiling"
+fi
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
