---
 python/Makefile.am |    1 +
 1 file changed, 1 insertion(+)

Index: libdnet-1.11/python/Makefile.am
===================================================================
--- libdnet-1.11.orig/python/Makefile.am
+++ libdnet-1.11/python/Makefile.am
@@ -11,6 +11,7 @@
 	pyrexc $(srcdir)/dnet.pyx
 
 python-build: $(srcdir)/dnet.c
+	CC="$(CC)" LDSHARED="$(CC) -shared" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" $(PYTHON) setup.py build_ext -I$(PYINCDIR) -L$(PYLIBDIR)
 	$(PYTHON) setup.py build
 	touch python-build
 
