From 5af7ad745323004984287e48b42712e7305de35c Mon Sep 17 00:00:00 2001
From: Per Hedeland <per@hedeland.org>
Date: Fri, 13 Dec 2013 14:43:06 +0100
Subject: [PATCH] Make --maxvars option work

From upstream: https://gitorious.org/libxslt/libxslt/commit/5af7ad745323004984287e48b42

Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 xsltproc/xsltproc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/xsltproc/xsltproc.c b/xsltproc/xsltproc.c
index b22df37..66d8cbb 100644
--- a/xsltproc/xsltproc.c
+++ b/xsltproc/xsltproc.c
@@ -803,6 +803,10 @@ main(int argc, char **argv)
             (!strcmp(argv[i], "--maxdepth"))) {
             i++;
             continue;
+        } else if ((!strcmp(argv[i], "-maxvars")) ||
+            (!strcmp(argv[i], "--maxvars"))) {
+            i++;
+            continue;
         } else if ((!strcmp(argv[i], "-maxparserdepth")) ||
             (!strcmp(argv[i], "--maxparserdepth"))) {
             i++;
-- 
2.1.0

