--- linux-2.6.20.oorig/net/ipsec/ipsec_alg_cryptoapi.c	2007-02-15 12:30:41.000000000 +0100
+++ linux-2.6.20/net/ipsec/ipsec_alg_cryptoapi.c	2007-02-15 13:47:07.000000000 +0100
@@ -197,7 +197,7 @@ static struct ipsec_alg_capi_cipher alg_
  */
 int setup_cipher(const char *ciphername)
 {
-	return crypto_alg_available(ciphername, 0);
+	return crypto_has_alg(ciphername, 0, CRYPTO_ALG_ASYNC);
 }
 
 /*
