diff -ur gettext-0.14.6/gettext-tools/lib/error.h gettext-0.14.6-patched/gettext-tools/lib/error.h
--- gettext-0.14.6/gettext-tools/lib/error.h	2005-05-20 16:03:42.000000000 -0500
+++ gettext-0.14.6-patched/gettext-tools/lib/error.h	2007-01-13 20:57:24.422168053 -0600
@@ -50,7 +50,7 @@
 /* If NULL, error will flush stdout, then print on stderr the program
    name, a colon and a space.  Otherwise, error will call this
    function without parameters instead.  */
-extern DLL_VARIABLE void (*error_print_progname) (void);
+void (*error_print_progname) (void);
 
 /* This variable is incremented each time `error' is called.  */
 extern DLL_VARIABLE unsigned int error_message_count;
