Honour LDFLAGS, especially required when static linking

Signed-off-by: Waldemar Brodkorb <wbx@openadk.org>

diff -Nur autossh-1.4e.orig/Makefile.in autossh-1.4e/Makefile.in
--- autossh-1.4e.orig/Makefile.in	2015-02-10 05:41:58.000000000 +0100
+++ autossh-1.4e/Makefile.in	2016-11-25 12:49:06.052122855 +0100
@@ -31,7 +31,7 @@
 
 
 $(TARGET):	$(OFILES)
-		$(CC) $(CPPFLAGS) -o $(TARGET) $(OFILES) $(LIBS)
+		$(CC) $(CPPFLAGS) $(LDFLAGS) -o $(TARGET) $(OFILES) $(LIBS)
 
 clean:
 		- /bin/rm -f *.o *.a *.core *~
