diff -durN sslh-v1.18.orig/genver.sh sslh-v1.18/genver.sh
--- sslh-v1.18.orig/genver.sh	2017-02-21 22:03:40.849161283 +0100
+++ sslh-v1.18/genver.sh	2017-02-21 22:03:59.173402356 +0100
@@ -25,7 +25,7 @@
         fi
 fi
 
-if [ -d .git ] && head=`git rev-parse --verify HEAD 2>/dev/null`; then
+if [ -d .git ] && head=`git rev-parse --verify HEAD 2>/dev/null`; then
 	# generate the version info based on the tag
 	release=`(git describe --tags || git --describe || git describe --all --long) \
 		2>/dev/null | tr -d '\n'`
