Make.rules: don't build .so libraries when static=1

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: b/Make.rules
===================================================================
--- a/Make.rules
+++ b/Make.rules
@@ -9,7 +9,11 @@
 CFLAGS_LIB ?= -fPIC
 CFLAGS += $(CFLAGS_LIB)
 
+ifeq ($(static),1)
+libraries = $(lib_name).a
+else
 libraries = $(lib_name).so $(lib_name).a
+endif
 
 .PHONY: library
 
