From 03bab84ca3f102274837e83ee6da4c997a9da018 Mon Sep 17 00:00:00 2001
From: Tzu-Jung Lee <tjlee@ambarella.com>
Date: Fri, 12 Jul 2013 20:00:57 +0800
Subject: [PATCH] build: modify hardcoded gcc to support buildroot

Signed-off-by: Tzu-Jung Lee <tjlee@ambarella.com>

diff --git a/src/Makefile b/src/Makefile
index 026b6ba..b87ae9f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -5,10 +5,10 @@ OBJFILES := main.o lookup.o\
 	 lookup_bfd.o lookup_kas.o
 
 dropwatch: $(OBJFILES) 
-	gcc -g -o dropwatch $(OBJFILES) $(LDFLAGS) 
+	$(CC) -g -o dropwatch $(OBJFILES) $(LDFLAGS)
 
 %.o: %.c
-	gcc $(CFLAGS) $<
+	$(CC) $(CFLAGS) $<
 clean:
 	rm -f dropwatch *.o
 
-- 
1.8.3.2

