Index: openssl/ssl/t1_lib.c
RCS File: /v/openssl/cvs/openssl/ssl/t1_lib.c,v
rcsdiff -q -kk '-r1.64.2.15' '-r1.64.2.16' -u '/v/openssl/cvs/openssl/ssl/t1_lib.c,v' 2>/dev/null
--- t1_lib.c	2010/11/16 13:26:24	1.64.2.15
+++ t1_lib.c	2010/11/16 22:41:07	1.64.2.16
@@ -779,8 +779,8 @@
 				{
 				if(s->session->tlsext_ecpointformatlist)
 					{
-					*al = TLS1_AD_DECODE_ERROR;
-					return 0;
+					OPENSSL_free(s->session->tlsext_ecpointformatlist);
+					s->session->tlsext_ecpointformatlist = NULL;
 					}
 				s->session->tlsext_ecpointformatlist_length = 0;
 				if ((s->session->tlsext_ecpointformatlist = OPENSSL_malloc(ecpointformatlist_length)) == NULL)
