--- busybox-1.22.1/procps/top.c
+++ busybox-1.22.1-ash/procps/top.c
@@ -917,17 +917,12 @@ enum {
 #if ENABLE_FEATURE_USE_TERMIOS
 static unsigned handle_input(unsigned scan_mask, unsigned interval)
 {
-	struct pollfd pfd[1];
-
 	if (option_mask32 & OPT_EOF) {
 		/* EOF on stdin ("top </dev/null") */
 		sleep(interval);
 		return scan_mask;
 	}
 
-	pfd[0].fd = 0;
-	pfd[0].events = POLLIN;
-
 	while (1) {
 		int32_t c;
 
