diff --git a/src/lfs.c b/src/lfs.c
index 85f01e3..ed49f50 100644
--- a/src/lfs.c
+++ b/src/lfs.c
@@ -22,13 +22,13 @@
 
 #ifndef _WIN32
 #ifndef _AIX
-#define _FILE_OFFSET_BITS 64 /* Linux, Solaris and HP-UX */
+//#define _FILE_OFFSET_BITS 64 /* Linux, Solaris and HP-UX */
 #else
 #define _LARGE_FILES 1 /* AIX */
 #endif
 #endif
 
-#define _LARGEFILE64_SOURCE
+//#define _LARGEFILE64_SOURCE
 
 #include <errno.h>
 #include <stdio.h>
