--- a/client/mount.cifs.c	2011-09-13 10:26:21.000000000 +0200
+++ b/client/mount.cifs.c	2011-09-13 10:27:16.000000000 +0200
@@ -39,7 +39,6 @@
 #include <mntent.h>
 #include <fcntl.h>
 #include <limits.h>
-#include <fstab.h>
 #include "mount.h"
 
 #define MOUNT_CIFS_VERSION_MAJOR "1"
@@ -255,6 +254,10 @@
 	return 0;
 }
 #else /* CIFS_LEGACY_SETUID_CHECK */
+
+#ifndef _PATH_FSTAB
+#define _PATH_FSTAB "/etc/fstab"
+#endif
 static int
 check_fstab(const char *progname, char *mountpoint, char *devname,
 	    char **options)
