--- a/source/client/mount.cifs.c
+++ b/source/client/mount.cifs.c
@@ -138,6 +138,7 @@ static size_t strlcat(char *d, const cha
 	return ret;
 }
 #endif
+#endif /* __GLIBC__ && !(__UCLIBC__ && __USE_BSD) */
 
 /* BB finish BB
 
@@ -178,7 +179,6 @@ static void mount_cifs_usage(void)
 	SAFE_FREE(mountpassword);
 	exit(EX_USAGE);
 }
-#endif /* __GLIBC__ && !(__UCLIBC__ && __USE_BSD) */
 
 /* caller frees username if necessary */
 static char * getusername(void) {
