--- a/source/client/mount.cifs.c	2009-04-01 13:48:54.000000000 +0200
+++ b/source/client/mount.cifs.c	2009-04-20 12:59:57.000000000 +0200
@@ -100,6 +100,7 @@
 
 /* glibc doesn't have strlcpy, strlcat. Ensure we do. JRA. We
  * don't link to libreplace so need them here. */
+#if defined(__GLIBC__) && !(defined(__UCLIBC__) && defined(__USE_BSD))
 
 /* like strncpy but does not 0 fill the buffer and always null
  *    terminates. bufsize is the size of the destination buffer */
@@ -181,6 +182,7 @@
 	SAFE_FREE(mountpassword);
 	exit(EX_USAGE);
 }
+#endif /* __GLIBC__ && !(__UCLIBC__ && __USE_BSD) */
 
 /* caller frees username if necessary */
 static char * getusername(void) {
