--- libXt-1.0.4/configure.ac.orig	2006-11-08 15:15:56.000000000 +0100
+++ libXt-1.0.4/configure.ac	2007-06-08 16:41:59.000000000 +0200
@@ -37,6 +37,16 @@
 AC_PROG_CC
 AC_PROG_LIBTOOL
 
+AC_MSG_CHECKING([for CC_FOR_BUILD])
+if test x$host != x$build ; then
+    CC_FOR_BUILD=${CC_FOR_BUILD-gcc}
+else
+    CC_FOR_BUILD=${CC}
+fi
+AC_MSG_RESULT([$CC_FOR_BUILD])
+AC_SUBST(CC_FOR_BUILD)
+
+
 PKG_CHECK_MODULES(XT, sm x11 xproto kbproto)
 
 # Needed for including Xalloca.h
--- libXt-1.0.4/util/Makefile.am.orig	2006-10-27 21:10:56.000000000 +0200
+++ libXt-1.0.4/util/Makefile.am	2007-06-08 16:44:13.000000000 +0200
@@ -4,6 +4,16 @@
 noinst_PROGRAMS = makestrs
 endif
 
+COMPILE = $(CC_FOR_BUILD) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) \
+	$(AM_CPPFLAGS) $(CPPFLAGS_FOR_BUILD) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD)
+LINK = $(CC_FOR_BUILD) $(AM_CFLAGS) $(CFLAGS_FOR_BUILD) $(AM_LDFLAGS) $(LDFLAGS_FOR_BUILD) -o $@
+
+$(srcdir)/makestrs-makestrs.o:
+	$(COMPILE) -c $< -o $@
+
+$(srcdir)/makestrs: $(srcdir)/makestrs-makestrs.o
+	$(LINK) $< -o $@
+
 EXTRA_DIST = \
 	Shell.ht \
 	StrDefs.ct \
