--- m4-1.4.8.orig/lib/regcomp.c	2006-09-24 06:28:37.000000000 +0200
+++ m4-1.4.8/lib/regcomp.c	2007-01-06 20:53:10.000000000 +0100
@@ -871,6 +871,10 @@ init_dfa (re_dfa_t *dfa, size_t pat_len)
   dfa->state_table = calloc (sizeof (struct re_state_table_entry), table_size);
   dfa->state_hash_mask = table_size - 1;
 
+#ifndef MB_CUR_MAX
+#define MB_CUR_MAX (0)
+#endif
+
   dfa->mb_cur_max = MB_CUR_MAX;
 #ifdef _LIBC
   if (dfa->mb_cur_max == 6
