diff -ur gdb-6.3.50.20050915/gdb/linux-nat.c gdb-6.3.50.20050915-patched/gdb/linux-nat.c
--- gdb-6.3.50.20050915/gdb/linux-nat.c	2005-09-10 13:11:02.000000000 -0500
+++ gdb-6.3.50.20050915-patched/gdb/linux-nat.c	2005-09-15 19:57:23.000000000 -0500
@@ -458,7 +458,7 @@
 	 generally not encounter vfork (vfork is defined to fork
 	 in libpthread.so).
 
-	 The holding part is very easy if we have VFORKDONE events;
+	 The holding part is very easy if we have VFORK_DONE events;
 	 but keeping track of both processes is beyond GDB at the
 	 moment.  So we don't expose the parent to the rest of GDB.
 	 Instead we quietly hold onto it until such time as we can
