Index: links-1.01pre1-no-ssl/links.h
===================================================================
--- links-1.01pre1-no-ssl.orig/links.h	2010-06-21 00:39:10.000000000 +0200
+++ links-1.01pre1-no-ssl/links.h	2010-06-21 00:39:52.000000000 +0200
@@ -2,6 +2,9 @@
 #define __EXTENSIONS__
 #endif
 
+#include <features.h>
+/* uclibc without largefile support #errors on _FILE_OFFSET_BITS=64 */
+#if !(defined(__UCLIBC__) && !defined(__UCLIBC_HAS_LFS__))
 #ifndef _LARGEFILE_SOURCE
 #define _LARGEFILE_SOURCE	1
 #endif
@@ -9,6 +12,7 @@
 #ifndef _FILE_OFFSET_BITS
 #define _FILE_OFFSET_BITS	64
 #endif
+#endif
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
