--- uClibc-0.9.28/extra/scripts/fix_includes.sh.orig	2006-12-11 20:55:46.000000000 -0700
+++ uClibc-0.9.28/extra/scripts/fix_includes.sh	2006-12-11 20:56:12.000000000 -0700
@@ -96,16 +96,16 @@
 if [ -z "$VERSION" -o -z "$PATCHLEVEL" -o -z "$SUBLEVEL" ]
 then
     echo "Unable to determine version for kernel headers"
-    echo -e "\tprovided in directory $KERNEL_SOURCE"
+    echo "    provided in directory $KERNEL_SOURCE"
     exit 1
 fi
 
 if [ "$MAKE_IS_SILENT" != "y" ]; then
 echo "Current kernel version is $VERSION.$PATCHLEVEL.$SUBLEVEL${EXTRAVERSION}"
-echo -e "\n"
+echo ""
 echo "Using kernel headers from $VERSION.$PATCHLEVEL.$SUBLEVEL${EXTRAVERSION} for architecture '$TARGET_ARCH'"
-echo -e "\tprovided in directory $KERNEL_SOURCE"
-echo -e "\n"
+echo "    provided in directory $KERNEL_SOURCE"
+echo ""
 fi
 
 # Create a symlink to include/asm
--- uClibc-0.9.28/Makefile.orig	2006-12-11 21:06:42.000000000 -0700
+++ uClibc-0.9.28/Makefile	2006-12-11 21:06:53.000000000 -0700
@@ -158,7 +158,7 @@
 	$(INSTALL) -d $(PREFIX)$(DEVEL_PREFIX)lib
 	$(INSTALL) -d $(PREFIX)$(DEVEL_PREFIX)include
 	-$(INSTALL) -m 644 lib/*.[ao] $(PREFIX)$(DEVEL_PREFIX)lib/
-	if [ "$(KERNEL_SOURCE)" == "$(DEVEL_PREFIX)" ] ; then \
+	if [ "$(KERNEL_SOURCE)" = "$(DEVEL_PREFIX)" ] ; then \
 		extra_exclude="--exclude include/linux --exclude include/asm'*'" ; \
 	else \
 		extra_exclude="" ; \
