--- a/libc/sysdeps/linux/xtensa/sysdep.h	2008-09-23 21:17:42.000000000 -0700
+++ b/libc/sysdeps/linux/xtensa/sysdep.h	2008-09-23 21:17:59.000000000 -0700
@@ -114,15 +114,7 @@
 
 #define ret_ERRVAL retw
 
-#if RTLD_PRIVATE_ERRNO
-# define SYSCALL_ERROR_HANDLER						      \
-0:	movi	a4, rtld_errno;						      \
-	neg	a2, a2;							      \
-	s32i	a2, a4, 0;						      \
-	movi	a2, -1;							      \
-	j	.Lpseudo_end;
-
-#elif defined _LIBC_REENTRANT
+#if defined _LIBC_REENTRANT
 
 # if USE___THREAD
 #  ifndef NOT_IN_libc
