---
 extra/locale/gen_wc8bit.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: uClibc-0.9.31/extra/locale/gen_wc8bit.c
===================================================================
--- uClibc-0.9.31.orig/extra/locale/gen_wc8bit.c
+++ uClibc-0.9.31/extra/locale/gen_wc8bit.c
@@ -120,7 +120,7 @@
 		}
 
  locale_failure:
-		printf("could not find a UTF8 locale ... please enable en_US.UTF-8\n");
+		fprintf(stderr, "could not find a UTF8 locale ... please enable en_US.UTF-8\n");
 		return EXIT_FAILURE;
  locale_success:
 		pclose(fp);
