--- uClibc-0.9.29-orig/libc/sysdeps/linux/xtensa/sys/procfs.h	2007-11-16 23:21:00.000000000 -0800
+++ uClibc-0.9.29/libc/sysdeps/linux/xtensa/sys/procfs.h	2007-11-16 23:26:59.000000000 -0800
@@ -32,19 +32,11 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/user.h>
+#include <asm/elf.h>
 
 __BEGIN_DECLS
 
-/* Type for a general-purpose register.  */
-typedef unsigned long elf_greg_t;
-
-#define ELF_NGREG 128
-typedef elf_greg_t elf_gregset_t[ELF_NGREG];
-
-/* Register set for the floating-point registers.  */
-#define ELF_NFPREG 18
-typedef unsigned long elf_fpreg_t;
-typedef elf_fpreg_t elf_fpregset_t[ELF_NFPREG];
+/* Note: we get elf_gregset_t and elf_fpregset_t from <asm/elf.h>.  */
 
 /* Signal info.  */
 struct elf_siginfo
