diff -Nura squid-3.1.10.orig/configure.ac squid-3.1.10/configure.ac
--- squid-3.1.10.orig/configure.ac	2010-12-22 02:48:27.000000000 -0300
+++ squid-3.1.10/configure.ac	2010-12-22 11:31:08.420533635 -0300
@@ -384,7 +384,7 @@
 
 dnl Nasty hack to get autoconf 2.64 on Linux to run.
 dnl all other uses of RUN_IFELSE are wrapped inside CACHE_CHECK which breaks on 2.64
-AC_RUN_IFELSE([AC_LANG_SOURCE([[ int main(int argc, char **argv) { return 0; } ]])],[],[],[:])
+dnl AC_RUN_IFELSE([AC_LANG_SOURCE([[ int main(int argc, char **argv) { return 0; } ]])],[],[],[:])
 
 dnl This is a developer only option.. developers know how to set defines
 dnl
@@ -1889,7 +1889,7 @@
     AC_MSG_NOTICE([Negotiate auth helpers built: $NEGOTIATE_AUTH_HELPERS])
 fi
 AC_SUBST(NEGOTIATE_AUTH_HELPERS)
-AC_CONFIG_SUBDIRS(helpers/negotiate_auth/squid_kerb_auth)
+dnl AC_CONFIG_SUBDIRS(helpers/negotiate_auth/squid_kerb_auth)
 
 dnl Select digest auth scheme helpers to build
 if test -n "$AUTH_MODULE_digest"; then
@@ -2202,6 +2202,18 @@
 	;;
 esac
 
+dnl Define CXX_FOR_BUILD
+CXX_FOR_BUILD="$CXX_FOR_BUILD"
+AC_SUBST(CXX_FOR_BUILD)
+
+dnl Define CXXFLAGS_FOR_BUILD
+CXXFLAGS_FOR_BUILD="$CXXFLAGS_FOR_BUILD"
+AC_SUBST(CXXFLAGS_FOR_BUILD)
+
+dnl Define LDFLAGS_FOR_BUILD
+LDFLAGS_FOR_BUILD="$LDFLAGS_FOR_BUILD"
+AC_SUBST(LDFLAGS_FOR_BUILD)
+
 dnl Check for programs
 AC_PROG_CPP
 AC_PROG_INSTALL
diff -Nura squid-3.1.10.orig/src/Makefile.am squid-3.1.10/src/Makefile.am
--- squid-3.1.10.orig/src/Makefile.am	2010-12-22 02:46:56.000000000 -0300
+++ squid-3.1.10/src/Makefile.am	2010-12-22 11:31:08.422533643 -0300
@@ -195,8 +195,6 @@
 cf_gen_SOURCES = cf_gen.cc
 nodist_cf_gen_HEADER = cf_gen_defines.cci
 cf_gen_LDADD= \
-	../compat/libcompat.la \
-	-L../lib -lmiscutil \
 	$(EPOLL_LIBS) \
 	$(MINGW_LIBS) \
 	$(XTRA_LIBS)
@@ -753,6 +751,12 @@
 squid.conf.default squid.conf.documented: cf_parser.cci
 	true
 
+cf_gen$(EXEEXT): $(cf_gen_OBJECTS) $(cf_gen_DEPENDENCIES)
+	$(CXX_FOR_BUILD) $(CXXFLAGS_FOR_BUILD) $(LDFLAGS_FOR_BUILD) -o $@ $(srcdir)/cf_gen.cc \
+		$(top_srcdir)/lib/util.c \
+		$(top_srcdir)/compat/assert.cc \
+		$(cf_gen_LDADD) $(LIBS) ${INCLUDES}
+
 cf_parser.cci: cf.data cf_gen$(EXEEXT)
 	./cf_gen cf.data $(srcdir)/cf.data.depend
 
