Add missing source files for aarch64.

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr libyuv-1ebf86795cb213a37f06eb1ef3713cff080568ea.org/CMakeLists.txt libyuv-1ebf86795cb213a37f06eb1ef3713cff080568ea/CMakeLists.txt
--- libyuv-1ebf86795cb213a37f06eb1ef3713cff080568ea.org/CMakeLists.txt	2015-06-20 00:04:06.000000000 +0200
+++ libyuv-1ebf86795cb213a37f06eb1ef3713cff080568ea/CMakeLists.txt	2015-09-07 19:29:42.898034988 +0200
@@ -14,6 +14,7 @@
   ${ly_src_dir}/compare.cc
   ${ly_src_dir}/compare_common.cc
   ${ly_src_dir}/compare_neon.cc
+  ${ly_src_dir}/compare_neon64.cc
   ${ly_src_dir}/compare_gcc.cc
   ${ly_src_dir}/compare_win.cc
   ${ly_src_dir}/convert.cc
@@ -31,10 +32,12 @@
   ${ly_src_dir}/rotate_argb.cc
   ${ly_src_dir}/rotate_mips.cc
   ${ly_src_dir}/rotate_neon.cc
+  ${ly_src_dir}/rotate_neon64.cc
   ${ly_src_dir}/row_any.cc
   ${ly_src_dir}/row_common.cc
   ${ly_src_dir}/row_mips.cc
   ${ly_src_dir}/row_neon.cc
+  ${ly_src_dir}/row_neon64.cc
   ${ly_src_dir}/row_gcc.cc
   ${ly_src_dir}/row_win.cc
   ${ly_src_dir}/scale.cc
@@ -43,6 +46,7 @@
   ${ly_src_dir}/scale_common.cc
   ${ly_src_dir}/scale_mips.cc
   ${ly_src_dir}/scale_neon.cc
+  ${ly_src_dir}/scale_neon64.cc
   ${ly_src_dir}/scale_gcc.cc
   ${ly_src_dir}/scale_win.cc
   ${ly_src_dir}/video_common.cc
