Change config files to match default buildroot install

Signed-off-by: Spenser Gilliland <spenser@gillilanding.com>
---
Index: gst-omx-1.0.0/config/bellagio/gstomx.conf
===================================================================
--- gst-omx-1.0.0.orig/config/bellagio/gstomx.conf	2013-03-12 04:12:09.000000000 -0500
+++ gst-omx-1.0.0/config/bellagio/gstomx.conf	2013-05-21 17:50:14.545692908 -0500
@@ -1,6 +1,6 @@
 [omxmpeg4videodec]
 type-name=GstOMXMPEG4VideoDec
-core-name=/usr/local/lib/libomxil-bellagio.so.0
+core-name=/usr/lib/libomxil-bellagio.so.0
 component-name=OMX.st.video_decoder.mpeg4
 rank=256
 in-port-index=0
@@ -9,7 +9,7 @@
 
 [omxh264dec]
 type-name=GstOMXH264Dec
-core-name=/usr/local/lib/libomxil-bellagio.so.0
+core-name=/usr/lib/libomxil-bellagio.so.0
 component-name=OMX.st.video_decoder.avc
 rank=256
 in-port-index=0
@@ -18,7 +18,7 @@
 
 [omxmpeg4videoenc]
 type-name=GstOMXMPEG4VideoEnc
-core-name=/usr/local/lib/libomxil-bellagio.so.0
+core-name=/usr/lib/libomxil-bellagio.so.0
 component-name=OMX.st.video_encoder.mpeg4
 rank=0
 in-port-index=0
@@ -27,7 +27,7 @@
 
 [omxaacenc]
 type-name=GstOMXAACEnc
-core-name=/usr/local/lib/libomxil-bellagio.so.0
+core-name=/usr/lib/libomxil-bellagio.so.0
 component-name=OMX.st.audio_encoder.aac
 rank=0
 in-port-index=0
Index: gst-omx-1.0.0/config/rpi/gstomx.conf
===================================================================
--- gst-omx-1.0.0.orig/config/rpi/gstomx.conf	2013-05-21 17:51:39.965693635 -0500
+++ gst-omx-1.0.0/config/rpi/gstomx.conf	2013-05-21 17:52:07.781693871 -0500
@@ -1,6 +1,6 @@
 [omxmpeg2videodec]
 type-name=GstOMXMPEG2VideoDec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -9,7 +9,7 @@
 
 [omxmpeg4videodec]
 type-name=GstOMXMPEG4VideoDec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -18,7 +18,7 @@
 
 [omxh263dec]
 type-name=GstOMXH263Dec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -27,7 +27,7 @@
 
 [omxh264dec]
 type-name=GstOMXH264Dec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -36,7 +36,7 @@
 
 [omxtheoradec]
 type-name=GstOMXTheoraDec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -45,7 +45,7 @@
 
 [omxvp8dec]
 type-name=GstOMXVP8Dec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -54,7 +54,7 @@
 
 [omxmjpegdec]
 type-name=GstOMXMJPEGDec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_decode
 rank=256
 in-port-index=130
@@ -63,7 +63,7 @@
 
 [omxvc1dec]
 type-name=GstOMXWMVDec
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_encode
 rank=256
 in-port-index=130
@@ -73,7 +73,7 @@
 
 [omxh264enc]
 type-name=GstOMXH264Enc
-core-name=/opt/vc/lib/libopenmaxil.so
+core-name=/usr/lib/libopenmaxil.so
 component-name=OMX.broadcom.video_encode
 rank=256
 in-port-index=200
