diff -ur udev-101/Makefile udev-101-patched/Makefile
--- udev-101/Makefile	2006-09-30 07:30:00.000000000 -0500
+++ udev-101-patched/Makefile	2006-10-20 12:28:18.000000000 -0500
@@ -275,6 +275,7 @@
 
 install-bin:
 	$(INSTALL) -d $(DESTDIR)$(udevdir)
+	$(INSTALL_PROGRAM) -D udevstart $(DESTDIR)$(sbindir)/udevstart
 	$(INSTALL_PROGRAM) -D udevd $(DESTDIR)$(sbindir)/udevd
 	$(INSTALL_PROGRAM) -D udevtrigger $(DESTDIR)$(sbindir)/udevtrigger
 	$(INSTALL_PROGRAM) -D udevsettle $(DESTDIR)$(sbindir)/udevsettle
