diff -urp udev-058/etc/udev/frugalware/udev.rules udev-058/etc/udev/frugalware/udev.rules
--- udev-058/etc/udev/frugalware/udev.rules	2005-05-20 21:18:28.000000000 +0100
+++ udev-058/etc/udev/frugalware/udev.rules	2005-10-26 02:26:25.000000000 +0100
@@ -40,6 +40,8 @@ KERNEL="ttyUB[0-9]",    NAME="%k", OWNER
 KERNEL="ttyUSB0",       NAME="%k", OWNER="root", GROUP="uucp", MODE="0660"
 KERNEL="ttyUSB1",       NAME="%k", OWNER="root"  GROUP="uucp", MODE="0666"
 
+KERNEL="s3c2410_serial*" NAME="%k", OWNER="root", GROUP="uucp", MODE="0660"
+
 # pty devices
 KERNEL="ptmx"           NAME="%k", OWNER="root", GROUP="tty",  MODE="0666"
 KERNEL="pty[p-za-e][0-9a-f]*" NAME="%k", OWNER="root", GROUP="tty",  MODE="0666"
