From 14d4d3e7b5c748118acffddd397fb67b37fd5054 Mon Sep 17 00:00:00 2001
From: "Arnout Vandecappelle (Essensium/Mind)" <arnout@mind.be>
Date: Thu, 24 Nov 2016 22:44:02 +0100
Subject: [PATCH] Tell qmake to add a pkgconfig file to ease usage with
 applications

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
---
 qextserialport.pro | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/qextserialport.pro b/qextserialport.pro
index 8d2c03a..2fcb39c 100644
--- a/qextserialport.pro
+++ b/qextserialport.pro
@@ -42,7 +42,8 @@ TEMPLATE=lib
 include(src/qextserialport.pri)
 
 #create_prl is needed, otherwise, MinGW can't found libqextserialport1.a
-CONFIG += create_prl
+CONFIG += create_prl create_pc
+QMAKE_PKGCONFIG_DESTDIR = pkgconfig
 
 #mac framework is designed for shared library
 macx:qesp_mac_framework:qesp_static: CONFIG -= qesp_static
-- 
2.10.2

