diff -urpN fbv-1.0b.orig/fb_display.c fbv-1.0b/fb_display.c
--- fbv-1.0b.orig/fb_display.c	2004-09-07 14:09:43.000000000 +0200
+++ fbv-1.0b/fb_display.c	2008-04-24 10:48:29.000000000 +0200
@@ -18,8 +18,6 @@
     Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
 */
 
-#include <linux/fb.h>
-
 #include <stdio.h>
 #include <stdlib.h>
 #include <sys/types.h>
@@ -31,6 +29,7 @@
 #include <asm/types.h>
 #include <string.h>
 #include <errno.h>
+#include <linux/fb.h>
 #include "config.h"
 /* Public Use Functions:
  *
