---
 Makefile |   20 +++++++++-----------
 1 file changed, 9 insertions(+), 11 deletions(-)

Index: mrouted-3.9.4/Makefile
===================================================================
--- mrouted-3.9.4.orig/Makefile
+++ mrouted-3.9.4/Makefile
@@ -42,11 +42,9 @@
 #MSTAT_OBJS    = mstat.o $(EXTRA_OBJS)
 
 ## Common
-CFLAGS        = $(MCAST_INCLUDE) $(SNMPDEF) $(RSRRDEF) $(INCLUDES) $(DEFS) $(USERCOMPILE)
-CFLAGS       += -O2 -W -Wall -Werror
-#CFLAGS       += -O -g
-LDLIBS        = $(SNMPLIBDIR) $(SNMPLIBS) $(EXTRA_LIBS)
-LDFLAGS      += -Wl,-Map,$@.map
+MROUTED_CFLAGS   = $(MCAST_INCLUDE) $(SNMPDEF) $(RSRRDEF) $(INCLUDES) $(DEFS) $(USERCOMPILE)
+LDLIBS           = $(SNMPLIBDIR) $(SNMPLIBS) $(EXTRA_LIBS)
+MROUTED_LDFLAGS += -Wl,-Map,$@.map
 OBJS          = $(IGMP_OBJS) $(ROUTER_OBJS) $(MAPPER_OBJS) $(MRINFO_OBJS) \
 		$(MTRACE_OBJS) $(MSTAT_OBJS)
 SRCS          = $(OBJS:.o=.c)
@@ -65,7 +63,7 @@
 
 .c.o:
 	@printf "  CC      $@\n"
-	@$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $@ $<
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(CPPFLAGS) -c -o $@ $<
 
 install: $(EXECS)
 	@install -d $(DESTDIR)$(prefix)/sbin
@@ -95,26 +93,26 @@
 
 mrouted: $(IGMP_OBJS) $(ROUTER_OBJS) $(CMULIBS)
 	@printf "  LINK    $@\n"
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(IGMP_OBJS) $(ROUTER_OBJS) $(LDLIBS)
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(LDFLAGS) $(MROUTED_LDFLAGS) -o $@ $(IGMP_OBJS) $(ROUTER_OBJS) $(LDLIBS)
 
 vers.c: Makefile
 	@echo $(VERSION) | sed -e 's/.*/char todaysversion[]="&";/' > vers.c
 
 map-mbone: $(IGMP_OBJS) $(MAPPER_OBJS)
 	@printf "  LINK    $@\n"
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(IGMP_OBJS) $(MAPPER_OBJS) $(LDLIBS)
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(LDFLAGS) $(MROUTED_LDFLAGS) -o $@ $(IGMP_OBJS) $(MAPPER_OBJS) $(LDLIBS)
 
 mrinfo: $(IGMP_OBJS) $(MRINFO_OBJS)
 	@printf "  LINK    $@\n"
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(IGMP_OBJS) $(MRINFO_OBJS) $(LDLIBS)
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(LDFLAGS) $(MROUTED_LDFLAGS) -o $@ $(IGMP_OBJS) $(MRINFO_OBJS) $(LDLIBS)
 
 mtrace: $(IGMP_OBJS) $(MTRACE_OBJS)
 	@printf "  LINK    $@\n"
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(IGMP_OBJS) $(MTRACE_OBJS) $(LDLIBS)
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(LDFLAGS) $(MROUTED_LDFLAGS) -o $@ $(IGMP_OBJS) $(MTRACE_OBJS) $(LDLIBS)
 
 mstat: $(MSTAT_OBJS) $(CMULIBS)
 	@printf "  LINK    $@\n"
-	@$(CC) $(CFLAGS) $(LDFLAGS) -o $@ $(MSTAT_OBJS) $(LDLIBS)
+	@$(CC) $(CFLAGS) $(MROUTED_CFLAGS) $(LDFLAGS) $(MROUTED_LDFLAGS) -o $@ $(MSTAT_OBJS) $(LDLIBS)
 
 clean: $(SNMPCLEAN)
 	-@$(RM) $(OBJS) $(EXECS)
