From db17dd9728b77485f26abb1ac5fcda65130f4a8d Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Sat, 27 Jun 2015 08:08:51 -0300
Subject: [PATCHv2 iproute2] tipc: make build conditional on having libmnl

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 tipc/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tipc/Makefile b/tipc/Makefile
index 4bda8c5..b3ef9b9 100644
--- a/tipc/Makefile
+++ b/tipc/Makefile
@@ -1,3 +1,6 @@
+include ../Config
+ifeq ($(HAVE_MNL),y)
+
 TIPCOBJ=bearer.o \
     cmdl.o link.o \
     media.o misc.o \
@@ -8,6 +11,8 @@ TIPCOBJ=bearer.o \
 TARGETS=tipc
 LDLIBS += -lmnl
 
+endif
+
 all: $(TARGETS) $(LIBS)
 
 tipc: $(TIPCOBJ)
-- 
2.3.6

