Patch to correct a missing header file issue.

Signed-off-by Clayton Shotwell <clshotwe@rockwellcollins.com>

--- a/src/exception.sh	2011-12-21 11:46:04.000000000 -0600
+++ b/src/exception.sh	2012-08-27 11:29:58.000000000 -0500
@@ -9,6 +9,6 @@
 }
 "
 }
-gcc -x c -c - -aux-info temp.aux < ../include/semanage/semanage.h
+gcc -x c -c - -aux-info temp.aux -I../include < ../include/semanage/semanage.h
 for i in `awk '/extern int/ { print $6 }' temp.aux`; do except $i ; done
 rm -f -- temp.aux -.o
