--- xdriver_xf86-video-xgi-1.5.1/configure.ac.orig	2009-09-07 08:03:18.000000000 +0300
+++ xdriver_xf86-video-xgi-1.5.1/configure.ac	2010-02-03 23:58:40.507035914 +0200
@@ -78,12 +78,18 @@ sdkdir=$(pkg-config --variable=sdkdir xo
 AC_HEADER_STDC
 
 if test "x$DRI" = xauto; then
-        AC_CHECK_FILE([${sdkdir}/dri.h],
+        if test "$cross_compiling" = "no" ; then
+                AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
-        AC_CHECK_FILE([${sdkdir}/sarea.h],
+                AC_CHECK_FILE([${sdkdir}/sarea.h],
                       [have_sarea_h="yes"], [have_sarea_h="no"])
-        AC_CHECK_FILE([${sdkdir}/dristruct.h],
+                AC_CHECK_FILE([${sdkdir}/dristruct.h],
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
+        else
+                have_dri_h="yes"
+                have_sarea_h="yes"
+                have_dristruct_h="yes"
+        fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
