diff -ur ltp-full-20050804/testcases/network/multicast/mc_member/member.c ltp-full-20050804-patched/testcases/network/multicast/mc_member/member.c
--- ltp-full-20050804/testcases/network/multicast/mc_member/member.c	2005-08-04 15:33:05.000000000 -0500
+++ ltp-full-20050804-patched/testcases/network/multicast/mc_member/member.c	2006-06-21 22:56:15.746866000 -0500
@@ -74,7 +74,7 @@
         }
 
         if(hp = gethostbyname(interface)) {
-           bcopy(hp->h_addr, &imr.imr_interface.s_addr, hp->h_length);
+           memcpy(&imr.imr_interface.s_addr, hp->h_addr, hp->h_length);
         } else 
            if((n = sscanf(interface, "%u.%u.%u.%u", &i1, &i2, &i3, &i4)) != 4) {
               fprintf(stderr, "bad group address\n" ); 
diff -ur ltp-full-20050804/testcases/network/tcp_cmds/echo/echoes.c ltp-full-20050804-patched/testcases/network/tcp_cmds/echo/echoes.c
--- ltp-full-20050804/testcases/network/tcp_cmds/echo/echoes.c	2005-08-04 15:33:05.000000000 -0500
+++ ltp-full-20050804-patched/testcases/network/tcp_cmds/echo/echoes.c	2006-06-21 22:55:13.000000000 -0500
@@ -121,7 +121,7 @@
 		exit(1);
 	}
 	port=sp->s_port;
-	bcopy(hp->h_addr_list[0],&hostaddr,sizeof(struct in_addr));
+	memcpy(&hostaddr,hp->h_addr_list[0],sizeof(struct in_addr));
 	bzero((char *)&sa,sizeof (sa));
 	sa.sin_port=port;
 	sa.sin_family=AF_INET;
