diff -ur ltp-full-20061121/testcases/kernel/syscalls/fmtmsg/Makefile ltp-full-20061121-patched/testcases/kernel/syscalls/fmtmsg/Makefile
--- ltp-full-20061121/testcases/kernel/syscalls/fmtmsg/Makefile	2006-11-21 14:31:28.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/fmtmsg/Makefile	2006-12-02 00:16:36.830948316 -0600
@@ -20,7 +20,7 @@
 LDLIBS += -L../../../../lib -lltp
 
 SRCS    = $(wildcard *.c)
-TARGETS = $(patsubst %.c,%,$(SRCS))
+TARGETS =#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/getcontext/Makefile ltp-full-20061121-patched/testcases/kernel/syscalls/getcontext/Makefile
--- ltp-full-20061121/testcases/kernel/syscalls/getcontext/Makefile	2006-11-21 14:31:27.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/getcontext/Makefile	2006-12-02 00:16:50.072004501 -0600
@@ -20,7 +20,7 @@
 LDLIBS += -L../../../../lib -lltp
 
 SRCS    = $(wildcard *.c)
-TARGETS = $(patsubst %.c,%,$(SRCS))
+TARGETS =#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/profil/Makefile ltp-full-20061121-patched/testcases/kernel/syscalls/profil/Makefile
--- ltp-full-20061121/testcases/kernel/syscalls/profil/Makefile	2006-11-21 14:31:28.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/profil/Makefile	2006-12-02 00:17:03.301059744 -0600
@@ -20,7 +20,7 @@
 LDLIBS += -L../../../../lib -lltp
 
 SRCS    = $(wildcard *.c)
-TARGETS = $(patsubst %.c,%,$(SRCS))
+TARGETS =#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/pselect/Makefile ltp-full-20061121-patched/testcases/kernel/syscalls/pselect/Makefile
--- ltp-full-20061121/testcases/kernel/syscalls/pselect/Makefile	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/pselect/Makefile	2006-12-02 00:17:14.661965984 -0600
@@ -20,7 +20,7 @@
 LDLIBS += -L../../../../lib -lltp
 
 SRCS    = $(wildcard *.c)
-TARGETS = $(patsubst %.c,%,$(SRCS))
+TARGETS =#$(patsubst %.c,%,$(SRCS))
 
 all: $(TARGETS)
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/swapon/swapon02.c ltp-full-20061121-patched/testcases/kernel/syscalls/swapon/swapon02.c
--- ltp-full-20061121/testcases/kernel/syscalls/swapon/swapon02.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/swapon/swapon02.c	2006-12-02 00:18:12.422573631 -0600
@@ -85,9 +85,7 @@
 #include <sys/wait.h>
 #include <sys/stat.h>
 #include <sys/swap.h>
-#ifdef OLDER_DISTRO_RELEASE
-#include <linux/swap.h>
-#endif
+#define MAX_SWAPFILES 32
 #include <fcntl.h>
 #include <pwd.h>
 #include <string.h>
@@ -96,10 +94,6 @@
 #include "test.h"
 #include "usctest.h"
 
-#ifndef OLDER_DISTRO_RELEASE
-#define MAX_SWAPFILES 32 
-#endif
-
 static void setup();
 static void cleanup();
 static int setup01();
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs01.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs01.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs01.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs01.c	2006-12-02 00:15:34.705993080 -0600
@@ -72,10 +72,6 @@
 #include <unistd.h>
 #include <syscall.h>
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 static void setup();
 static void cleanup();
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs02.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs02.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs02.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs02.c	2006-12-02 00:15:34.705993080 -0600
@@ -70,10 +70,6 @@
 #include <unistd.h>
 #include <syscall.h>
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 static void setup();
 static void cleanup();
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs03.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs03.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs03.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs03.c	2006-12-02 00:15:34.705993080 -0600
@@ -70,10 +70,6 @@
 #include <unistd.h>
 #include <syscall.h>
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 static void setup();
 static void cleanup();
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs04.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs04.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs04.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs04.c	2006-12-02 00:15:34.705993080 -0600
@@ -73,10 +73,6 @@
 #include "test.h"
 #include "usctest.h"
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 #define INVALID_OPTION 100
 static void setup();
 static void cleanup();
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs05.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs05.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs05.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs05.c	2006-12-02 00:22:55.657171949 -0600
@@ -74,10 +74,6 @@
 #include "test.h"
 #include "usctest.h"
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 static void setup();
 static void cleanup();
 
diff -ur ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs06.c ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs06.c
--- ltp-full-20061121/testcases/kernel/syscalls/sysfs/sysfs06.c	2006-11-21 14:31:29.000000000 -0600
+++ ltp-full-20061121-patched/testcases/kernel/syscalls/sysfs/sysfs06.c	2006-12-02 00:22:47.124491054 -0600
@@ -75,10 +75,6 @@
 #include "test.h"
 #include "usctest.h"
 
-#ifndef _syscall2
-#include <linux/unistd.h>
-#endif
-
 static void setup();
 static void cleanup();
 
