diff -ru genext2fs-1.4/genext2fs.c genext2fs-1.4-remove_ugly_warnings/genext2fs.c
--- genext2fs-1.4/genext2fs.c	2007-02-09 10:54:56.000000000 +0100
+++ genext2fs-1.4-remove_ugly_warnings/genext2fs.c	2007-02-09 10:54:30.000000000 +0100
@@ -1606,7 +1606,7 @@
 static void
 add2fs_from_dir(filesystem *fs, uint32 this_nod, int squash_uids, int squash_perms, uint32 fs_timestamp, struct stats *stats)
 {
-	uint32 nod;
+	uint32 nod = 0;
 	uint32 uid, gid, mode, ctime, mtime;
 	const char *name;
 	FILE *fh;
@@ -2056,7 +2056,7 @@
 static filesystem *
 load_fs(FILE * fh, int swapit)
 {
-	size_t fssize;
+	size_t fssize = 0;
 	filesystem *fs;
 	if((fseek(fh, 0, SEEK_END) < 0) || ((ssize_t)(fssize = ftell(fh)) == -1))
 		perror_msg_and_die("input filesystem image");
