diff -ru sfdisk_llseek-fix/sfdisk.c sfdisk_index-rindex-fix/sfdisk.c
--- sfdisk_llseek-fix/sfdisk.c	2008-04-17 08:42:35.000000000 +0000
+++ sfdisk_index-rindex-fix/sfdisk.c	2008-04-17 09:50:17.000000000 +0000
@@ -58,6 +58,17 @@
 
 #define SIZE(a)	(sizeof(a)/sizeof(a[0]))
 
+/* 
+ * Why these wouldn't be defined in string.h, I don't know, there is a block
+ * testing for BSD, and no other mention of these functions ... NK
+ */
+#if !defined(index)
+#  define index(s,c) strchr((s), (c))
+#endif
+#if !defined(rindex)
+#  define rindex(s,c) strrchr((s), (c))
+#endif
+
 /*
  * Table of contents:
  *  A. About seeking
