From b0fb0d1f69ceaceba05500602747d9f85f360fb9 Mon Sep 17 00:00:00 2001
From: Max Filippov <jcmvbkbc@gmail.com>
Date: Sat, 25 Oct 2014 14:29:08 +0400
Subject: [PATCH] Config.in.arch: enable long double math for xtensa

Signed-off-by: Max Filippov <jcmvbkbc@gmail.com>
---
 extra/Configs/Config.in.arch | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/extra/Configs/Config.in.arch b/extra/Configs/Config.in.arch
index 2338e3c..68e03bf 100644
--- a/extra/Configs/Config.in.arch
+++ b/extra/Configs/Config.in.arch
@@ -191,7 +191,7 @@ config UCLIBC_HAS_FENV
 config UCLIBC_HAS_LONG_DOUBLE_MATH
 	bool "Enable long double support"
 	depends on DO_C99_MATH
-	depends on TARGET_i386 || TARGET_m68k || TARGET_sparc || TARGET_x86_64 || TARGET_powerpc || TARGET_sh || TARGET_microblaze
+	depends on TARGET_i386 || TARGET_m68k || TARGET_sparc || TARGET_x86_64 || TARGET_powerpc || TARGET_sh || TARGET_microblaze || TARGET_xtensa
 	default y
 	help
 	  If you want the uClibc math library to contain the full set of C99
-- 
1.8.1.4

