Makefile: Don't call "ln" directly

Signed-off-by: Vicente Olivert Riera <vincent.riera@imgtec.com>

--- keyutils-1.5.9/Makefile.orig	2014-09-22 16:05:14.117007430 +0100
+++ keyutils-1.5.9/Makefile	2014-09-22 16:06:26.053219336 +0100
@@ -123,10 +123,10 @@ endif
 ifeq ($(NO_SOLIB),0)
 all: $(DEVELLIB)
 $(DEVELLIB): $(SONAME)
-	ln -sf $< $@
+	$(LNS) $< $@
 
 $(SONAME): $(LIBNAME)
-	ln -sf $< $@
+	$(LNS) $< $@
 
 LIBVERS := -shared -Wl,-soname,$(SONAME) -Wl,--version-script,version.lds
 
