diff -ru popt-1.14_vanilla/popthelp.c popt-1.14_no-wchar/popthelp.c
--- popt-1.14_vanilla/popthelp.c	2008-03-27 17:33:08.000000000 +0000
+++ popt-1.14_no-wchar/popthelp.c	2008-04-10 05:44:43.000000000 +0000
@@ -15,7 +15,7 @@
 #include <sys/ioctl.h>
 #endif
 
-#define	POPT_WCHAR_HACK
+/* #define	POPT_WCHAR_HACK */
 #ifdef 	POPT_WCHAR_HACK
 #include <wchar.h>			/* for mbsrtowcs */
 /*@access mbstate_t @*/
