diff -ur netplug-1.2.9/netlink.c netplug-1.2.9-patched/netlink.c
--- netplug-1.2.9/netlink.c	2005-01-07 23:57:09.000000000 -0600
+++ netplug-1.2.9-patched/netlink.c	2008-02-27 17:59:48.373864047 -0600
@@ -286,7 +286,7 @@
 
     int addr_len = sizeof(addr);
 
-    if (getsockname(fd, (struct sockaddr *) &addr, &addr_len) == -1) {
+    if (getsockname(fd, (struct sockaddr *) &addr, (socklen_t *) &addr_len) == -1) {
         do_log(LOG_ERR, "Could not get socket details: %m");
         exit(1);
     }
