--- gtk+-2.8.10/gtk/gtkentry.c.set-invisible-char-to-bullet	2006-01-19 15:12:18.000000000 -0500
+++ gtk+-2.8.10/gtk/gtkentry.c	2006-01-19 15:36:51.000000000 -0500
@@ -542,7 +542,7 @@
                                    g_param_spec_unichar ("invisible-char",
 							 P_("Invisible character"),
 							 P_("The character to use when masking entry contents (in \"password mode\")"),
-							 '*',
+							 (gunichar) 0x25cf,
 							 GTK_PARAM_READWRITE));
 
   g_object_class_install_property (gobject_class,
@@ -1000,7 +1000,7 @@
 
   entry->editable = TRUE;
   entry->visible = TRUE;
-  entry->invisible_char = '*';
+  entry->invisible_char = (gunichar) 0x25cf;
   entry->dnd_position = -1;
   entry->width_chars = -1;
   entry->is_cell_renderer = FALSE;
@@ -4019,9 +4019,9 @@
  * gtk_entry_set_visibility() has been called to set text visibility
  * to %FALSE. i.e. this is the character used in "password mode" to
  * show the user how many characters have been typed. The default
- * invisible char is an asterisk ('*').  If you set the invisible char
- * to 0, then the user will get no feedback at all; there will be
- * no text on the screen as they type.
+ * invisible char is a small bullet (Unicode character 2022). If you
+ * set the invisible char to 0, then the user will get no feedback at
+ * all; there will be no text on the screen as they type.
  * 
  **/
 void
