misc.m4: ac_cv_c_stack_direction is unset.

Signed-off-by: Marcelo Gutierrez (UTN/FRH) <kuyurix@gmail.com>

--- mysql-5.1.70.orig/config/ac-macros/misc.m4
+++ mysql-5.1.70/config/ac-macros/misc.m4
@@ -477,7 +477,7 @@
     exit(ptr_f(&a) < 0);
   }
   ], ac_cv_c_stack_direction=1, ac_cv_c_stack_direction=-1,
-   ac_cv_c_stack_direction=)])
+   ac_cv_c_stack_direction=0)])
  AC_DEFINE_UNQUOTED(STACK_DIRECTION, $ac_cv_c_stack_direction)
 ])dnl
 
