--- a/configure.in.orig	2009-07-08 15:23:44.000000000 +0200
+++ b/configure.in	2009-07-08 15:24:52.000000000 +0200
@@ -997,7 +997,25 @@
 fi
 
 AC_FUNC_GETPGRP
-AC_FUNC_SETPGRP  
+
+dnl AC_FUNC_SETPGRP does not work if cross compiling
+dnl Instead, assume we will have a prototype for setpgrp if cross compiling.
+if test "$cross_compiling" = no; then
+  AC_FUNC_SETPGRP
+else
+  AC_CACHE_CHECK([whether setpgrp takes no argument], ac_cv_func_setpgrp_void,
+    [AC_TRY_COMPILE([
+#include <unistd.h>
+], [
+  if (setpgrp(1,1) == -1)
+    exit (0);
+  else
+    exit (1);
+], ac_cv_func_setpgrp_void=no, ac_cv_func_setpgrp_void=yes)])
+if test $ac_cv_func_setpgrp_void = yes; then
+  AC_DEFINE(SETPGRP_VOID, 1)
+fi
+fi
 
 AC_C_BIGENDIAN
 AC_C_CONST
