diff -urpN busybox-1.18.1/modutils/modprobe-small.c busybox-1.18.1-modprobe-small/modutils/modprobe-small.c
--- busybox-1.18.1/modutils/modprobe-small.c	2010-12-20 01:41:27.000000000 +0100
+++ busybox-1.18.1-modprobe-small/modutils/modprobe-small.c	2011-01-09 21:00:26.000000000 +0100
@@ -844,13 +844,15 @@ int modprobe_main(int argc UNUSED_PARAM,
 		void *map;
 
 		len = MAXINT(ssize_t);
-		map = xmalloc_xopen_read_close(*argv, &len);
+		map = xmalloc_open_zipped_read_close(*argv, &len);
 		if (init_module(map, len,
 			IF_FEATURE_MODPROBE_SMALL_OPTIONS_ON_CMDLINE(options ? options : "")
 			IF_NOT_FEATURE_MODPROBE_SMALL_OPTIONS_ON_CMDLINE("")
-				) != 0)
+			) != 0
+		) {
 			bb_error_msg_and_die("can't insert '%s': %s",
 					*argv, moderror(errno));
+		}
 		return 0;
 	}
 
