--- util-linux-2.11z/mount/fstab.c.orig	2003-12-03 15:28:22.000000000 -0700
+++ util-linux-2.11z/mount/fstab.c	2003-12-03 15:28:41.000000000 -0700
@@ -342,7 +342,7 @@
 /* Ensure that the lock is released if we are interrupted.  */
 static void
 handler (int sig) {
-     die (EX_USER, "%s", sys_siglist[sig]);
+     die (EX_USER, "%s", strsignal(sig));
 }
 
 static void
--- /dev/null	2003-09-07 01:55:59.000000000 -0600
+++ util-linux-2.11z/mount/swapargs.h	2003-12-03 15:44:50.000000000 -0700
@@ -0,0 +1,3 @@
+#define SWAPON_HAS_TWO_ARGS
+#include <asm/page.h>
+#include <sys/swap.h>
--- util-linux-2.11z/mount/swap.configure.orig	2003-12-03 15:43:24.000000000 -0700
+++ util-linux-2.11z/mount/swap.configure	2003-12-03 15:45:33.000000000 -0700
@@ -1,6 +1,8 @@
 # Find out whether we can include <sys/swap.h>
 # and whether libc thinks that swapon() has two arguments.
 
+exit 0
+
 # Prepare test
 CC=${CC-cc}
 compile="$CC -o conftest conftest.c >/dev/null 2>&1"
--- util-linux-2.12/fdisk/sfdisk.c.orig	2005-01-11 16:46:36.000000000 -0700
+++ util-linux-2.12/fdisk/sfdisk.c	2005-01-11 16:47:14.000000000 -0700
@@ -47,7 +47,7 @@
 #include <sys/stat.h>
 #include <sys/utsname.h>
 #ifdef __linux__
-#include <linux/unistd.h>	/* _syscall */
+#include <sys/syscall.h>	/* _syscall */
 #endif
 #include "nls.h"
 #include "common.h"
--- util-linux-2.12/fdisk/llseek.c.orig	2005-01-11 17:05:55.000000000 -0700
+++ util-linux-2.12/fdisk/llseek.c	2005-01-11 17:07:11.000000000 -0700
@@ -24,7 +24,7 @@
 #define my_llseek lseek
 
 #else
-#include <linux/unistd.h>	/* for __NR__llseek */
+#include <sys/syscall.h>	/* for __NR__llseek */
 
 static int _llseek (unsigned int, unsigned long,
 		   unsigned long, long long *, unsigned int);
@@ -33,7 +33,7 @@
 
 static _syscall5(int,_llseek,unsigned int,fd,unsigned long,offset_high,
 		 unsigned long, offset_low,long long *,result,
-		 unsigned int, origin)
+		 unsigned int, origin);
 
 #else
 
