diff -urN util-linux-2.12q/disk-utils/fsck.cramfs.c util-linux-2.12q-patched/disk-utils/fsck.cramfs.c
--- util-linux-2.12q/disk-utils/fsck.cramfs.c	2004-12-11 08:53:16.000000000 -0600
+++ util-linux-2.12q-patched/disk-utils/fsck.cramfs.c	2005-10-03 17:27:13.000000000 -0500
@@ -77,15 +77,7 @@
 #define PAD_SIZE 512
 
 #include <asm/page.h>
-#ifdef PAGE_SIZE
-#define PAGE_CACHE_SIZE ((int) PAGE_SIZE)
-#elif defined __ia64__
-#define PAGE_CACHE_SIZE (16384)
-#elif defined __alpha__
-#define PAGE_CACHE_SIZE (8192)
-#else
 #define PAGE_CACHE_SIZE (4096)
-#endif
 
 /* Guarantee access to at least 8kB at a time */
 #define ROMBUFFER_BITS	13
@@ -95,7 +87,7 @@
 static unsigned long read_buffer_block = ~0UL;
 
 /* Uncompressing data structures... */
-static char outbuffer[PAGE_CACHE_SIZE*2];
+static char outbuffer[4096*2];
 z_stream stream;
 
 #endif /* INCLUDE_FS_TESTS */
