Signed-off-by: Khem Raj <raj.khem@gmail.com>

Index: util-linux-2.19.1/configure.ac
===================================================================
--- util-linux-2.19.1.orig/configure.ac
+++ util-linux-2.19.1/configure.ac
@@ -250,7 +250,7 @@ esac


 AC_MSG_CHECKING(whether program_invocation_short_name is defined)
-AC_TRY_COMPILE([#include <argp.h>],
+AC_TRY_COMPILE([#include <errno.h>],
                [program_invocation_short_name = "test";],
                AC_DEFINE(HAVE_PROGRAM_INVOCATION_SHORT_NAME, 1,
                          [Define if program_invocation_short_name is defined])

