From 6bc1d15250841cf17d307cfb4f35c960c23d8797 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Sun, 29 May 2016 19:31:50 +0200
Subject: [PATCH] Remove man from BUILDDIRS in configure

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 config-scripts/cups-common.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config-scripts/cups-common.m4 b/config-scripts/cups-common.m4
index fbba715..77d0f5c 100644
--- a/config-scripts/cups-common.m4
+++ b/config-scripts/cups-common.m4
@@ -446,7 +446,7 @@ AC_ARG_WITH(components, [  --with-components       set components to build:
 
 case "$COMPONENTS" in
 	all)
-		BUILDDIRS="filter backend berkeley cgi-bin monitor notifier ppdc scheduler systemv conf data desktop locale man doc examples templates"
+		BUILDDIRS="filter backend berkeley cgi-bin monitor notifier ppdc scheduler systemv conf data desktop locale doc examples templates"
 		;;
 
 	core)
-- 
2.8.1

