--- cups-1.3.5/Makefile	2007-12-06 15:09:39.000000000 -0500
+++ cups-1.3.5/Makefile	2008-02-09 21:53:03.000000000 -0500
@@ -19,8 +19,8 @@ include Makedefs
 # Directories to make...
 #
 
-DIRS	=	cups backend berkeley cgi-bin filter locale man monitor \
-		notifier $(PDFTOPS) scheduler systemv test \
+DIRS	=	cups backend berkeley cgi-bin filter locale monitor \
+		notifier $(PDFTOPS) scheduler systemv \
 		$(PHPDIR) \
 		conf data doc $(FONTS) ppd templates
 
