From 43c4bbf3e0e61f5f487a00a0066816913503656c Mon Sep 17 00:00:00 2001
From: Waldemar Brodkorb <wbx@uclibc-ng.org>
Date: Sun, 9 Oct 2016 20:15:00 +0200
Subject: [PATCH] fix mips/mips64 build for old compilers

Signed-off-by: Waldemar Brodkorb <wbx@uclibc-ng.org>
---
 Rules.mak | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Rules.mak b/Rules.mak
index 67189ff..b9dbf25 100644
--- a/Rules.mak
+++ b/Rules.mak
@@ -404,8 +404,8 @@ ifeq ($(TARGET_ARCH),mips)
 	CPU_CFLAGS-$(CONFIG_MIPS_N64_ABI)+=-mabi=64
 	CPU_CFLAGS-$(CONFIG_MIPS_O32_ABI)+=-mabi=32
 	CPU_CFLAGS-$(CONFIG_MIPS_N32_ABI)+=-mabi=n32
-	CPU_CFLAGS-$(CONFIG_MIPS_NAN_LEGACY)+=-mnan=legacy
-	CPU_CFLAGS-$(CONFIG_MIPS_NAN_2008)+=-mnan=2008
+	CPU_CFLAGS-$(CONFIG_MIPS_NAN_LEGACY)+=$(call check_gcc,-mnan=legacy)
+	CPU_CFLAGS-$(CONFIG_MIPS_NAN_2008)+=$(call check_gcc,-mnan=2008)
 	CPU_LDFLAGS-y += $(CPU_CFLAGS)
 endif
 
-- 
2.1.4

