From a9c7314e069036f74b503dfaef261e5f45477b7e Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 7 Mar 2017 22:30:45 +0100
Subject: [PATCH] Add an option to disable the zlib module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Peter: update for 2.7.16]
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 85df7b1478..3848b16b9f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2875,6 +2875,12 @@ AC_ARG_ENABLE(bz2,
             DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} bz2"
          fi])
 
+AC_ARG_ENABLE(zlib,
+       AS_HELP_STRING([--disable-zlib], [disable ZLIB]),
+       [ if test "$enableval" = "no"; then
+            DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} zlib"
+         fi])
+
 AC_ARG_ENABLE(dbm,
 	AS_HELP_STRING([--disable-dbm], [disable DBM]),
 	[ if test "$enableval" = "no"; then
-- 
2.11.0

