From 6af0f179d3db956146bbabac8748675e3abc1f78 Mon Sep 17 00:00:00 2001
From: Zoltan Gyarmati <mr.zoltan.gyarmati@gmail.com>
Date: Tue, 7 Mar 2017 22:31:22 +0100
Subject: [PATCH] Add an option to disable the ossaudiodev module

Signed-off-by: Zoltan Gyarmati <mr.zoltan.gyarmati@gmail.com>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 291ba35..8dc573e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2898,6 +2898,12 @@ if test "$CURSES" = "no"; then
    DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _curses _curses_panel"
 fi
 
+AC_ARG_ENABLE(ossaudiodev,
+       AS_HELP_STRING([--disable-ossaudiodev], [disable OSSAUDIODEV]),
+       [ if test "$enableval" = "no"; then
+            DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ossaudiodev"
+         fi])
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
2.7.4

