From 479bef8182c4f6b678a86820ccc06760ca60c286 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 7 Mar 2017 22:33:02 +0100
Subject: [PATCH] Add option to disable the hashlib module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Peter: update for 2.7.16]
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index 6d19cdee95..ddccc79c2c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2869,6 +2869,12 @@ AC_ARG_ENABLE(ssl,
     	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _ssl"
   	  fi])
 
+AC_ARG_ENABLE(hashlib,
+	AS_HELP_STRING([--disable-hashlib], [disable hashlib]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _hashlib"
+	  fi])
+
 AC_ARG_ENABLE(bz2,
        AS_HELP_STRING([--disable-bz2], [disable BZIP2]),
        [ if test "$enableval" = "no"; then
-- 
2.11.0

