From abea92958b073dfbf0c78bcb31ce9cfc709c20ca Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 7 Mar 2017 22:33:55 +0100
Subject: [PATCH] Add an option to disable readline module

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
[Peter: update for 2.7.16]
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure.ac b/configure.ac
index ddccc79c2c..dae266326e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -2875,6 +2875,12 @@ AC_ARG_ENABLE(hashlib,
 	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _hashlib"
 	  fi])
 
+AC_ARG_ENABLE(readline,
+	AS_HELP_STRING([--disable-readline], [disable readline]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} readline"
+	  fi])
+
 AC_ARG_ENABLE(bz2,
        AS_HELP_STRING([--disable-bz2], [disable BZIP2]),
        [ if test "$enableval" = "no"; then
-- 
2.11.0

