diff -rduNp Python-2.4.2-010/Python/import.c Python-2.4.2/Python/import.c
--- Python-2.4.2-010/Python/import.c	2005-09-14 20:15:03.000000000 +0200
+++ Python-2.4.2/Python/import.c	2007-01-22 19:49:18.000000000 +0100
@@ -822,8 +822,12 @@ static void
 write_compiled_module(PyCodeObject *co, char *cpathname, long mtime)
 {
 	FILE *fp;
-
-	fp = open_exclusive(cpathname);
+   char *py_dontcompile = getenv("PYTHON_DONTCOMPILE");
+   
+   if (!py_dontcompile) 
+     fp = open_exclusive(cpathname);
+   else
+     fp = NULL;
 	if (fp == NULL) {
 		if (Py_VerboseFlag)
 			PySys_WriteStderr(
