Index: Python-2.7.2/configure
===================================================================
--- Python-2.7.2.orig/configure
+++ Python-2.7.2/configure
@@ -3007,6 +3007,7 @@
 	darwin*) MACHDEP="darwin";;
 	atheos*) MACHDEP="atheos";;
         irix646) MACHDEP="irix6";;
+        linux*) MACHDEP="linux2";;
 	'')	MACHDEP="unknown";;
 	esac
 fi
Index: Python-2.7.2/configure.in
===================================================================
--- Python-2.7.2.orig/configure.in
+++ Python-2.7.2/configure.in
@@ -297,6 +297,7 @@
 	darwin*) MACHDEP="darwin";;
 	atheos*) MACHDEP="atheos";;
         irix646) MACHDEP="irix6";;
+        linux*) MACHDEP="linux2";;
 	'')	MACHDEP="unknown";;
 	esac
 fi
