diff -ur diffutils-2.8.4/lib/regex.c diffutils-2.8.4-patched/lib/regex.c
--- diffutils-2.8.4/lib/regex.c	2002-03-01 01:44:52.000000000 -0600
+++ diffutils-2.8.4-patched/lib/regex.c	2005-12-08 00:44:03.101524750 -0600
@@ -8314,7 +8314,7 @@
       if (msg_size > errbuf_size)
         {
 #if defined HAVE_MEMPCPY || defined _LIBC
-	  *((char *) __mempcpy (errbuf, msg, errbuf_size - 1)) = '\0';
+	  *((char *) mempcpy (errbuf, msg, errbuf_size - 1)) = '\0';
 #else
           memcpy (errbuf, msg, errbuf_size - 1);
           errbuf[errbuf_size - 1] = 0;
