--- a/configure.in
+++ b/configure.in
@@ -11,7 +11,10 @@
 AC_PROG_LN_S
 
 dnl Checks for libraries.
-AC_CHECK_LIB(fltk,numericsort,,AC_MSG_ERROR("missing fltk"))
+AC_CHECK_LIB(fltk,numericsort,,
+AC_CHECK_LIB(fltk,fl_numericsort,,
+AC_MSG_ERROR("missing fltk"))
+)
 AM_PATH_ALSA(0.9.0)
 
 dnl Checks for header files.
--- a/src/alsamixer.cxx	2007-10-21 12:39:11.000000000 -0400
+++ b/src/alsamixer.cxx	2007-10-21 12:49:10.000000000 -0400
@@ -2081,8 +2081,8 @@ static void
 mixer_signal_handler (int signal)
 {
   if (signal != SIGSEGV)
-    mixer_abort (ERR_SIGNAL, sys_siglist[signal], 0);
-  else
+  //  mixer_abort (ERR_SIGNAL, sys_siglist[signal], 0);
+  //else
     {
       fprintf (stderr, "\nSegmentation fault.\n");
       _exit (11);
--- a/src/alsamixer.cxx	2006-01-16 12:41:54.000000000 -0600
+++ b/src/alsamixer.cxx	2006-01-16 12:45:44.000000000 -0600
@@ -653,8 +653,8 @@
     if (vleft >= 0 && vright >= 0) {
       if (joined) {
 #ifdef ALSAMIXER_GUI
-	for (chn = snd_mixer_selem_channel_id_t(0); chn < SND_MIXER_SCHN_LAST; 
-	snd_mixer_selem_channel_id_t(int(chn)++))
+	for (chn = snd_mixer_selem_channel_id_t(0); chn < SND_MIXER_SCHN_LAST;
+	snd_mixer_selem_channel_id_t(int(chn)+1))
 #else
 	for (chn = 0; chn < SND_MIXER_SCHN_LAST; chn++)
 #endif
