From 2e1b003600156e4adcb88998eabf18addee45be1 Mon Sep 17 00:00:00 2001
From: Giulio Benetti <giulio.benetti@benettiengineering.com>
Date: Sun, 8 Dec 2019 11:57:45 +0100
Subject: [PATCH] Bug 1602288 - Fix build failure due to missing posix signal.h

Signed-off-by: Giulio Benetti <giulio.benetti@benettiengineering.com>
---
 nss/coreconf/Linux.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nss/coreconf/Linux.mk b/nss/coreconf/Linux.mk
index d07f8a3c5..854d3ca96 100644
--- a/nss/coreconf/Linux.mk
+++ b/nss/coreconf/Linux.mk
@@ -21,7 +21,7 @@ ifeq ($(USE_PTHREADS),1)
 endif
 
 DEFAULT_COMPILER = gcc
-DEFINES += -D_DEFAULT_SOURCE -D_BSD_SOURCE
+DEFINES += -D_DEFAULT_SOURCE -D_BSD_SOURCE -D_POSIX_SOURCE
 
 ifeq ($(OS_TARGET),Android)
 ifndef ANDROID_NDK
-- 
2.20.1

