From b1f2f1f71894fa0af92041fb4d5b6146101d63e5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Stefan=20S=C3=B8rensen?= <stefan.sorensen@spectralink.com>
Date: Thu, 2 Feb 2017 11:17:26 +0100
Subject: [PATCH] Fix memory leak in settings when using bad word list.

Signed-off-by: Stefan Sørensen <stefan.sorensen@spectralink.com>

---

Status: accepted upstream, not yet released.

diff --git a/src/settings.c b/src/settings.c
index 5c38b30..4b026b1 100644
--- a/src/settings.c
+++ b/src/settings.c
@@ -46,6 +46,7 @@ pwquality_free_settings(pwquality_settings_t *pwq)
 {
         if (pwq) {
                 free(pwq->dict_path);
+                free(pwq->bad_words);
                 free(pwq);
         }
 }
-- 
2.9.3

