Fix build failure with gcc 4.7+
From bugtracker: http://sourceforge.net/p/dbus-cplusplus/patches/13/

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff --git a/include/dbus-c++/eventloop-integration.h b/include/dbus-c++/eventloop-integration.h
index 1b0302e..3e44304 100644
--- a/include/dbus-c++/eventloop-integration.h
+++ b/include/dbus-c++/eventloop-integration.h
@@ -26,6 +26,7 @@
 #define __DBUSXX_EVENTLOOP_INTEGRATION_H
 
 #include <errno.h>
+#include <unistd.h>
 #include "api.h"
 #include "dispatcher.h"
 #include "util.h"
