diff -urpN busybox-1.12.1/coreutils/basename.c busybox-1.12.1-basename/coreutils/basename.c
--- busybox-1.12.1/coreutils/basename.c	2008-09-28 20:04:18.000000000 +0200
+++ busybox-1.12.1-basename/coreutils/basename.c	2008-11-06 16:10:54.000000000 +0100
@@ -48,5 +48,5 @@ int basename_main(int argc, char **argv)
 
 	/* puts(s) will do, but we can do without stdio this way: */
 	s[m++] = '\n';
-	return full_write(STDOUT_FILENO, s, m) == (ssize_t)m;
+	return full_write(STDOUT_FILENO, s, m) != (ssize_t)m;
 }
