--- busybox-1.12.1/libbb/lineedit.c	Sun Sep 28 20:04:20 2008
+++ busybox-1.12.1-lineedit/libbb/lineedit.c	Sun Nov  2 01:39:52 2008
@@ -1415,7 +1415,8 @@
 	if ((state->flags & SAVE_HISTORY) && state->hist_file)
 		load_history(state->hist_file);
 #endif
-	state->cur_history = state->cnt_history;
+	if (state->flags & DO_HISTORY)
+		state->cur_history = state->cnt_history;
 
 	/* prepare before init handlers */
 	cmdedit_y = 0;  /* quasireal y, not true if line > xt*yt */
