--- busybox-1.13.1/miscutils/crond.c	Sun Nov  9 18:28:17 2008
+++ busybox-1.13.1-crond/miscutils/crond.c	Thu Dec  4 14:56:10 2008
@@ -779,6 +779,8 @@
 			xmove_fd(mailFd, mail_filename ? 1 : 0);
 			dup2(1, 2);
 		}
+		/* crond 3.0pl1-100 puts tasks in separate process groups */
+		bb_setpgrp();
 		execlp(prog, prog, cmd, arg, NULL);
 		crondlog(ERR20 "can't exec, user %s cmd %s %s %s", user, prog, cmd, arg);
 		if (mail_filename) {
@@ -914,6 +916,8 @@
 		if (DebugOpt) {
 			crondlog(LVL5 "child running %s", DEFAULT_SHELL);
 		}
+		/* crond 3.0pl1-100 puts tasks in separate process groups */
+		bb_setpgrp();
 		execl(DEFAULT_SHELL, DEFAULT_SHELL, "-c", line->cl_Shell, NULL);
 		crondlog(ERR20 "can't exec, user %s cmd %s %s %s", user,
 				 DEFAULT_SHELL, "-c", line->cl_Shell);
