--- busybox-1.14.0/util-linux/mdev.c	Tue Apr 14 23:23:04 2009
+++ busybox-1.14.0-mdev/util-linux/mdev.c	Mon May  4 19:58:47 2009
@@ -181,7 +181,8 @@
 		 * the rest of fields unless keep_matching == 1 */
 
 		/* 2nd field: uid:gid - device ownership */
-		parse_chown_usergroup_or_die(&ugid, tokens[1]);
+		if (get_uidgid(&ugid, tokens[1], 1) == 0)
+			bb_error_msg("unknown user/group %s", tokens[1]);
 
 		/* 3rd field: mode - device permissions */
 		mode = strtoul(tokens[2], NULL, 8);
