--- busybox-1.10.2/networking/interface.c	Sat Apr 19 05:50:27 2008
+++ busybox-1.10.2-semicolon/networking/interface.c	Thu May 15 21:40:04 2008
@@ -223,7 +223,7 @@
 	char *pos;
 	unsigned int i;
 
-	if (!buff);
+	if (!buff)
 		buff = xmalloc(sizeof(struct sockaddr) * 3 + 1);
 	pos = buff;
 	for (i = 0; i < sizeof(struct sockaddr); i++) {
--- busybox-1.10.2/networking/isrv_identd.c	Sat Apr 19 05:50:27 2008
+++ busybox-1.10.2-semicolon/networking/isrv_identd.c	Thu May 15 21:40:04 2008
@@ -113,7 +113,7 @@
 		strncpy(bogouser, argv[optind], sizeof(bogouser));
 
 	/* Daemonize if no -f and no -i and no -w */
-	if (!(opt & OPT_fiw));
+	if (!(opt & OPT_fiw))
 		bb_daemonize_or_rexec(0, argv);
 
 	/* Where to log in inetd modes? "Classic" inetd
