diff -urN busybox-1.9.0/Makefile.help busybox-1.9.0-allno/Makefile.help
--- busybox-1.9.0/Makefile.help	2007-12-21 22:00:33.000000000 +0000
+++ busybox-1.9.0-allno/Makefile.help	2008-01-04 20:03:30.000000000 +0000
@@ -16,7 +16,6 @@
 	@echo 'Configuration:'
 	@echo '  allnoconfig		- disable all symbols in .config'
 	@echo '  allyesconfig		- enable all symbols in .config (see defconfig)'
-	@echo '  allbareconfig		- enable all applets without any sub-features'
 	@echo '  config		- text based configurator (of last resort)'
 	@echo '  defconfig		- set .config to largest generic configuration'
 	@echo '  menuconfig		- interactive curses-based configurator'
diff -urN busybox-1.9.0/applets/applet_tables.c busybox-1.9.0-allno/applets/applet_tables.c
--- busybox-1.9.0/applets/applet_tables.c	2007-12-24 14:08:25.000000000 +0000
+++ busybox-1.9.0-allno/applets/applet_tables.c	2008-01-04 20:00:12.000000000 +0000
@@ -71,7 +71,7 @@
 
 	puts("/* This is a generated file, don't edit */");
 
-	puts("const char applet_names[] ALIGN1 =");
+	puts("const char applet_names[] ALIGN1 = \"\" \n");
 	for (i = 0; i < NUM_APPLETS; i++) {
 		printf("\"%s\" \"\\0\"\n", applets[i].name);
 	}
