--- busybox-1.6.0/shell/Config.in	Fri Jun  1 13:48:33 2007
+++ busybox-1.6.0-hushdep/shell/Config.in	Sun Jun 10 01:02:47 2007
@@ -179,6 +179,7 @@
 config HUSH_HELP
 	bool "help builtin"
 	default n
+	depends on HUSH
 	help
 	  Enable help builtin in hush. Code size + ~1 kbyte.
 
@@ -206,18 +207,21 @@
 config HUSH_TICK
 	bool "Process substitution"
 	default n
+	depends on HUSH
 	help
 	  Enable process substitution `command` and $(command) in hush.
 
 config HUSH_IF
 	bool "Support if/then/elif/else/fi"
 	default n
+	depends on HUSH
 	help
 	  Enable if/then/elif/else/fi in hush.
 
 config HUSH_LOOPS
 	bool "Support for, while and until loops"
 	default n
+	depends on HUSH
 	help
 	  Enable for, while and until loops in hush.
 
