--- busybox-1.14.0/networking/udhcp/leases.c	Tue Apr 14 01:42:51 2009
+++ busybox-1.14.0-dhcp_hostname/networking/udhcp/leases.c	Tue Apr 21 02:41:22 2009
@@ -63,7 +63,8 @@
 	if (oldest) {
 		oldest->hostname[0] = '\0';
 		if (hostname) {
-        		hostname_length = hostname[-1]; /* look at option size byte */
+			/* option size byte, + 1 for NUL */
+        		hostname_length = hostname[-1] + 1;
 			if (hostname_length > sizeof(oldest->hostname))
 				hostname_length = sizeof(oldest->hostname);
             		hostname = (uint8_t*) safe_strncpy((char*)oldest->hostname, (char*)hostname, hostname_length);
