--- busybox-1.12.2/libbb/lineedit.c	Sun Nov  9 18:20:16 2008
+++ busybox-1.12.2-lineedit/libbb/lineedit.c	Tue Nov 11 21:02:04 2008
@@ -1415,8 +1415,10 @@
 	if ((state->flags & SAVE_HISTORY) && state->hist_file)
 		load_history(state->hist_file);
 #endif
+#if MAX_HISTORY > 0
 	if (state->flags & DO_HISTORY)
 		state->cur_history = state->cnt_history;
+#endif
 
 	/* prepare before init handlers */
 	cmdedit_y = 0;  /* quasireal y, not true if line > xt*yt */
