diff -urpN busybox-1.15.0/modutils/modprobe-small.c busybox-1.15.0-modprobe/modutils/modprobe-small.c
--- busybox-1.15.0/modutils/modprobe-small.c	2009-08-21 00:26:14.000000000 +0200
+++ busybox-1.15.0-modprobe/modutils/modprobe-small.c	2009-09-07 02:43:01.000000000 +0200
@@ -218,6 +218,7 @@ static void parse_module(module_info *in
 	bksp(); /* remove last ' ' */
 	appendc('\0');
 	info->aliases = copy_stringbuf();
+	replace(info->aliases, '-', '_');
 
 	/* "dependency1 depandency2" */
 	reset_stringbuf();
