diff -urN busybox-1.9.0/networking/nameif.c busybox-1.9.0-nameif/networking/nameif.c
--- busybox-1.9.0/networking/nameif.c	2007-12-24 14:08:25.000000000 +0000
+++ busybox-1.9.0-nameif/networking/nameif.c	2008-01-04 20:05:48.000000000 +0000
@@ -50,18 +50,18 @@
 #define ETHTOOL_BUSINFO_LEN 32
 /* these strings are set to whatever the driver author decides... */
 struct ethtool_drvinfo {
-	__u32 cmd;
-	char  driver[32]; /* driver short name, "tulip", "eepro100" */
-	char  version[32];  /* driver version string */
-	char  fw_version[32]; /* firmware version string, if applicable */
-	char  bus_info[ETHTOOL_BUSINFO_LEN];  /* Bus info for this IF. */
+	uint32_t cmd;
+	char driver[32]; /* driver short name, "tulip", "eepro100" */
+	char version[32];  /* driver version string */
+	char fw_version[32]; /* firmware version string, if applicable */
+	char bus_info[ETHTOOL_BUSINFO_LEN];  /* Bus info for this IF. */
         /* For PCI devices, use pci_dev->slot_name. */
-	char  reserved1[32];
-	char  reserved2[16];
-	__u32 n_stats;  /* number of u64's from ETHTOOL_GSTATS */
-	__u32 testinfo_len;
-	__u32 eedump_len; /* Size of data from ETHTOOL_GEEPROM (bytes) */
-	__u32 regdump_len;  /* Size of data from ETHTOOL_GREGS (bytes) */
+	char reserved1[32];
+	char reserved2[16];
+	uint32_t n_stats;  /* number of u64's from ETHTOOL_GSTATS */
+	uint32_t testinfo_len;
+	uint32_t eedump_len; /* Size of data from ETHTOOL_GEEPROM (bytes) */
+	uint32_t regdump_len;  /* Size of data from ETHTOOL_GREGS (bytes) */
 };
 #define ETHTOOL_GDRVINFO  0x00000003 /* Get driver info. */
 #endif
