diff -urpN busybox-1.13.3/coreutils/tail.c busybox-1.13.3-tail/coreutils/tail.c
--- busybox-1.13.3/coreutils/tail.c	2009-02-26 12:47:02.000000000 +0100
+++ busybox-1.13.3-tail/coreutils/tail.c	2009-03-27 03:34:57.000000000 +0100
@@ -104,7 +104,7 @@ int tail_main(int argc, char **argv)
 	if (argv[1] && (argv[1][0] == '+' || argv[1][0] == '-')
 	 && isdigit(argv[1][1])
 	) {
-		count = eat_num(&argv[1][1]);
+		count = eat_num(argv[1]);
 		argv++;
 		argc--;
 	}
