--- busybox-1.4.1/include/libbb.h	Wed Jan 24 22:34:48 2007
+++ busybox-1.4.1-binhex/include/libbb.h	Sat Jan 27 00:32:01 2007
@@ -690,7 +690,7 @@
 extern const char bb_msg_standard_output[];
 
 extern const char bb_str_default[];
-/* NB: (bb_hexdigits_upcase[i] | 0x10) -> lowercase hex digit */
+/* NB: (bb_hexdigits_upcase[i] | 0x20) -> lowercase hex digit */
 extern const char bb_hexdigits_upcase[];
 
 extern const char bb_path_mtab_file[];
--- busybox-1.4.1/libbb/xfuncs.c	Wed Jan 24 22:49:25 2007
+++ busybox-1.4.1-binhex/libbb/xfuncs.c	Sat Jan 27 00:32:01 2007
@@ -340,8 +340,8 @@
 	while (count) {
 		unsigned char c = *cp++;
 		/* put lowercase hex digits */
-		*p++ = 0x10 | bb_hexdigits_upcase[c >> 4];
-		*p++ = 0x10 | bb_hexdigits_upcase[c & 0xf];
+		*p++ = 0x20 | bb_hexdigits_upcase[c >> 4];
+		*p++ = 0x20 | bb_hexdigits_upcase[c & 0xf];
 		count--;
 	}
 	return p;
