diff -urpN busybox-1.15.1/miscutils/hdparm.c busybox-1.15.1-buildsys/miscutils/hdparm.c
--- busybox-1.15.1/miscutils/hdparm.c	2009-09-12 17:55:36.000000000 +0200
+++ busybox-1.15.1-buildsys/miscutils/hdparm.c	2009-09-22 23:13:25.000000000 +0200
@@ -15,6 +15,9 @@
 /* must be _after_ libbb.h: */
 #include <linux/hdreg.h>
 #include <sys/mount.h>
+#if !defined(BLKGETSIZE64)
+# define BLKGETSIZE64 _IOR(0x12,114,size_t)
+#endif
 
 /* device types */
 /* ------------ */
diff -urpN busybox-1.15.1/util-linux/fdisk.c busybox-1.15.1-buildsys/util-linux/fdisk.c
--- busybox-1.15.1/util-linux/fdisk.c	2009-09-12 17:55:37.000000000 +0200
+++ busybox-1.15.1-buildsys/util-linux/fdisk.c	2009-09-22 23:13:25.000000000 +0200
@@ -9,13 +9,16 @@
 
 #ifndef _LARGEFILE64_SOURCE
 /* For lseek64 */
-#define _LARGEFILE64_SOURCE
+# define _LARGEFILE64_SOURCE
 #endif
 #include <assert.h>             /* assert */
 #include <sys/mount.h>
 #if !defined(BLKSSZGET)
 # define BLKSSZGET _IO(0x12, 104)
 #endif
+#if !defined(BLKGETSIZE64)
+# define BLKGETSIZE64 _IOR(0x12,114,size_t)
+#endif
 #include "libbb.h"
 
 /* Looks like someone forgot to add this to config system */
