--- busybox-1.9.1/networking/httpd.c	Tue Feb 12 17:03:01 2008
+++ busybox-1.9.1-httpd-POST/networking/httpd.c	Mon Mar 17 13:58:09 2008
@@ -1950,7 +1950,7 @@
 			if ((STRNCASECMP(iobuf, "Content-length:") == 0)) {
 				/* extra read only for POST */
 				if (prequest != request_GET) {
-					tptr = iobuf + sizeof("Content-length:") - 1;
+					tptr = tptr = skip_whitespace(iobuf + sizeof("Content-length:") - 1);
 					if (!tptr[0])
 						send_headers_and_exit(HTTP_BAD_REQUEST);
 					errno = 0;
