--- busybox-1.11.1/modutils/insmod.c	Wed Jun 25 14:51:18 2008
+++ busybox-1.11.1-modutils/modutils/insmod.c	Wed Jul 23 00:46:08 2008
@@ -840,7 +840,8 @@
 				ElfW(RelM) *rel, ElfW(Addr) v)
 {
 #if defined(__arm__) || defined(__i386__) || defined(__mc68000__) \
- || defined(__sh__) || defined(__s390__) || defined(__x86_64__)
+ || defined(__sh__) || defined(__s390__) || defined(__x86_64__) \
+ || defined(__powerpc__) || defined(__mips__)
 	struct arch_file *ifile = (struct arch_file *) f;
 #endif
 	enum obj_reloc ret = obj_reloc_ok;
