diff -urpN busybox-1.16.0/shell/hush.c busybox-1.16.0-hush/shell/hush.c
--- busybox-1.16.0/shell/hush.c	2010-01-25 17:51:24.000000000 +0100
+++ busybox-1.16.0-hush/shell/hush.c	2010-03-22 01:46:39.000000000 +0100
@@ -3568,7 +3568,9 @@ static void execvp_or_die(char **argv)
 {
 	debug_printf_exec("execing '%s'\n", argv[0]);
 	sigprocmask(SIG_SETMASK, &G.inherited_set, NULL);
-	execvp(argv[0], argv);
+	/* if FEATURE_SH_STANDALONE, "exec <applet_name>" should work,
+	 * therefore we should use BB_EXECVP, not execvp */
+	BB_EXECVP(argv[0], argv);
 	bb_perror_msg("can't execute '%s'", argv[0]);
 	_exit(127); /* bash compat */
 }
