diff -urpN busybox-1.16.1/networking/udhcp/leases.c busybox-1.16.1-dhcpd/networking/udhcp/leases.c
--- busybox-1.16.1/networking/udhcp/leases.c	2010-03-28 19:43:36.000000000 +0200
+++ busybox-1.16.1-dhcpd/networking/udhcp/leases.c	2010-05-15 20:47:08.000000000 +0200
@@ -64,6 +64,8 @@ struct dyn_lease* FAST_FUNC add_lease(
 		oldest->hostname[0] = '\0';
 		if (hostname) {
 			char *p;
+
+			hostname_len++; /* include NUL */
 			if (hostname_len > sizeof(oldest->hostname))
 				hostname_len = sizeof(oldest->hostname);
 			p = safe_strncpy(oldest->hostname, hostname, hostname_len);
