--- busybox-1.10.0/miscutils/taskset.c	Thu Mar 20 12:46:55 2008
+++ busybox-1.10.0-taskset/miscutils/taskset.c	Sat Mar 22 18:46:21 2008
@@ -50,6 +50,11 @@
 	char *pid_str;
 	char *aff = aff; /* for compiler */
 
+	/* NB: we mimic util-linux's taskset: -p does not take
+	 * an argument, i.e., "-pN" is NOT valid, only "-p N"!
+	 * Indeed, util-linux-2.13-pre7 uses:
+	 * getopt_long(argc, argv, "+pchV", ...), not "...p:..." */
+
 	opt_complementary = "-1"; /* at least 1 arg */
 	opt_p = getopt32(argv, "+p");
 	argv += optind;
--- busybox-1.10.0/testsuite/taskset.tests	Thu Mar 20 12:47:04 2008
+++ busybox-1.10.0-taskset/testsuite/taskset.tests	Sat Mar 22 18:46:21 2008
@@ -7,8 +7,8 @@
 a="taskset"
 
 # testing "test name"              "opts" "expected result" "file inp" "stdin"
-testing "taskset (get from pid 1)" "$a -p1 >/dev/null;echo \$?" "0\n" "" ""
-testing "taskset (invalid pid)"    "$a -p0 >/dev/null 2>&1;echo \$?" "1\n" "" ""
+testing "taskset (get from pid 1)" "$a -p 1 >/dev/null;echo \$?" "0\n" "" ""
+testing "taskset (invalid pid)"    "$a -p 0 >/dev/null 2>&1;echo \$?" "1\n" "" ""
 testing "taskset (set_aff, needs CAP_SYS_NICE)" \
                                    "$a 0x1 $SHELL -c $a\ -p\ \$$\|grep\ \"current\ affinity\ mask:\ 1\" >/dev/null;echo \$?" \
 				                                "0\n" "" ""
