Fix cross compilation, based on OpenWRT patch.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura libtorrent-0.13.3.orig/configure.ac libtorrent-0.13.3/configure.ac
--- libtorrent-0.13.3.orig/configure.ac	2012-10-31 07:38:59.345752076 -0300
+++ libtorrent-0.13.3/configure.ac	2012-10-31 07:39:06.454799361 -0300
@@ -19,7 +19,6 @@
 
 AM_INIT_AUTOMAKE
 AM_CONFIG_HEADER(config.h)
-AM_PATH_CPPUNIT(1.9.6)
 
 AC_PROG_CXX
 
diff -Nura libtorrent-0.13.3.orig/scripts/checks.m4 libtorrent-0.13.3/scripts/checks.m4
--- libtorrent-0.13.3.orig/scripts/checks.m4	2012-10-31 07:38:59.342752057 -0300
+++ libtorrent-0.13.3/scripts/checks.m4	2012-10-31 07:39:06.454799361 -0300
@@ -96,7 +96,7 @@
 AC_DEFUN([TORRENT_CHECK_KQUEUE_SOCKET_ONLY], [
   AC_MSG_CHECKING(whether kqueue supports pipes and ptys)
 
-  AC_RUN_IFELSE([AC_LANG_SOURCE([
+  AC_LINK_IFELSE([AC_LANG_SOURCE([
       #include <fcntl.h>
       #include <stdlib.h>
       #include <unistd.h>
diff -Nura libtorrent-0.13.3.orig/scripts/common.m4 libtorrent-0.13.3/scripts/common.m4
--- libtorrent-0.13.3.orig/scripts/common.m4	2012-10-31 07:38:59.342752057 -0300
+++ libtorrent-0.13.3/scripts/common.m4	2012-10-31 07:39:32.550972826 -0300
@@ -223,7 +223,7 @@
 AC_DEFUN([TORRENT_CHECK_EXECINFO], [
   AC_MSG_CHECKING(for execinfo.h)
 
-  AC_RUN_IFELSE([AC_LANG_SOURCE([
+  AC_LINK_IFELSE([AC_LANG_SOURCE([
       #include <execinfo.h>
       int main() { backtrace((void**)0, 0); backtrace_symbols((char**)0, 0); return 0;}
       ])],
