From c1822a19e75bae8e044cd46a257a234cd9649896 Mon Sep 17 00:00:00 2001
From: Olivier Schonken <olivier.schonken@gmail.com>
Date: Wed, 20 Jan 2016 23:46:50 +0200
Subject: [PATCH] Changed Makefile.am to allow cross-compilation of HPLIP

Signed-off-by: Olivier Schonken <olivier.schonken@gmail.com>
---
 Makefile.am | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 73fed2e..23aa0ec 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -92,7 +92,7 @@ libhpmud_la_SOURCES += io/hpmud/musb_libusb01.c
 libhpmud_la_LDFLAGS += -lusb
 else
 libhpmud_la_SOURCES += io/hpmud/musb.c 
-libhpmud_la_CFLAGS  += -I/usr/include/libusb-1.0
+libhpmud_la_CFLAGS  += -I$(includedir)/libusb-1.0
 libhpmud_la_LDFLAGS += -lusb-1.0
 endif
 
@@ -314,7 +314,7 @@ hpmudext_la_LIBADD = libhpmud.la libhpdiscovery.la
 hpmudext_la_CFLAGS = -I$(PYTHONINCLUDEDIR) -Iprotocol/discovery
 
 if !LIBUSB01_BUILD
-hpmudext_la_CFLAGS +=-I/usr/include/libusb-1.0 
+hpmudext_la_CFLAGS +=-I$(includedir)/libusb-1.0 
 endif
 
 # ui (qt3)
-- 
2.5.0

