diff -urpN busybox-1.17.3/libbb/unicode.c busybox-1.17.3-unicode/libbb/unicode.c
--- busybox-1.17.3/libbb/unicode.c	2010-10-09 21:58:13.000000000 +0200
+++ busybox-1.17.3-unicode/libbb/unicode.c	2010-10-29 00:48:36.951494079 +0200
@@ -1006,7 +1006,7 @@ static char* FAST_FUNC unicode_conv_to_p
 			}
 		}
 		if (stats)
-			stats->byte_count = stats->unicode_count = (d - dst);
+			stats->byte_count = stats->unicode_count = stats->unicode_width = (d - dst);
 		return dst;
 	}
 
