diff -urpN busybox-1.17.3/networking/libiproute/iproute.c busybox-1.17.3-iproute/networking/libiproute/iproute.c
--- busybox-1.17.3/networking/libiproute/iproute.c	2010-10-09 21:57:14.000000000 +0200
+++ busybox-1.17.3-iproute/networking/libiproute/iproute.c	2010-11-04 09:13:06.307919550 +0100
@@ -791,8 +791,8 @@ static int iproute_get(char **argv)
 				}
 				req.r.rtm_dst_len = addr.bitlen;
 			}
-			argv++;
 		}
+		argv++;
 	}
 
 	if (req.r.rtm_dst_len == 0) {
