diff -urpN busybox-1.17.3/networking/dnsd.c busybox-1.17.3-dnsd/networking/dnsd.c
--- busybox-1.17.3/networking/dnsd.c	2010-10-09 21:57:14.000000000 +0200
+++ busybox-1.17.3-dnsd/networking/dnsd.c	2010-10-29 00:37:43.066043841 +0200
@@ -388,7 +388,7 @@ static int process_packet(struct dns_ent
 	query_len = strlen(query_string) + 1;
 	/* may be unaligned! */
 	unaligned_type_class = (void *)(query_string + query_len);
-	query_len += sizeof(unaligned_type_class);
+	query_len += sizeof(*unaligned_type_class);
 	/* where to append answer block */
 	answb = (void *)(unaligned_type_class + 1);
 
