Add a pkgconfig file to ease usage with applications

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: qextserialport-ef4af2a2ee3f/qextserialport.pc
===================================================================
--- /dev/null
+++ qextserialport-ef4af2a2ee3f/qextserialport.pc
@@ -0,0 +1,10 @@
+prefix=/usr
+exec_prefix=${prefix}
+libdir=${prefix}/lib
+includedir=${prefix}/include/QExtSerialPort
+
+Name: QtExtSerialPort
+Description: QtExtSerialPort library
+Version: 1.2.0
+Libs: -L${libdir} -lqextserialport
+Cflags: -I${includedir}
