Disable compilation of examples

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>

Index: qextserialport-ef4af2a2ee3f/qextserialport.pro
===================================================================
--- qextserialport-ef4af2a2ee3f.orig/qextserialport.pro
+++ qextserialport-ef4af2a2ee3f/qextserialport.pro
@@ -1,7 +1,4 @@
 #
 TEMPLATE = subdirs
 CONFIG   += ordered
-SUBDIRS  = src \
-           examples/enumerator \
-           examples/event
-
+SUBDIRS  = src
