From fba2627bebc7d497f64827a0517316e9b5c64c0a Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@e-labworks.com>
Date: Tue, 26 Jul 2016 13:24:00 -0300
Subject: [PATCH 1/2] Add cross-compilation support

Signed-off-by: Sergio Prado <sergio.prado@e-labworks.com>
---
 configure | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 0d90a4f0acde..a4afea8e1880 100755
--- a/configure
+++ b/configure
@@ -502,8 +502,9 @@ if test -n "$_host"; then
 			_host_os=win32
 			;;
 		*)
-			echo "Cross-compiling to unknown target, please add your target to configure."
-			exit 1
+			echo "Cross-compiling to $_host target"
+			DEFINES="$DEFINES -DUNIX"
+			_host_os=unix
 			;;
 	esac
 	
-- 
1.9.1

