From ef1cffc526d3cf9562d50efbf8addc2aa4c06b52 Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@e-labworks.com>
Date: Sun, 30 Jul 2017 18:23:23 -0300
Subject: [PATCH 1/2] Add cross-compilation support

Signed-off-by: Sergio Prado <sergio.prado@e-labworks.com>
---
 configure | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index c2f304729651..b05321aac3da 100755
--- a/configure
+++ b/configure
@@ -499,8 +499,9 @@ if test -n "$_host"; then
 			_host_os=win32
 			;;
 		*)
-			echo "Cross-compiling to unknown target, please add your target to configure."
-			exit 1
+			echo "Cross-compiling to $_host target"
+			DEFINES="$DEFINES -DUNIX"
+			_host_os=unix
 			;;
 	esac
 
-- 
1.9.1

