--- lzma-4.32.0beta3.orig/src/lzma/lzmp.cpp	2006-07-06 23:58:46.000000000 +0200
+++ lzma-4.32.0beta3/src/lzma/lzmp.cpp	2006-08-29 20:21:22.000000000 +0200
@@ -47,7 +47,11 @@ typedef vector<string> stringVector;
 
 // For Solaris
 #ifndef HAVE_FUTIMES
+#ifdef HAVE_FUTIMESAT
 #define futimes(fd, tv) futimesat(fd, NULL, tv)
+#else
+#define futimes(fd, tv) 0 /* utimes(fd, NULL, tv)*/
+#endif
 #endif
 
 #if defined(_WIN32) || defined(OS2) || defined(MSDOS)
