diff -urN xlib_libXfont-1.2.7.orig/src/builtins/file.c xlib_libXfont-1.2.7/src/builtins/file.c
--- xlib_libXfont-1.2.7.orig/src/builtins/file.c	2007-01-17 23:49:10.000000000 +0200
+++ xlib_libXfont-1.2.7/src/builtins/file.c	2007-09-30 03:09:29.000000000 +0300
@@ -50,7 +50,7 @@
     len = BUFFILESIZE;
     if (len > left)
 	len = left;
-    bcopy (io->file->bits + io->offset, f->buffer, len);
+    memmove (f->buffer, io->file->bits + io->offset, len);
     io->offset += len;
     f->left = len - 1;
     f->bufp = f->buffer + 1;
