From 9e95f88906f850caa789977dab4b0f256b575254 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Tue, 7 Mar 2017 23:31:11 +0100
Subject: [PATCH] Add options to disable zlib, bzip2 and xz modules

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 configure.ac | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/configure.ac b/configure.ac
index 484f6ebaca..79538df0cf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3172,6 +3172,24 @@ AC_ARG_ENABLE(readline,
 	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} readline"
 	  fi])
 
+AC_ARG_ENABLE(bzip2,
+	AS_HELP_STRING([--disable-bzip2], [disable bzip2]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _bz2"
+	  fi])
+
+AC_ARG_ENABLE(zlib,
+	AS_HELP_STRING([--disable-zlib], [disable zlib]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} zlib"
+	  fi])
+
+AC_ARG_ENABLE(xz,
+	AS_HELP_STRING([--disable-xz], [disable xz]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} _lzma"
+	  fi])
+
 AC_SUBST(TK)
 AC_ARG_ENABLE(tk,
 	AS_HELP_STRING([--disable-tk], [disable tk]),
-- 
2.13.5

