From e151b6863bd8f697f17b50eda2cecef57e433c3e Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 23 Dec 2015 13:58:00 +0100
Subject: [PATCH] Support PGEN_FOR_BUILD and FREEZE_IMPORTLIB_FOR_BUILD

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 Makefile.pre.in | 10 ++++++++++
 configure.ac    |  3 +++
 2 files changed, 13 insertions(+)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 0c26bcf..d1237eb 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -719,10 +719,15 @@ Programs/_testembed: Programs/_testembed.o $(LIBRARY) $(LDLIBRARY) $(PY3LIBRARY)
 ############################################################################
 # Importlib
 
+ifeq (@FREEZE_IMPORTLIB_FOR_BUILD@,)
 Programs/_freeze_importlib.o: Programs/_freeze_importlib.c Makefile
 
 Programs/_freeze_importlib: Programs/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN)
 	$(LINKCC) $(PY_LDFLAGS) -o $@ Programs/_freeze_importlib.o $(LIBRARY_OBJS_OMIT_FROZEN) $(LIBS) $(MODLIBS) $(SYSLIBS) $(LDLAST)
+else
+Programs/_freeze_importlib: @FREEZE_IMPORTLIB_FOR_BUILD@
+	cp $^ $@
+endif
 
 Python/importlib_external.h: $(srcdir)/Lib/importlib/_bootstrap_external.py Programs/_freeze_importlib
 	if test "$(cross_compiling)" != "yes"; then \
@@ -811,8 +816,13 @@ $(GRAMMAR_C): $(GRAMMAR_H)
 		mv $(GRAMMAR_C).tmp $(GRAMMAR_C); \
 	fi
 
+ifeq (@PGEN_FOR_BUILD@,)
 $(PGEN): $(PGENOBJS)
 		$(CC) $(OPT) $(PY_LDFLAGS) $(PGENOBJS) $(LIBS) -o $(PGEN)
+else
+$(PGEN): @PGEN_FOR_BUILD@
+		cp $^ $@
+endif
 
 Parser/grammar.o:	$(srcdir)/Parser/grammar.c \
 				$(srcdir)/Include/token.h \
diff --git a/configure.ac b/configure.ac
index 4e2b88e..16a827c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -80,6 +80,9 @@ else
 fi
 AC_SUBST(PYTHON_FOR_BUILD)
 
+AC_SUBST(PGEN_FOR_BUILD)
+AC_SUBST(FREEZE_IMPORTLIB_FOR_BUILD)
+
 dnl Ensure that if prefix is specified, it does not end in a slash. If
 dnl it does, we get path names containing '//' which is both ugly and
 dnl can cause trouble.
-- 
2.7.4

