From: Nicolas Cavallari <nicolas.cavallari@green-communications.fr>
Subject: [PATCH] Add an option to disable openssl support.

Signed-off-by: Nicolas Cavallari <nicolas.cavallari@green-communications.fr>
---
 configure.ac | 6 ++++++
 1 file changed, 6 insertions(+)

--- a/configure.ac
+++ b/configure.ac
@@ -3027,6 +3027,12 @@
     	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} unicodedata"
   	  fi])
 
+AC_ARG_ENABLE(openssl,
+	AS_HELP_STRING([--disable-openssl], [disable openssl support]),
+	[ if test "$enableval" = "no"; then
+	     DISABLED_EXTENSIONS="${DISABLED_EXTENSIONS} ssl _ssl _hashlib"
+	  fi])
+
 AC_SUBST(TK)
 AC_ARG_ENABLE(tk,
 	AS_HELP_STRING([--disable-tk], [disable tk]),
