diff -urpN busybox-1.17.0/networking/wget.c busybox-1.17.0-wget/networking/wget.c
--- busybox-1.17.0/networking/wget.c	2010-06-24 04:40:43.000000000 +0200
+++ busybox-1.17.0-wget/networking/wget.c	2010-07-12 03:27:57.000000000 +0200
@@ -50,7 +50,7 @@ static void progress_meter(int flag)
 	}
 
 	bb_progress_update(&G.pmt, G.curfile, G.beg_range, G.transferred,
-			   G.chunked ? 0 : G.content_len + G.beg_range);
+			   G.chunked ? 0 : G.beg_range + G.transferred + G.content_len);
 
 	if (flag == 0) {
 		/* last call to progress_meter */
