From 283676e2e8f15bedca20c3d94caeecfdebe11ffd Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Krause?= <joerg.krause@embedded.rocks>
Date: Mon, 23 Nov 2015 22:53:09 +0100
Subject: [PATCH 2/2] Install static library for static builds
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

[Gustavo: update for 4.1.2]

Signed-off-by: Jörg Krause <joerg.krause@embedded.rocks>
Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
 Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Makefile b/Makefile
index 86c8fac..fbaff63 100644
--- a/Makefile
+++ b/Makefile
@@ -163,6 +163,9 @@ ifeq ($(BUILD_SHARED),yes)
 	ln -sf $(LIBRARY_FULL_VERSION_SHARED) $(LIBDIR)/$(LIBRARY_VERSION_SHARED)
 	ln -sf $(LIBRARY_FULL_VERSION_SHARED) $(LIBDIR)/$(LIBRARY_SHARED)
 endif
+ifeq ($(BUILD_STATIC),yes)
+	install -m 755 $(BUILD_PATH)/$(LIBRARY_ARCHIVE) $(LIBDIR)/$(LIBRARY_ARCHIVE)
+endif
 	if [ -f $(BUILD_PATH)/$(CLIENT_TARGET) ]; then install -m 755 $(BUILD_PATH)/$(CLIENT_TARGET) $(BINDIR); fi
 	$(LDCONFIG)
 	@echo "use make installcheck to test the install"
-- 
2.4.10

