Fix libusb check when cross-compiling.

Signed-off-by: Julien Boibessot <julien.boibessot@armadeus.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>

Index: openobex-1.5/acinclude.m4
===================================================================
--- openobex-1.5.orig/acinclude.m4	2010-06-16 10:32:47.000000000 +0200
+++ openobex-1.5/acinclude.m4	2010-06-16 10:35:33.000000000 +0200
@@ -158,7 +158,7 @@
 		;;
 	*)
 		PKG_CHECK_MODULES(USB, libusb, usb_lib_found=yes, AC_MSG_RESULT(no))
-		AC_CHECK_FILE(${prefix}/lib/pkgconfig/libusb.pc, REQUIRES="libusb")
+		REQUIRES="libusb"
 		;;
 	esac
 	AC_SUBST(USB_CFLAGS)
