diff -uNr alsa-lib-1.2.2-orig/include/sound/uapi/asound.h alsa-lib-1.2.2-patched/include/sound/uapi/asound.h
--- alsa-lib-1.2.2-orig/include/sound/uapi/asound.h	2020-08-12 14:42:09.597313755 +0300
+++ alsa-lib-1.2.2-patched/include/sound/uapi/asound.h	2020-08-12 14:43:15.576887697 +0300
@@ -876,6 +876,11 @@
 #define SNDRV_TIMER_IOCTL_PAUSE		_IO('T', 0xa3)
 #define SNDRV_TIMER_IOCTL_TREAD64	_IOW('T', 0xa4, int)
 
+#ifndef __kernel_long_t
+typedef long        __kernel_long_t;
+typedef unsigned long    __kernel_ulong_t;
+#endif
+
 #if __BITS_PER_LONG == 64
 #define SNDRV_TIMER_IOCTL_TREAD SNDRV_TIMER_IOCTL_TREAD_OLD
 #else
