diff -ru rpm-5.2.0_vanilla/tools/Makefile.am rpm-5.2.0_exclude-some-tools/tools/Makefile.am
--- rpm-5.2.0_vanilla/tools/Makefile.am	2009-06-03 01:24:42.000000000 +0000
+++ rpm-5.2.0_exclude-some-tools/tools/Makefile.am	2009-12-20 07:47:13.000000000 +0000
@@ -45,9 +45,7 @@
 bin_PROGRAMS =		rpm2cpio
 
 pkgbindir =		@USRLIBRPM@/bin
-pkgbin_PROGRAMS =	\
-	rpmcache rpmdigest grep mtree rpmrepo rpmspecdump wget \
-	rpmcmp rpmdeps @WITH_KEYUTILS_RPMKEY@ @WITH_LIBELF_DEBUGEDIT@
+pkgbin_PROGRAMS =	
 dist_man_MANS =		rpmgrep.1
 
 debugedit_SOURCES =	debugedit.c hashtab.c
diff -ru rpm-5.2.0_vanilla/tools/Makefile.in rpm-5.2.0_exclude-some-tools/tools/Makefile.in
--- rpm-5.2.0_vanilla/tools/Makefile.in	2009-07-07 21:14:06.000000000 +0000
+++ rpm-5.2.0_exclude-some-tools/tools/Makefile.in	2009-12-20 07:47:37.000000000 +0000
@@ -39,11 +39,7 @@
 target_triplet = @target@
 EXTRA_PROGRAMS = rpmkey$(EXEEXT) debugedit$(EXEEXT)
 bin_PROGRAMS = rpm2cpio$(EXEEXT)
-pkgbin_PROGRAMS = rpmcache$(EXEEXT) rpmdigest$(EXEEXT) grep$(EXEEXT) \
-	mtree$(EXEEXT) rpmrepo$(EXEEXT) rpmspecdump$(EXEEXT) \
-	wget$(EXEEXT) rpmcmp$(EXEEXT) rpmdeps$(EXEEXT) \
-	@WITH_KEYUTILS_RPMKEY@ @WITH_LIBELF_DEBUGEDIT@ $(am__EXEEXT_1) \
-	$(am__EXEEXT_2)
+pkgbin_PROGRAMS = 
 @WITH_XAR_TRUE@am__append_1 = txar
 @WITH_DB_INTERNAL_TRUE@@WITH_DB_TOOLS_INTEGRATED_TRUE@am__append_2 = db_tool
 @WITH_DB_INTERNAL_TRUE@@WITH_DB_RPC_TRUE@@WITH_DB_TOOLS_INTEGRATED_TRUE@am__append_3 = \
