diff -ru rpm-5.2.0_vanilla/db/env/env_open.c rpm-5.2.0_test/db/env/env_open.c
--- rpm-5.2.0_vanilla/db/env/env_open.c	2008-05-28 01:23:27.000000000 +0000
+++ rpm-5.2.0_test/db/env/env_open.c	2009-12-24 14:54:55.000000000 +0000
@@ -124,7 +124,7 @@
 		}
 	}
 
-#ifdef HAVE_MUTEX_THREAD_ONLY
+#ifdef NK_HAVE_MUTEX_THREAD_ONLY
 	/*
 	 * Currently we support one kind of mutex that is intra-process only,
 	 * POSIX 1003.1 pthreads, because a variety of systems don't support
