diff -ru rpm-4.4.9_vanilla/rpmio/rpmio.h rpm-4.4.9_uclibc-vs-cookie-seek-pointer/rpmio/rpmio.h
--- rpm-4.4.9_vanilla/rpmio/rpmio.h	2007-04-28 15:55:33.000000000 +0000
+++ rpm-4.4.9_uclibc-vs-cookie-seek-pointer/rpmio/rpmio.h	2008-03-22 07:28:07.000000000 +0000
@@ -31,7 +31,7 @@
  * typedefs to isolate the lossage.
  */
 /*@{*/
-#if !defined(__LCLINT__) && defined(__GLIBC__) && \
+#if !defined(__LCLINT__) && defined(__GLIBC__) && !defined(__UCLIBC__) && \
 	(__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 2))
 #define USE_COOKIE_SEEK_POINTER 1
 typedef _IO_off64_t 	_libio_off_t;
