diff -ru rpm-5.0.3_vanilla/rpmio/rpmdav.c rpm-5.0.3_dirty-ssl-hack/rpmio/rpmdav.c
--- rpm-5.0.3_vanilla/rpmio/rpmdav.c	2008-01-29 13:57:50.000000000 +0000
+++ rpm-5.0.3_dirty-ssl-hack/rpmio/rpmdav.c	2008-04-09 17:06:39.000000000 +0000
@@ -84,7 +84,7 @@
 /* =============================================================== */
 void davDestroy(void)
 {
-#ifdef NE_FEATURE_SSL
+#if 0
     if (ne_has_support(NE_FEATURE_SSL)) {
 /* XXX http://www.nabble.com/Memory-Leaks-in-SSL_Library_init()-t3431875.html */
 	ENGINE_cleanup();
