Fix musl build

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr dvbsnoop-1.4.50.org/src/misc/helper.h dvbsnoop-1.4.50/src/misc/helper.h
--- dvbsnoop-1.4.50.org/src/misc/helper.h	2006-01-02 20:04:56.000000000 +0100
+++ dvbsnoop-1.4.50/src/misc/helper.h	2016-01-24 14:35:02.000000000 +0100
@@ -16,6 +16,7 @@
 #ifndef __HELPER_H
 #define __HELPER_H
 
+#include <sys/types.h>
 
 u_long outBit_Sx (int verbosity, const char *text, u_char *buf, int startbit, int bitlen);
 u_long outBit_Sx_NL (int verbosity, const char *text, u_char *buf, int startbit, int bitlen);
