--- ipsec-tools-0.6.6.oorig/src/racoon/cftoken.c	2006-06-13 10:49:01.000000000 +0200
+++ ipsec-tools-0.6.6/src/racoon/cftoken.c	2006-11-22 21:20:30.000000000 +0100
@@ -4076,8 +4076,11 @@
 			"Includes nested too deeply");
 		return -1;
 	}
-
+#ifdef GLOB_TILDE
 	if (glob(path, GLOB_TILDE, NULL, &incstack[incstackp].matches) != 0 ||
+#else
+	if (glob(path, 0, NULL, &incstack[incstackp].matches) != 0 ||
+#endif
 	    incstack[incstackp].matches.gl_pathc == 0) {
 		plog(LLV_ERROR, LOCATION, NULL,
 			"glob found no matches for path");
