--- curl-7.10.4/lib/Makefile.am
+++ curl-7.10.4/lib/Makefile.am
@@ -78,8 +78,8 @@
 
 install-data-hook:
 	@if test -n "@CURL_CA_BUNDLE@"; then \
-	  $(mkinstalldirs) `dirname $(DESTDIR)@CURL_CA_BUNDLE@`; \
-	  @INSTALL_DATA@ $(srcdir)/ca-bundle.crt $(DESTDIR)@CURL_CA_BUNDLE@; \
+	  $(mkinstalldirs) `dirname $(prefix)/../@CURL_CA_BUNDLE@`; \
+	  @INSTALL_DATA@ $(srcdir)/ca-bundle.crt $(prefix)/../@CURL_CA_BUNDLE@; \
         fi
 
 # this hook is mainly for non-unix systems to build even if configure
