Fix the broken libjson makefile.

Signed-off-by: Simon Dawson <spdawson@gmail.com>

diff -Nurp a/makefile b/makefile
--- a/makefile	2012-05-30 10:15:42.000000000 +0100
+++ b/makefile	2013-03-27 18:45:34.814670534 +0000
@@ -170,7 +170,7 @@ include_path     = $(prefix)/$(includedi
 # Usage check
 ifdef CXXFLAGS
 ifdef BUILD_TYPE
-	$(error CXXFLAGS and BUILD_TYPE are mutually exclusive)
+$(error CXXFLAGS and BUILD_TYPE are mutually exclusive)
 endif
 endif
 
