Fillers to make autoreconf (automake) happy.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura thrift-0.9.1.orig/AUTHORS thrift-0.9.1/AUTHORS
--- thrift-0.9.1.orig/AUTHORS	1969-12-31 21:00:00.000000000 -0300
+++ thrift-0.9.1/AUTHORS	2013-10-07 20:45:15.745315148 -0300
@@ -0,0 +1,2 @@
+
+
diff -Nura thrift-0.9.1.orig/ChangeLog thrift-0.9.1/ChangeLog
--- thrift-0.9.1.orig/ChangeLog	1969-12-31 21:00:00.000000000 -0300
+++ thrift-0.9.1/ChangeLog	2013-10-07 20:45:23.078554836 -0300
@@ -0,0 +1,2 @@
+
+
diff -Nura thrift-0.9.1.orig/NEWS thrift-0.9.1/NEWS
--- thrift-0.9.1.orig/NEWS	1969-12-31 21:00:00.000000000 -0300
+++ thrift-0.9.1/NEWS	2013-10-07 20:45:20.351465699 -0300
@@ -0,0 +1,2 @@
+
+
