From f31bb2561776ad2ce4b77339da6fbfc9e3b2ffbb Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Sun, 26 Oct 2014 14:18:29 +0100
Subject: [PATCH] Do not pass -Werror when --enable-debug is used

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
---
 Make_global.am | 2 +-
 configure.ac   | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/Make_global.am b/Make_global.am
index 7953451..25c4d64 100644
--- a/Make_global.am
+++ b/Make_global.am
@@ -77,7 +77,7 @@ AM_CPPFLAGS = $(kinclude_CFLAGS) $(all_includes) -I$(top_srcdir)/include \
 AM_CFLAGS = -std=gnu99
 
 if ENABLE_DEBUG
-AM_CFLAGS += -g -g3 -ggdb -gdwarf-2 -DIPSET_DEBUG -Wall -Werror
+AM_CFLAGS += -g -g3 -ggdb -gdwarf-2 -DIPSET_DEBUG -Wall
 else
 AM_CFLAGS += -O2 -DNDEBUG
 endif
diff --git a/configure.ac b/configure.ac
index 8fdac15..ee3b147 100644
--- a/configure.ac
+++ b/configure.ac
@@ -345,7 +345,6 @@ AX_CFLAGS_GCC_OPTION(-Waggregate-return)
 AX_CFLAGS_GCC_OPTION(-Wbad-function-cast)
 AX_CFLAGS_GCC_OPTION(-Wcast-align)
 AX_CFLAGS_GCC_OPTION(-Wcast-qual)
-AX_CFLAGS_GCC_OPTION(-Werror)
 AX_CFLAGS_GCC_OPTION(-Wextra)
 AX_CFLAGS_GCC_OPTION(-Wfloat-equal)
 AX_CFLAGS_GCC_OPTION(-Wformat=2)
-- 
2.0.0

