diff -urpN busybox-1.17.2/networking/libiproute/iplink.c busybox-1.17.2-compat/networking/libiproute/iplink.c
--- busybox-1.17.2/networking/libiproute/iplink.c	2010-08-23 02:44:35.000000000 +0200
+++ busybox-1.17.2-compat/networking/libiproute/iplink.c	2010-08-30 23:31:12.000000000 +0200
@@ -15,6 +15,11 @@
 #include "rt_names.h"
 #include "utils.h"
 
+#ifndef IFLA_LINKINFO
+# define IFLA_LINKINFO 18
+# define IFLA_INFO_KIND 1
+#endif
+
 /* taken from linux/sockios.h */
 #define SIOCSIFNAME	0x8923		/* set interface name */
 
