From d076aa6182dc6df6bb311e60bbddb03573b9483b Mon Sep 17 00:00:00 2001
From: Gregory Hermant <gregory.hermant@calao-systems.com>
Date: Fri, 3 Aug 2012 11:25:49 +0200
Subject: [PATCH] Enable pull-up on Rx serial ports for the CALAO MB-QIL-A9260


Signed-off-by: Gregory Hermant <gregory.hermant@calao-systems.com>
---
 arch/arm/boards/qil-a9260/init.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boards/qil-a9260/init.c b/arch/arm/boards/qil-a9260/init.c
index 305d733..b43cace 100644
--- a/arch/arm/boards/qil-a9260/init.c
+++ b/arch/arm/boards/qil-a9260/init.c
@@ -196,11 +196,17 @@ device_initcall(qil_a9260_devices_init);
 static int qil_a9260_console_init(void)
 {
 	at91_register_uart(0, 0);
+	at91_set_A_periph(AT91_PIN_PB14, 1);    /* Enable pull-up on DRXD */
+
 	at91_register_uart(1, ATMEL_UART_CTS | ATMEL_UART_RTS
 			   | ATMEL_UART_DTR | ATMEL_UART_DSR | ATMEL_UART_DCD
 			   | ATMEL_UART_RI);
+
 	at91_register_uart(2, ATMEL_UART_CTS | ATMEL_UART_RTS);
+	at91_set_A_periph(AT91_PIN_PB7, 1);	/* Enable pull-up on RXD1 */
+
 	at91_register_uart(3, ATMEL_UART_CTS | ATMEL_UART_RTS);
+	at91_set_A_periph(AT91_PIN_PB9, 1);	/* Enable pull-up on RXD2 */
 
 	return 0;
 }
-- 
1.5.6.3

