diff -Nura bind-9.5.1-P1/lib/isc/unix/ifiter_ioctl.c bind-9.5.1-P1.nosusv3/lib/isc/unix/ifiter_ioctl.c
--- bind-9.5.1-P1/lib/isc/unix/ifiter_ioctl.c	2007-08-30 03:02:28.000000000 -0300
+++ bind-9.5.1-P1.nosusv3/lib/isc/unix/ifiter_ioctl.c	2009-03-04 16:31:20.000000000 -0200
@@ -17,6 +17,8 @@
 
 /* $Id: ifiter_ioctl.c,v 1.58 2007/08/30 06:02:28 marka Exp $ */
 
+#include <string.h>
+
 /*! \file
  * \brief
  * Obtain the list of network interfaces using the SIOCGLIFCONF ioctl.
@@ -479,8 +481,8 @@
 	for (i = 0; i < 16; i++) {
 		unsigned char byte;
 		static const char hex[] = "0123456789abcdef";
-		byte = ((index(hex, address[i * 2]) - hex) << 4) |
-		       (index(hex, address[i * 2 + 1]) - hex);
+		byte = ((strchr(hex, address[i * 2]) - hex) << 4) |
+			(strchr(hex, address[i * 2 + 1]) - hex);
 		addr6.s6_addr[i] = byte;
 	}
 	iter->current.af = AF_INET6;
