From 395f063b6af8885cc9d688eeeb44acf9d75c6045 Mon Sep 17 00:00:00 2001
From: Peter Seiderer <ps.report@gmx.net>
Date: Thu, 3 Oct 2019 16:31:22 +0200
Subject: [PATCH] double-conversion: enable for microblaze

Signed-off-by: Peter Seiderer <ps.report@gmx.net>
---
 source/i18n/double-conversion-utils.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/source/i18n/double-conversion-utils.h b/source/i18n/double-conversion-utils.h
index 0cf9939..a00037e 100644
--- a/source/i18n/double-conversion-utils.h
+++ b/source/i18n/double-conversion-utils.h
@@ -104,6 +104,7 @@ int main(int argc, char** argv) {
     defined(__AARCH64EL__) || defined(__aarch64__) || defined(__AARCH64EB__) || \
     defined(__riscv) || \
     defined(__or1k__) || defined(__arc__) || \
+    defined(__microblaze__) || \
     defined(__EMSCRIPTEN__)
 #define DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS 1
 #elif defined(__mc68000__) || \
--
2.23.0

