--- libsndfile-1.0.17/Makefile.in.orig	2006-12-06 17:26:46.000000000 +0100
+++ libsndfile-1.0.17/Makefile.in	2006-12-06 17:31:11.000000000 +0100
@@ -209,7 +209,7 @@
 target_cpu = @target_cpu@
 target_os = @target_os@
 target_vendor = @target_vendor@
-SUBDIRS = man doc Win32 Octave src examples regtest tests
+SUBDIRS = src
 DIST_SUBDIRS = $(SUBDIRS)
 EXTRA_DIST = reconfigure.mk acinclude.m4 libsndfile.spec.in \
 			sndfile.pc.in Mingw-make-dist.sh
