Index: xserver-xorg-input-evdev/src/inotify-syscalls.h
===================================================================
--- xserver-xorg-input-evdev.orig/src/inotify-syscalls.h	2006-11-01 19:37:12.000000000 +1100
+++ xserver-xorg-input-evdev/src/inotify-syscalls.h	2006-11-01 19:59:51.000000000 +1100
@@ -43,6 +43,24 @@
 # define __NR_inotify_init	269
 # define __NR_inotify_add_watch	270
 # define __NR_inotify_rm_watch	271
+#elif defined (__mips__)
+# if _MIPS_SIM == _MIPS_SIM_ABI32
+#  define __NR_inotify_init		4284
+#  define __NR_inotify_add_watch	4285
+#  define __NR_inotify_rm_watch		4286
+# elif _MIPS_SIM == _MIPS_SIM_NABI32
+#  define __NR_inotify_init		6247
+#  define __NR_inotify_add_watch	6248
+#  define __NR_inotify_rm_watch		6249
+# elif _MIPS_SIM == _MIPS_SIM_ABI64
+#  define __NR_inotify_init		5243
+#  define __NR_inotify_add_watch	5244
+#  define __NR_inotify_rm_watch		5245
+# endif
+#elif defined (__mc68000__)
+# define __NR_inotify_init	284
+# define __NR_inotify_add_watch	285
+# define __NR_inotify_rm_watch	286
 #else
 # error "Unsupported architecture!"
 #endif
