--- a/configure.ac	2008-06-23 11:17:07.000000000 -0400
+++ b/configure.ac	2008-06-23 11:16:31.000000000 -0400
@@ -75,12 +75,20 @@ CPPFLAGS="$SAVE_CPPFLAGS"
 AC_HEADER_STDC
 
 if test "$DRI" != no; then
-        AC_CHECK_FILE([${sdkdir}/dri.h],
-                      [have_dri_h="yes"], [have_dri_h="no"])
-        AC_CHECK_FILE([${sdkdir}/sarea.h],
-                      [have_sarea_h="yes"], [have_sarea_h="no"])
-        AC_CHECK_FILE([${sdkdir}/dristruct.h],
-                      [have_dristruct_h="yes"], [have_dristruct_h="no"])
+  	if test "$cross_compiling" = "no" ; then
+ 		AC_CHECK_FILE([${sdkdir}/dri.h],
+                        [have_dri_h="yes"], [have_dri_h="no"])
+ 		AC_CHECK_FILE([${sdkdir}/sarea.h],
+                        [have_sarea_h="yes"], [have_sarea_h="no"])
+ 		AC_CHECK_FILE([${sdkdir}/dristruct.h],
+                        [have_dristruct_h="yes"], [have_dristruct_h="no"])
+ 		AC_CHECK_FILE([${sdkdir}/damage.h],
+                        [have_damage_h="yes"], [have_damage_h="no"])
+  	else
+  		have_dri_h="yes"
+  		have_sarea_h="yes"
+  		have_dristruct_h="yes"
+  	fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
