--- xf86-video-glint-1.1.1/configure.ac.orig	2007-06-10 20:58:34.000000000 +0200
+++ xf86-video-glint-1.1.1/configure.ac	2007-06-10 20:58:58.000000000 +0200
@@ -69,12 +69,18 @@
 AC_HEADER_STDC
 
 if test "$DRI" != no; then
+	if test "$cross_compiling" = "no" ; then
         AC_CHECK_FILE([${sdkdir}/dri.h],
                       [have_dri_h="yes"], [have_dri_h="no"])
         AC_CHECK_FILE([${sdkdir}/sarea.h],
                       [have_sarea_h="yes"], [have_sarea_h="no"])
         AC_CHECK_FILE([${sdkdir}/dristruct.h],
                       [have_dristruct_h="yes"], [have_dristruct_h="no"])
+	else
+		have_dri_h="yes"
+		have_sarea_h="yes"
+		have_dristruct_h="yes"
+	fi
 fi
 
 AC_MSG_CHECKING([whether to include DRI support])
