From c8aae8c96494132759319b0eab6cd2a51d1d4889 Mon Sep 17 00:00:00 2001
From: Frank Hunleth <fhunleth@troodon-software.com>
Date: Tue, 6 Mar 2012 07:20:26 -0800
Subject: [PATCH] am335x_evm.h: Specify ext2 and ro for booting rootfs

Signed-off-by: Frank Hunleth <fhunleth@troodon-software.com>
---
 include/configs/am335x_evm.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/include/configs/am335x_evm.h b/include/configs/am335x_evm.h
index e0ac494..b45c306 100755
--- a/include/configs/am335x_evm.h
+++ b/include/configs/am335x_evm.h
@@ -53,11 +53,11 @@
 	"script_addr=0x81900000\0" \
 	"console=ttyO0,115200n8\0" \
 	"mmc_dev=0\0" \
-	"mmc_root=/dev/mmcblk0p2 rw\0" \
+	"mmc_root=/dev/mmcblk0p2 ro\0" \
 	"nand_root=ubi0:rootfs rw ubi.mtd=7,2048\0" \
 	"spi_root=/dev/mtdblock4 rw\0" \
 	"nor_root=/dev/mtdblock3 rw\0" \
-	"mmc_root_fs_type=ext3 rootwait\0" \
+	"mmc_root_fs_type=ext2 rootwait\0" \
 	"nand_root_fs_type=ubifs rootwait=1\0" \
 	"spi_root_fs_type=jffs2\0" \
 	"nor_root_fs_type=jffs2\0" \
-- 
1.7.4.1

