--- a/lib/Xm/Makefile.in	2007-08-09 03:27:55.000000000 -0400
+++ b/lib/Xm/Makefile.in	2008-02-27 10:45:45.000000000 -0500
@@ -1130,7 +1130,7 @@ $(STRING_FILES): Makefile $(XMSTRING_LIS
 	test -f XmStrDefs23.ht || $(LN_S) $(srcdir)/XmStrDefs23.ht XmStrDefs23.ht
 	test -f XmStrDefsI.ht || $(LN_S) $(srcdir)/XmStrDefsI.ht XmStrDefsI.ht
 	test -f XmStrDefs.ct || $(LN_S) $(srcdir)/XmStrDefs.ct XmStrDefs.ct
-	$(top_builddir)/config/util/makestrs -f $(XMSTRING_LIST) >XmStrDefs.c
+	$(top_builddir)/config/util/makestrs-host -f $(XMSTRING_LIST) >XmStrDefs.c
 
 @MessageCatalog_TRUE@$(CAT_FILES): Xm.msg
 @MessageCatalog_TRUE@	$(mkinstalldirs) $(top_builddir)/localized/C/msg

--- a/lib/Xm/Makefile.am	2005-12-17 10:04:02.000000000 -0500
+++ b/lib/Xm/Makefile.am	2008-02-27 10:40:01.000000000 -0500
@@ -284,7 +284,7 @@ $(STRING_FILES): Makefile $(XMSTRING_LIS
 	test -f XmStrDefs23.ht || $(LN_S) $(srcdir)/XmStrDefs23.ht XmStrDefs23.ht
 	test -f XmStrDefsI.ht || $(LN_S) $(srcdir)/XmStrDefsI.ht XmStrDefsI.ht
 	test -f XmStrDefs.ct || $(LN_S) $(srcdir)/XmStrDefs.ct XmStrDefs.ct
-	$(top_builddir)/config/util/makestrs -f $(XMSTRING_LIST) >XmStrDefs.c
+	$(top_builddir)/config/util/makestrs-host -f $(XMSTRING_LIST) >XmStrDefs.c
 
 if MessageCatalog

--- a/tools/wml/Makefile.in	2008-02-27 11:24:14.000000000 -0500
+++ b/tools/wml/Makefile.in	2008-02-27 11:32:33.000000000 -0500
@@ -617,9 +617,9 @@ UilLexPars.c: Uil.c
 
 $(WMLTARGETS): wmluiltok wml UilLexPars.c $(TABLE)
 	$(RM) $(WMLTARGETS) wml.report wml-uil.mm
-	./wmluiltok <$(srcdir)/Uil.y >tokens.dat
+	./wmluiltok-host <$(srcdir)/Uil.y >tokens.dat
 	$(RM) wml-uil.mm
-	./wml $(srcdir)/$(TABLE)
+	./wml-host $(srcdir)/$(TABLE)
 	$(RM) tokens.dat
 
 all: motif.wmd

--- a/tools/wml/Makefile.am	2002-10-06 09:08:48.000000000 -0400
+++ b/tools/wml/Makefile.am	2008-02-27 11:33:04.000000000 -0500
@@ -50,9 +50,9 @@ UilLexPars.c: Uil.c
 
 $(WMLTARGETS): wmluiltok wml UilLexPars.c $(TABLE)
 	$(RM) $(WMLTARGETS) wml.report wml-uil.mm
-	./wmluiltok <$(srcdir)/Uil.y >tokens.dat
+	./wmluiltok-host <$(srcdir)/Uil.y >tokens.dat
 	$(RM) wml-uil.mm
-	./wml $(srcdir)/$(TABLE)
+	./wml-host $(srcdir)/$(TABLE)
 	$(RM) tokens.dat
 
 all: motif.wmd

--- a/demos/lib/Exm/wml/Makefile.am	2002-10-06 09:08:47.000000000 -0400
+++ b/demos/lib/Exm/wml/Makefile.am	2008-02-27 13:29:08.000000000 -0500
@@ -65,11 +65,11 @@ $(TARGETS):	$(TABLE) $(WMLDIR)/motif.wml
 # 	$(YACC) -d $(YACCFLAGS) $(UILY)
 # 	$(MV) y.tab.c UilLexPars.c
 # 	$(MV) y.tab.h UilLexPars.h
-	$(top_builddir)/tools/wml/wmluiltok < $(UILY) > tokens.dat
+	$(top_builddir)/tools/wml/wmluiltok-host < $(UILY) > tokens.dat
 # 	$(RM) wml-uil.mm
 	$(RM) Exm.h
 	$(LN_S) $(srcdir)/$(TABLE) ./Exm.h
-	$(CPP) -I$(WMLDIR) Exm.h | $(top_builddir)/tools/wml/wml
+	$(CPP) -I$(WMLDIR) Exm.h | $(top_builddir)/tools/wml/wml-host
 	$(RM) Exm.h
 # 	$(RM) tokens.dat
 # 

--- a/tools/wml/wmluiltok.c	2007-08-09 03:47:17.000000000 -0400
+++ b/tools/wml/wmluiltok.c	2008-02-27 12:13:02.000000000 -0500
@@ -1689,13 +1689,13 @@ void *ptr;
 	free( ptr );
 	}
 
-#if YY_MAIN
+//#if YY_MAIN
 int main()
 	{
 	yylex();
 	return 0;
 	}
-#endif
+//#endif
 #line 183 "wmluiltok.l"


--- a/ltmain.sh	2006-07-13 03:12:13.000000000 -0400
+++ b/ltmain.sh	2008-02-27 14:50:40.000000000 -0500
@@ -5427,13 +5427,13 @@ else
 	# Backslashes separate directories on plain windows
 	*-*-mingw | *-*-os2*)
 	  $echo >> $output "\
-      exec \"\$progdir\\\\\$program\" \${1+\"\$@\"}
+      exec \"\$progdir\\\\\$program-host\" \${1+\"\$@\"}
 "
 	  ;;
 
 	*)
 	  $echo >> $output "\
-      exec \"\$progdir/\$program\" \${1+\"\$@\"}
+      exec \"\$progdir/\$program-host\" \${1+\"\$@\"}
 "
 	  ;;
 	esac
