Host version of hamcorebuilder is needed during cross-compile

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824.org/src/bin/hamcore/Makefile.am SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824/src/bin/hamcore/Makefile.am
--- SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824.org/src/bin/hamcore/Makefile.am	2014-09-27 12:56:53.108725793 +0200
+++ SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824/src/bin/hamcore/Makefile.am	2014-09-27 12:57:50.280009025 +0200
@@ -26,8 +26,8 @@
 
 noinst_PROGRAMS = hamcore.se2
 
-hamcore.se2$(EXEEXT): $(HAMCOREBUILDER)
-	$(HAMCOREBUILDER) $(top_srcdir)/src/bin/hamcore $@
+hamcore.se2$(EXEEXT):
+	hamcorebuilder $(top_srcdir)/src/bin/hamcore $@
 
 install-exec-local: hamcore.se2$(EXEEXT)
 	$(INSTALL_DATA) -D hamcore.se2$(EXEEXT) $(DESTDIR)$(localstatedir)/lib/softether/hamcore.se2
diff -uNr SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824.org/src/Makefile.am SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824/src/Makefile.am
--- SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824.org/src/Makefile.am	2014-09-27 12:56:53.136726809 +0200
+++ SoftEtherVPN-16b713b98da8dba29f0f845d5a8c36d6f7c34824/src/Makefile.am	2014-09-27 12:58:12.959723109 +0200
@@ -20,7 +20,7 @@
 SUBDIRS = Mayaqua Cedar
 
 # This is a nodist helper.
-SUBDIRS += hamcorebuilder
+# SUBDIRS += hamcorebuilder
 
 # These are shared components.
 SUBDIRS += libsoftether bin/hamcore
