Allow cross compilation. Adapted from crosstool-ng.

Signed-off-by: Baruch Siach <baruch at tkos.co.il>

Index: b/GNUmakefile
===================================================================
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -93,10 +93,6 @@
 # also define 'WIN32'
 
 # some defaults:
-CC=gcc
-CXX=g++
-AR=ar
-RANLIB=ranlib
 INSTALL=install
 RM=rm
 RMFORCE=rm -f
@@ -471,7 +467,7 @@
 
 createconf$(EXEPOSTFIX): createconf.o
 	- $(RMFORCE) createconf$(EXEPOSTFIX)
-	$(CC) $(CFLAGS) $(DUMA_OPTIONS) createconf.o -o createconf$(EXEPOSTFIX)
+	$(CC_FOR_BUILD) $(HOST_CFLAGS) $(DUMA_OPTIONS) createconf.o -o createconf$(EXEPOSTFIX)
 
 tstheap$(EXEPOSTFIX): libduma.a tstheap.o
 	- $(RMFORCE) tstheap$(EXEPOSTFIX)
@@ -532,7 +528,7 @@
 # define rules how to build objects for createconf
 #
 createconf.o:
-	$(CC) $(CFLAGS) $(DUMA_OPTIONS) -c createconf.c -o $@
+	$(CC_FOR_BUILD) $(HOST_CFLAGS) $(DUMA_OPTIONS) -c createconf.c -o $@
 
 
 #
