--- a/libsylph/Makefile.in	2007-12-20 23:48:26.000000000 -0500
+++ b/libsylph/Makefile.in	2008-02-14 16:34:17.000000000 -0500
@@ -143,7 +143,7 @@ manualdir = @manualdir@
 AM_CPPFLAGS =  	-DG_LOG_DOMAIN=\"LibSylph\" 	-DSYSCONFDIR=\""$(sysconfdir)"\"
 
 
-INCLUDES = $(GLIB_CFLAGS) -I$(top_srcdir) -I$(includedir)
+INCLUDES = $(GLIB_CFLAGS) -I$(top_srcdir) 
 
 #lib_LTLIBRARIES = libsylph.la
 noinst_LTLIBRARIES = libsylph.la

--- a/src/Makefile.in	2007-12-20 23:48:26.000000000 -0500
+++ b/src/Makefile.in	2008-02-14 16:47:46.000000000 -0500
@@ -153,7 +153,7 @@ BUILT_SOURCES =  	quote_fmt_lex.c 	quote
 EXTRA_DIST =  	quote_fmt_parse.h 	sylpheed-marshal.list 	version.h.in 	sylpheed.rc.in
 
 
-INCLUDES =  	-DG_LOG_DOMAIN=\"Sylpheed\" 	$(GTK_CFLAGS) 	$(GDK_PIXBUF_CFLAGS) 	$(GPGME_CFLAGS) 	-I$(includedir) 	-I$(top_srcdir)/libsylph
+INCLUDES =  	-DG_LOG_DOMAIN=\"Sylpheed\" 	$(GTK_CFLAGS) 	$(GDK_PIXBUF_CFLAGS) 	$(GPGME_CFLAGS) 	-I$(top_srcdir)/libsylph
 
 @HAVE_WINDRES_TRUE@SYLPHEED_RES = sylpheed.res
 @HAVE_WINDRES_FALSE@SYLPHEED_RES = 
