Drop bcopy since it's deprecated.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura libmbus-0.7.0/mbus/mbus-tcp.c libmbus-0.7.0-bcopy/mbus/mbus-tcp.c
--- libmbus-0.7.0/mbus/mbus-tcp.c	2011-12-08 13:21:19.000000000 -0300
+++ libmbus-0.7.0-bcopy/mbus/mbus-tcp.c	2012-05-03 09:08:36.987106612 -0300
@@ -78,7 +78,7 @@
         return NULL;
     }
 
-    bcopy((void *)(host_addr->h_addr), (void *)(&s.sin_addr), host_addr->h_length);
+    memmove((void *)(&s.sin_addr), (void *)(host_addr->h_addr), host_addr->h_length);
 
     if (connect(handle->sock, (struct sockaddr *)&s, sizeof(s)) < 0)
     {
