diff -ur module-init-tools-3.2.2/modprobe.c module-init-tools-3.2.2-patched/modprobe.c
--- module-init-tools-3.2.2/modprobe.c	2005-12-01 17:42:09.000000000 -0600
+++ module-init-tools-3.2.2-patched/modprobe.c	2006-12-04 19:50:50.353237649 -0600
@@ -270,7 +270,7 @@
 	char *modname;
 
 	/* Ignore lines without : or which start with a # */
-	ptr = index(line, ':');
+	ptr = strchr(line, ':');
 	if (ptr == NULL || line[strspn(line, "\t ")] == '#')
 		return 0;
 
