---
 Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: shared-mime-info-0.60/Makefile.am
===================================================================
--- shared-mime-info-0.60.orig/Makefile.am
+++ shared-mime-info-0.60/Makefile.am
@@ -72,7 +72,7 @@
 
 check: freedesktop.org.xml
 	if test -n $(XMLLINT) ; then \
-		xmllint --noout --valid $(top_srcdir)/freedesktop.org.xml;	\
+		$(XMLLINT) --noout --valid $(top_srcdir)/freedesktop.org.xml;	\
 	fi
 	if test -d CVS/ && test -x ../xdgmime/src/test-mime-data && test -x $(top_builddir)/test-tree-magic ; then \
 		mkdir -p $(top_builddir)/temp-mime-dir/mime/packages ; \
