From 0b701c37d8a409904ea80fb6b74f096d1f28cc84 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Tue, 1 May 2018 22:27:21 +0200
Subject: [PATCH] Disable building of statically linked vlc binary

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 Makefile.am     | 2 +-
 bin/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index febdc63fe4..dca6354ec9 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -933,7 +933,7 @@ libvlc: libvlccore
 	cd lib && $(MAKE) $(AM_MAKEFLAGS) libvlc.la
 
 core: libvlc vlc$(EXEEXT)
-	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT) vlc-static$(EXEEXT)
+	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT)
 
 doc:
 	cd doc && $(MAKE) $(AM_MAKEFLAGS) doc
diff --git a/bin/Makefile.am b/bin/Makefile.am
index 869ac561dc..5e48b70d0a 100644
--- a/bin/Makefile.am
+++ b/bin/Makefile.am
@@ -1,7 +1,7 @@
 # Building vlc
 #
 bin_PROGRAMS = vlc
-noinst_PROGRAMS = vlc-static
+noinst_PROGRAMS =
 noinst_DATA =
 vlclib_PROGRAMS = vlc-cache-gen
 EXTRA_PROGRAMS = vlc-wrapper
-- 
2.14.3

