From 7c2b3343a30ee53a1cded2ca5e712d8cd7d50f5f Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@t-online.de>
Date: Tue, 1 May 2018 22:27:21 +0200
Subject: [PATCH] Disable building of statically linked vlc binary

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>
---
 Makefile.am     | 2 +-
 bin/Makefile.am | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index 3ac82623ad..637504a3f5 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -108,7 +108,7 @@ libvlc: libvlccore
 	cd lib && $(MAKE) $(AM_MAKEFLAGS) libvlc.la
 
 core: libvlc vlc$(EXEEXT)
-	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT) vlc-static$(EXEEXT)
+	cd bin && $(MAKE) $(AM_MAKEFLAGS) vlc$(EXEEXT)
 
 doc:
 	cd doc && $(MAKE) $(AM_MAKEFLAGS) doc
diff --git a/bin/Makefile.am b/bin/Makefile.am
index 2877e8b0ea..bef4160e8b 100644
--- a/bin/Makefile.am
+++ b/bin/Makefile.am
@@ -6,7 +6,7 @@ bin_PROGRAMS = vlc-osx
 noinst_PROGRAMS = vlc-osx-static
 else
 bin_PROGRAMS = vlc
-noinst_PROGRAMS = vlc-static
+noinst_PROGRAMS =
 endif
 endif
 EXTRA_DIST = vlc_win32_rc.rc.in
-- 
2.14.4

