libX11 is an optional dependency for the qt4 module

Signed-off-by: Bernd Kuhls <bernd.kuhls@t-online.de>

diff -uNr vlc-2.2.1.org/modules/gui/qt4/Makefile.am vlc-2.2.1/modules/gui/qt4/Makefile.am
--- vlc-2.2.1.org/modules/gui/qt4/Makefile.am	2014-11-29 12:34:15.000000000 +0100
+++ vlc-2.2.1/modules/gui/qt4/Makefile.am	2015-04-27 18:53:36.968611490 +0200
@@ -22,7 +22,7 @@
 if HAVE_WIN32
 libqt4_plugin_la_LIBADD += -lole32 -lcomctl32 -luuid
 else
-libqt4_plugin_la_LIBADD += $(X_LIBS) $(X_PRE_LIB) -lX11
+libqt4_plugin_la_LIBADD += $(X_LIBS) $(X_PRE_LIB)
 endif
 endif
 if HAVE_DARWIN
