diff -urN fakeroot-1.9.3.orig/communicate.h fakeroot-1.9.3/communicate.h
--- fakeroot-1.9.3.orig/communicate.h	2008-03-11 09:42:05.000000000 -0400
+++ fakeroot-1.9.3/communicate.h	2008-03-11 10:58:10.000000000 -0400
@@ -35,7 +35,9 @@
 #endif
 
 /* Then decide whether we do or do not use the stat64 support */
-#if (defined(sun) && !defined(__SunOS_5_5_1) && !defined(_LP64)) || __GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1)
+#if (defined(sun) && !defined(__SunOS_5_5_1) && !defined(_LP64)) \
+    || (!defined __UCLIBC__ && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 1))) \
+    || (defined __UCLIBC__ && defined UCLIBC_HAS_LFS)
 # define STAT64_SUPPORT
 #else
 # warning Not using stat64 support