Fix "underquoted definition of AM_PATH_LOG4C" warning.

Signed-off-by: Danomi Manchego <danomimanchego123@gmail.com> 

diff -urN liblog4c-localtime-1.0/log4c.m4 liblog4c-localtime-1.0.orig/log4c.m4
--- liblog4c-localtime-1.0/log4c.m4	2012-10-27 16:11:46.219894716 -0400
+++ liblog4c-localtime-1.0.orig/log4c.m4	2012-10-27 16:10:31.393336864 -0400
@@ -4,7 +4,7 @@
 dnl AM_PATH_LOG4C([MINIMUM-VERSION, [ACTION-IF-FOUND [, ACTION-IF-NOT-FOUND]]])
 dnl Test for LOG4C, and define LOG4C_CFLAGS and LOG4C_LIBS
 dnl
-AC_DEFUN(AM_PATH_LOG4C,
+AC_DEFUN([AM_PATH_LOG4C],
 [dnl 
 dnl Get the cflags and libraries from the log4c-config script
 dnl
