diff --git a/acinclude.m4 b/acinclude.m4
index a7bc0fa..1a5bffc 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -76,13 +76,14 @@ case "${host}" in
     i386-emx-os2 | i[3456]86-pc-os2*emx | i386-pc-msdosdjgpp)
         ac_cv_sys_symbol_underscore=yes
         ;;
-    *)
-      if test "$cross_compiling" = yes; then
-        if test "x$ac_cv_sys_symbol_underscore" = x ; then
-           ac_cv_sys_symbol_underscore=yes
-        fi
-      else
-         tmp_do_check="yes"
+    *) if test -z "$ac_cv_sys_symbol_underscore"; then
+         if test "$cross_compiling" = yes; then
+           if test "x$ac_cv_sys_symbol_underscore" = x ; then
+              ac_cv_sys_symbol_underscore=yes
+           fi
+         else
+            tmp_do_check="yes"
+         fi
       fi
        ;;
 esac
 
