diff -u libgcrypt-1.2.4.oorig/acinclude.m4 libgcrypt-1.2.4/acinclude.m4
--- libgcrypt-1.2.4.oorig/acinclude.m4	2005-07-29 15:45:48.000000000 +0200
+++ libgcrypt-1.2.4/acinclude.m4	2007-09-19 13:33:19.000000000 +0200
@@ -99,12 +99,14 @@
     i386-emx-os2 | i[3456]86-pc-os2*emx | i386-pc-msdosdjgpp)
         ac_cv_sys_symbol_underscore=yes
         ;;
-    *)
-      if test "$cross_compiling" = yes; then
-         ac_cv_sys_symbol_underscore=yes
-      else
-         tmp_do_check="yes"
-      fi
+    *) if test -z "$ac_cv_sys_symbol_underscore"; then
+         if test "$cross_compiling" = yes && 
+         then
+           ac_cv_sys_symbol_underscore=yes
+         else
+           tmp_do_check="yes"
+         fi
+        fi
        ;;
 esac
 if test "$tmp_do_check" = "yes"; then
