diff --git a/configure b/configure
index 70a6d50..8f55d95 100755
--- a/configure
+++ b/configure
@@ -22306,14 +22306,15 @@ case "${host}" in
     i386-emx-os2 | i345686-pc-os2*emx | i386-pc-msdosdjgpp)
         ac_cv_sys_symbol_underscore=yes
         ;;
-    *)
-      if test "$cross_compiling" = yes; then
-        if test "x$ac_cv_sys_symbol_underscore" = x ; then
-           ac_cv_sys_symbol_underscore=yes
+    *)if test -z "$ac_cv_sys_symbol_underscore";then
+        if test "$cross_compiling" = yes; then
+            if test "x$ac_cv_sys_symbol_underscore" = x ; then
+                ac_cv_sys_symbol_underscore=yes
+            fi
+        else
+            tmp_do_check="yes"
+        fi
         fi
-      else
-         tmp_do_check="yes"
-      fi
        ;;
 esac
 if test "$tmp_do_check" = "yes"; then
