--- binutils-2.16/ld/configure.tgt	2005-04-22 21:53:27.895421000 -0500
+++ binutils-2.16-patched/ld/configure.tgt	2005-04-22 21:49:26.757507746 -0500
@@ -444,10 +444,10 @@
 			targ_extra_emuls="elf32ltsmipn32 elf32btsmip elf32ltsmip elf64btsmip elf64ltsmip"
 			targ_extra_libpath="elf32btsmip elf64btsmip"
 			;;
-mips*el-*-linux-gnu*)	targ_emul=elf32ltsmip
+mips*el-*-linux-*)	targ_emul=elf32ltsmip
 			targ_extra_emuls="elf32btsmip elf32ltsmipn32 elf64ltsmip elf32btsmipn32 elf64btsmip"
 			;;
-mips*-*-linux-gnu*)	targ_emul=elf32btsmip
+mips*-*-linux-*)	targ_emul=elf32btsmip
 			targ_extra_emuls="elf32ltsmip elf32btsmipn32 elf64btsmip elf32ltsmipn32 elf64ltsmip"
 			;;
 mips*-*-lnews*)		targ_emul=mipslnews ;;
