Allow cache answer for strcasestr discovery.

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>

diff -Nura php-5.6.7.orig/ext/fileinfo/config.m4 php-5.6.7/ext/fileinfo/config.m4
--- php-5.6.7.orig/ext/fileinfo/config.m4	2015-04-08 22:19:45.798770792 -0300
+++ php-5.6.7/ext/fileinfo/config.m4	2015-04-08 22:26:33.110654338 -0300
@@ -14,6 +14,7 @@
     libmagic/readcdf.c libmagic/softmagic.c"
 
   AC_MSG_CHECKING([for strcasestr])
+  AC_CACHE_VAL(ac_cv_func_strcasestr,
   AC_TRY_RUN([
 #include <string.h>
 #include <strings.h>
@@ -46,7 +47,7 @@
     AC_MSG_RESULT(no)
     AC_MSG_NOTICE(using libmagic strcasestr implementation)
     libmagic_sources="$libmagic_sources libmagic/strcasestr.c"
-  ])
+  ]))
 
   PHP_NEW_EXTENSION(fileinfo, fileinfo.c $libmagic_sources, $ext_shared,,-I@ext_srcdir@/libmagic)
   PHP_ADD_BUILD_DIR($ext_builddir/libmagic)
