From b60398837ddf39d4b605600abc167d4366e454a6 Mon Sep 17 00:00:00 2001
From: Samuel Martin <s.martin49@gmail.com>
Date: Thu, 5 Jun 2014 00:06:08 +0200
Subject: [PATCH] interface/vmcs_host: add install rules for libvcilcs

Signed-off-by: Samuel Martin <s.martin49@gmail.com>
---
 interface/vmcs_host/CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/interface/vmcs_host/CMakeLists.txt b/interface/vmcs_host/CMakeLists.txt
index 64ba2f8..8510f81 100755
--- a/interface/vmcs_host/CMakeLists.txt
+++ b/interface/vmcs_host/CMakeLists.txt
@@ -33,4 +33,5 @@ target_link_libraries(vchostif vchiq_arm vcos vcfiled_check)
 
 add_subdirectory(linux/vcfiled)
 install(TARGETS vchostif DESTINATION lib)
+install(TARGETS vcilcs DESTINATION lib)
 
-- 
2.0.0

