diff -rdup fontconfig-2.4.2.orig/Makefile.in fontconfig-2.4.2/Makefile.in
--- fontconfig-2.4.2.orig/Makefile.in	2006-12-03 00:27:34.000000000 +0100
+++ fontconfig-2.4.2/Makefile.in	2007-01-16 18:08:30.000000000 +0100
@@ -122,6 +122,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
diff -rdup fontconfig-2.4.2.orig/doc/Makefile.in fontconfig-2.4.2/doc/Makefile.in
--- fontconfig-2.4.2.orig/doc/Makefile.in	2006-12-03 00:27:32.000000000 +0100
+++ fontconfig-2.4.2/doc/Makefile.in	2007-01-16 18:10:09.000000000 +0100
@@ -79,10 +79,10 @@ DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(t
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(AM_CFLAGS) $(BUILD_CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(BUILD_CFLAGS)
 CCLD = $(CC)
 SOURCES = $(edit_sgml_SOURCES)
 DIST_SOURCES = $(edit_sgml_SOURCES)
@@ -118,6 +118,7 @@ AWK = @AWK@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
@@ -254,7 +255,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(BUILD_CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 DOC_SRC = $(srcdir)
 DOC_MODULE = fontconfig
 DOC2HTML = docbook2html
diff -rdup fontconfig-2.4.2.orig/fc-arch/Makefile.in fontconfig-2.4.2/fc-arch/Makefile.in
--- fontconfig-2.4.2.orig/fc-arch/Makefile.in	2006-12-03 00:27:32.000000000 +0100
+++ fontconfig-2.4.2/fc-arch/Makefile.in	2007-01-16 18:24:19.000000000 +0100
@@ -80,10 +80,10 @@ DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(t
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(AM_CFLAGS) $(BUILD_CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(BUILD_CFLAGS)
 CCLD = $(CC)
 SOURCES = fc-arch.c
 DIST_SOURCES = fc-arch.c
@@ -105,6 +105,7 @@ AWK = @AWK@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
@@ -241,7 +242,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(BUILD_CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
 TMPL = fcarch.tmpl.h
 STMPL = ${top_srcdir}/fc-arch/fcarch.tmpl.h
diff -rdup fontconfig-2.4.2.orig/fc-case/Makefile.in fontconfig-2.4.2/fc-case/Makefile.in
--- fontconfig-2.4.2.orig/fc-case/Makefile.in	2006-12-03 00:27:32.000000000 +0100
+++ fontconfig-2.4.2/fc-case/Makefile.in	2007-01-16 18:23:16.000000000 +0100
@@ -80,10 +80,10 @@ DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(t
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(AM_CFLAGS) $(BUILD_CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(BUILD_CFLAGS)
 CCLD = $(CC)
 SOURCES = fc-case.c
 DIST_SOURCES = fc-case.c
@@ -105,6 +105,7 @@ AWK = @AWK@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
@@ -241,7 +242,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(BUILD_CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
 TMPL = fccase.tmpl.h
 STMPL = ${top_srcdir}/fc-case/${TMPL}
diff -rdup fontconfig-2.4.2.orig/fc-cat/Makefile.in fontconfig-2.4.2/fc-cat/Makefile.in
--- fontconfig-2.4.2.orig/fc-cat/Makefile.in	2006-12-03 00:27:32.000000000 +0100
+++ fontconfig-2.4.2/fc-cat/Makefile.in	2007-01-16 18:13:33.000000000 +0100
@@ -110,6 +110,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
diff -rdup fontconfig-2.4.2.orig/fc-glyphname/Makefile.in fontconfig-2.4.2/fc-glyphname/Makefile.in
--- fontconfig-2.4.2.orig/fc-glyphname/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/fc-glyphname/Makefile.in	2007-01-16 18:10:59.000000000 +0100
@@ -80,10 +80,10 @@ DEFAULT_INCLUDES = -I. -I$(srcdir) -I$(t
 depcomp = $(SHELL) $(top_srcdir)/depcomp
 am__depfiles_maybe = depfiles
 COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
-	$(CPPFLAGS) $(AM_CFLAGS) $(CFLAGS)
+	$(CPPFLAGS) $(AM_CFLAGS) $(BUILD_CFLAGS)
 LTCOMPILE = $(LIBTOOL) --tag=CC --mode=compile $(CC) $(DEFS) \
 	$(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) $(CPPFLAGS) \
-	$(AM_CFLAGS) $(CFLAGS)
+	$(AM_CFLAGS) $(BUILD_CFLAGS)
 CCLD = $(CC)
 SOURCES = fc-glyphname.c
 DIST_SOURCES = fc-glyphname.c
@@ -105,6 +105,7 @@ AWK = @AWK@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
@@ -241,7 +242,7 @@ sbindir = @sbindir@
 sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+LINK = $(CCLD) $(AM_CFLAGS) $(BUILD_CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
 INCLUDES = -I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
 TMPL = fcglyphname.tmpl.h
 STMPL = ${top_srcdir}/fc-glyphname/${TMPL}
diff -rdup fontconfig-2.4.2.orig/fc-list/Makefile.in fontconfig-2.4.2/fc-list/Makefile.in
--- fontconfig-2.4.2.orig/fc-list/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/fc-list/Makefile.in	2007-01-16 18:20:18.000000000 +0100
@@ -110,6 +110,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
diff -rdup fontconfig-2.4.2.orig/fontconfig/Makefile.in fontconfig-2.4.2/fontconfig/Makefile.in
--- fontconfig-2.4.2.orig/fontconfig/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/fontconfig/Makefile.in	2007-01-16 18:12:57.000000000 +0100
@@ -74,6 +74,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
diff -rdup fontconfig-2.4.2.orig/src/Makefile.in fontconfig-2.4.2/src/Makefile.in
--- fontconfig-2.4.2.orig/src/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/src/Makefile.in	2007-01-16 18:21:17.000000000 +0100
@@ -119,6 +119,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/conf.d/Makefile.in	2006-12-03 00:27:31.000000000 +0100
+++ fontconfig-2.4.2/conf.d/Makefile.in	2007-01-16 18:35:32.000000000 +0100
@@ -94,6 +94,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/fc-lang/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/fc-lang/Makefile.in	2007-01-16 18:38:07.000000000 +0100
@@ -105,6 +105,7 @@ AWK = @AWK@
 CC = @CC_FOR_BUILD@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/test/Makefile.in	2006-12-03 00:27:34.000000000 +0100
+++ fontconfig-2.4.2/test/Makefile.in	2007-01-16 18:38:20.000000000 +0100
@@ -61,6 +61,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/fc-cache/Makefile.in	2006-12-03 00:27:32.000000000 +0100
+++ fontconfig-2.4.2/fc-cache/Makefile.in	2007-01-16 18:38:26.000000000 +0100
@@ -110,6 +110,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/fc-match/Makefile.in	2006-12-03 00:27:33.000000000 +0100
+++ fontconfig-2.4.2/fc-match/Makefile.in	2007-01-16 18:40:17.000000000 +0100
@@ -110,6 +110,7 @@ AWK = @AWK@
 CC = @CC@
 CCDEPMODE = @CCDEPMODE@
 CC_FOR_BUILD = @CC_FOR_BUILD@
+BUILD_CFLAGS = @BUILD_CFLAGS@
 CFLAGS = @CFLAGS@
 CONFDIR = @CONFDIR@
 CPP = @CPP@
--- fontconfig-2.4.2.orig/configure	2006-12-03 00:27:35.000000000 +0100
+++ fontconfig-2.4.2/configure	2007-01-16 18:54:49.000000000 +0100
@@ -871,6 +871,7 @@ WARN_CFLAGS
 CROSS_COMPILING_TRUE
 CROSS_COMPILING_FALSE
 CC_FOR_BUILD
+BUILD_CFLAGS
 EXEEXT_FOR_BUILD
 ARCHITECTURE
 ICONV_LIBS
@@ -20231,7 +20232,9 @@ else
   CROSS_COMPILING_FALSE=
 fi
 
-
+if test "x$BUILD_CFLAGS" = "x"; then
+  BUILD_CFLAGS="-O2 "
+fi
 
 # Setup for compiling build tools (fc-glyphname, etc)
 { echo "$as_me:$LINENO: checking for a C compiler for build tools" >&5
@@ -24541,6 +24544,7 @@ WARN_CFLAGS!$WARN_CFLAGS$ac_delim
 CROSS_COMPILING_TRUE!$CROSS_COMPILING_TRUE$ac_delim
 CROSS_COMPILING_FALSE!$CROSS_COMPILING_FALSE$ac_delim
 CC_FOR_BUILD!$CC_FOR_BUILD$ac_delim
+BUILD_CFLAGS!$BUILD_CFLAGS$ac_delim
 EXEEXT_FOR_BUILD!$EXEEXT_FOR_BUILD$ac_delim
 ARCHITECTURE!$ARCHITECTURE$ac_delim
 ICONV_LIBS!$ICONV_LIBS$ac_delim
@@ -24575,7 +24579,7 @@ LIBOBJS!$LIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 54; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 55; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
