--- fontconfig-2.4.2/fc-lang/fc-lang.orig.c	2007-02-03 13:39:49.000000000 -0700
+++ fontconfig-2.4.2/fc-lang/fc-lang.c	2007-02-03 13:46:08.000000000 -0700
@@ -116,6 +116,20 @@
     return f;
 }
 
+char* trim(char *s)
+{
+	int len = strlen(s);
+
+	/* nuke trailing whitespace */
+	while (len > 0 && (isspace(s[len - 1]) || s[len - 1]=='\n'))
+		s[--len] = '\0';
+
+	/* skip leading whitespace */
+	s += strspn(s, " \n\r\t\v");
+
+	return s;
+}
+
 /*
  * build a single charset from a source file
  *
@@ -142,8 +156,7 @@
             if (!file)
                 fatal (line, lineno, 
                        "invalid syntax, expected: include filename");
-	    while (isspace(*file))
-		file++;
+	    file = trim(file);
 	    f = scanopen (file);
 	    if (!f)
 		fatal (file, 0, "can't open");
