--- fontconfig-2.4.2/fc-lang/Makefile.am.orig	2007-06-24 07:02:54.000000000 +0200
+++ fontconfig-2.4.2/fc-lang/Makefile.am	2007-06-24 07:05:14.000000000 +0200
@@ -23,10 +23,16 @@
 #
 
 CC = @CC_FOR_BUILD@
+LD = @CC_FOR_BUILD@
+AM_CPPFLAGS =
+AM_CFLAGS = @CFLAGS_FOR_BUILD@
+AM_LDFLAGS = @LDFLAGS_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(AM_CFLAGS)
+LINK = $(LD) $(AM_CFLAGS) $(AM_LDFLAGS) -o $@
 
-INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
+INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
 TMPL=fclang.tmpl.h
 STMPL=${top_srcdir}/fc-lang/fclang.tmpl.h
--- fontconfig-2.4.2/fc-case/Makefile.am.orig	2007-06-24 07:02:54.000000000 +0200
+++ fontconfig-2.4.2/fc-case/Makefile.am	2007-06-24 07:05:14.000000000 +0200
@@ -23,10 +23,16 @@
 #
 
 CC = @CC_FOR_BUILD@
+LD = @CC_FOR_BUILD@
+AM_CPPFLAGS =
+AM_CFLAGS = @CFLAGS_FOR_BUILD@
+AM_LDFLAGS = @LDFLAGS_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(AM_CFLAGS)
+LINK = $(LD) $(AM_CFLAGS) $(AM_LDFLAGS) -o $@
 
-INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
+INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
 TMPL=fccase.tmpl.h
 STMPL=${top_srcdir}/fc-case/${TMPL}
--- fontconfig-2.4.2/fc-glyphname/Makefile.am.orig	2007-06-24 07:02:54.000000000 +0200
+++ fontconfig-2.4.2/fc-glyphname/Makefile.am	2007-06-24 07:05:14.000000000 +0200
@@ -23,10 +23,16 @@
 #
 
 CC = @CC_FOR_BUILD@
+LD = @CC_FOR_BUILD@
+AM_CPPFLAGS =
+AM_CFLAGS = @CFLAGS_FOR_BUILD@
+AM_LDFLAGS = @LDFLAGS_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(AM_CFLAGS)
+LINK = $(LD) $(AM_CFLAGS) $(AM_LDFLAGS) -o $@
 
-INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
+INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
 TMPL=fcglyphname.tmpl.h
 STMPL=${top_srcdir}/fc-glyphname/${TMPL}
--- fontconfig-2.4.2/configure.in.orig	2007-06-24 07:02:54.000000000 +0200
+++ fontconfig-2.4.2/configure.in	2007-06-24 07:02:59.000000000 +0200
@@ -101,6 +101,8 @@
 AC_MSG_CHECKING([for a C compiler for build tools])
 if test $cross_compiling = yes; then
   AC_CHECK_PROGS(CC_FOR_BUILD, gcc cc)
+  AC_SUBST(CFLAGS_FOR_BUILD)
+  AC_SUBST(LDFLAGS_FOR_BUILD)
 else
   CC_FOR_BUILD=$CC
 fi
--- fontconfig-2.4.2/fc-arch/Makefile.am.orig	2007-06-24 07:02:54.000000000 +0200
+++ fontconfig-2.4.2/fc-arch/Makefile.am	2007-06-24 07:05:14.000000000 +0200
@@ -23,10 +23,16 @@
 #
 
 CC = @CC_FOR_BUILD@
+LD = @CC_FOR_BUILD@
+AM_CPPFLAGS =
+AM_CFLAGS = @CFLAGS_FOR_BUILD@
+AM_LDFLAGS = @LDFLAGS_FOR_BUILD@
 EXEEXT = @EXEEXT_FOR_BUILD@
-LINK = $(CCLD) $(AM_CFLAGS) $(CFLAGS) $(AM_LDFLAGS) $(LDFLAGS) -o $@
+COMPILE = $(CC) $(DEFS) $(DEFAULT_INCLUDES) $(INCLUDES) $(AM_CPPFLAGS) \
+	$(AM_CFLAGS)
+LINK = $(LD) $(AM_CFLAGS) $(AM_LDFLAGS) -o $@
 
-INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(FREETYPE_CFLAGS) $(WARN_CFLAGS)
+INCLUDES=-I${top_srcdir}/src -I${top_srcdir} $(WARN_CFLAGS)
 
 TMPL=fcarch.tmpl.h
 STMPL=${top_srcdir}/fc-arch/fcarch.tmpl.h
