--- a/alsactl/Makefile.in	2009-01-15 10:59:38.000000000 +0000
+++ b/alsactl/Makefile.in	2009-01-15 10:59:34.000000000 +0000
@@ -204,7 +204,7 @@ sharedstatedir = @sharedstatedir@
 sysconfdir = @sysconfdir@
 target_alias = @target_alias@
 SUBDIRS = init
-man_MANS = alsactl.1 alsactl_init.7
+man_MANS = alsactl.1
 EXTRA_DIST = alsactl.1 alsactl_init.xml
 alsactl_SOURCES = alsactl.c state.c utils.c init_parse.c
 noinst_HEADERS = alsactl.h list.h init_sysdeps.c init_utils_string.c init_utils_run.c init_sysfs.c
