---
 configure |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: ifplugd-0.28/configure
===================================================================
--- ifplugd-0.28.orig/configure
+++ ifplugd-0.28/configure
@@ -5430,7 +5430,7 @@
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   if test "$cross_compiling" = yes; then
-  ac_cv_func_malloc_0_nonnull=no
+  ac_cv_func_malloc_0_nonnull=yes
 else
   cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
