Makefile: use LDFLAGS when linking

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN pifmrds-c67306ea9b8d827f45e0d90279d367e97119bcb1.orig/src/Makefile pifmrds-c67306ea9b8d827f45e0d90279d367e97119bcb1/src/Makefile
--- pifmrds-c67306ea9b8d827f45e0d90279d367e97119bcb1.orig/src/Makefile	2014-06-21 16:46:49.101118754 +0200
+++ pifmrds-c67306ea9b8d827f45e0d90279d367e97119bcb1/src/Makefile	2014-06-21 16:47:47.801745683 +0200
@@ -1,10 +1,10 @@
 CC = gcc
 
 app: rds.o waveforms.o pi_fm_rds.o fm_mpx.o control_pipe.o
-	$(CC) -o pi_fm_rds rds.o waveforms.o pi_fm_rds.o fm_mpx.o control_pipe.o -lm -lsndfile
+	$(CC) $(LDFLAGS) -o pi_fm_rds rds.o waveforms.o pi_fm_rds.o fm_mpx.o control_pipe.o -lm -lsndfile
 
 rds_wav: rds.o waveforms.o rds_wav.o fm_mpx.o
-	$(CC) -o rds_wav rds_wav.o rds.o waveforms.o fm_mpx.o -lm -lsndfile
+	$(CC) $(LDFLAGS) -o rds_wav rds_wav.o rds.o waveforms.o fm_mpx.o -lm -lsndfile
 
 rds.o: rds.c waveforms.h
 	$(CC) $(CFLAGS) rds.c
