tools/watchgnupg: select() is from sys/select.h

Signed-off-by: "Yann E. MORIN" <yann.morin.1998@free.fr>

diff -durN gnupg2-2.0.29.orig/tools/watchgnupg.c gnupg2-2.0.29/tools/watchgnupg.c
--- gnupg2-2.0.29.orig/tools/watchgnupg.c	2015-09-08 14:39:24.000000000 +0200
+++ gnupg2-2.0.29/tools/watchgnupg.c	2015-12-16 18:17:17.758057787 +0100
@@ -32,6 +32,7 @@
 #include <sys/un.h>
 #include <fcntl.h>
 #include <time.h>
+#include <sys/select.h>
 
 #define PGM "watchgnupg"
 
