From adcf320db5cc2844209d001f9484b0ff0cab4e0b Mon Sep 17 00:00:00 2001
From: Gustavo Zacarias <gustavo@zacarias.com.ar>
Date: Tue, 11 Nov 2014 20:22:37 -0300
Subject: [PATCH] build: uclinux is also linux

Signed-off-by: Gustavo Zacarias <gustavo@zacarias.com.ar>
---
Status: pull request in github

diff --git a/configure.ac b/configure.ac
index 3059e52..10388fd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -70,7 +70,7 @@ fi
 PKG_CHECK_MODULES([CONFUSE], [libconfuse], [], AC_MSG_ERROR([requires libconfuse]))
 
 case ${target_os} in
-    linux*)
+    linux*|uclinux*)
         PKG_CHECK_MODULES([LIBNL], [libnl-3.0], [], AC_MSG_ERROR([requires libnl3-dev]))
         PKG_CHECK_MODULES([LIBNL_ROUTE], [libnl-route-3.0], [], AC_MSG_ERROR([requires libnl3-route]))
     ;;
-- 
2.0.4

