diff -Nura bmon/configure bmon.libnl/configure
--- bmon/configure	2005-04-05 12:01:33.000000000 -0300
+++ bmon.libnl/configure	2009-06-10 16:10:52.000000000 -0300
@@ -7327,13 +7327,13 @@
 echo "${ECHO_T}$ac_cv_lib_nl_nl_connect" >&6
 if test $ac_cv_lib_nl_nl_connect = yes; then
 
-
-cat >>confdefs.h <<_ACEOF
-#define HAVE_NL "1"
-_ACEOF
-
-	LIBNL="-lnl"
-	NL="Yes"
+	case ${target_os} in
+	    *linux*)
+		echo
+		echo "*** Warning: Building bmon on Linux w/o libnl ***"
+		echo
+	    ;;
+	esac
 
 else
 
