---
 Makefile |   14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

Index: config/Makefile
===================================================================
--- config.orig/Makefile
+++ config/Makefile
@@ -174,18 +174,30 @@
 ifeq ($(MAKECMDGOALS),nconfig)
 	hostprogs-y += nconf
 endif
+ifeq ($(findstring nconf,$(MAKECMDGOALS)),nconf)
+       hostprogs-y += nconf
+endif
 
 ifeq ($(MAKECMDGOALS),menuconfig)
 	hostprogs-y += mconf
 endif
+ifeq ($(findstring mconf,$(MAKECMDGOALS)),mconf)
+       hostprogs-y += mconf
+endif
 
 ifeq ($(MAKECMDGOALS),xconfig)
 	qconf-target := 1
 endif
+ifeq ($(findstring qconf,$(MAKECMDGOALS)),qconf)
+	qconf-target := 1
+endif
+
 ifeq ($(MAKECMDGOALS),gconfig)
 	gconf-target := 1
 endif
-
+ifeq ($(findstring gconf,$(MAKECMDGOALS)),gconf)
+	gconf-target := 1
+endif
 
 ifeq ($(qconf-target),1)
 qconf-cxxobjs	:= qconf.o
