---
 gconf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: kconfig/gconf.c
===================================================================
--- kconfig.orig/gconf.c
+++ kconfig/gconf.c
@@ -1462,7 +1462,7 @@ int main(int ac, char *av[])
 	/* Determine GUI path */
 	env = getenv(SRCTREE);
 	if (env)
-		glade_file = g_strconcat(env, "/scripts/kconfig/gconf.glade", NULL);
+		glade_file = g_strconcat(env, "/support/kconfig/gconf.glade", NULL);
 	else if (av[0][0] == '/')
 		glade_file = g_strconcat(av[0], ".glade", NULL);
 	else
