---
 gconf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: b/gconf.c
===================================================================
--- a/gconf.c
+++ b/gconf.c
@@ -1486,7 +1486,7 @@
 	/* Determine GUI path */
 	env = getenv(SRCTREE);
 	if (env)
-		glade_file = g_strconcat(env, "/scripts/kconfig/gconf.glade", NULL);
+		glade_file = g_strconcat(env, "/support/kconfig/gconf.glade", NULL);
 	else if (av[0][0] == '/')
 		glade_file = g_strconcat(av[0], ".glade", NULL);
 	else
