From 4426de35f0e5dd5b0152474ce6a885b73de60656 Mon Sep 17 00:00:00 2001
From: Romain Naour <romain.naour@openwide.fr>
Date: Tue, 16 Dec 2014 23:50:33 +0100
Subject: [PATCH 2/2] use correct include path for libnl-3.0

Signed-off-by: Romain Naour <romain.naour@openwide.fr>
---
 src/drivers/drivers.mak | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/drivers/drivers.mak b/src/drivers/drivers.mak
index cdb913e..e9fc83c 100644
--- a/src/drivers/drivers.mak
+++ b/src/drivers/drivers.mak
@@ -35,7 +35,7 @@ NEED_RFKILL=y
 ifdef CONFIG_LIBNL32
   DRV_LIBS += -lnl-3
   DRV_LIBS += -lnl-genl-3
-  DRV_CFLAGS += -DCONFIG_LIBNL20 -I/usr/include/libnl3
+  DRV_CFLAGS += -DCONFIG_LIBNL20 `pkg-config --cflags libnl-3.0`
 ifdef CONFIG_LIBNL3_ROUTE
   DRV_LIBS += -lnl-route-3
   DRV_CFLAGS += -DCONFIG_LIBNL3_ROUTE
-- 
1.9.3

