From 2d0450bfa3fb7003098da1a7fe24c3ae140912dc Mon Sep 17 00:00:00 2001
From: Gregory Hermant <gregory.hermant@calao-systems.com>
Date: Mon, 22 Oct 2012 15:11:53 +0200
Subject: [PATCH] pthsem build on Linux 3.x host


Signed-off-by: Michael Markstaller <mm@elabnet.de>
Signed-off-by: Gregory Hermant <gregory.hermant@calao-systems.com>
---
 acinclude.m4 |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/acinclude.m4 b/acinclude.m4
index 03b42de..fea9bc8 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -892,6 +892,8 @@ case $PLATFORM in
         case "x`uname -r`" in
 changequote(, )dnl
             x2.[23456789]* ) ;;
+changequote(, )dnl
+            x3.* ) ;;
 changequote([, ])
             * ) braindead=yes ;;
         esac
-- 
1.7.9.5

