diff --git a/Makefile b/Makefile
index 83d2536..d9a300c 100644
--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 ALL_PROGS = v4l2copy v4l2convert_yuv v4l2source_yuv v4l2dump
-CFLAGS = -W -Wall -pthread -g -pipe $(CFLAGS_EXTRA) -I include
+CFLAGS = -W -Wall -pthread -g -pipe $(CFLAGS_EXTRA) -I include -ljpeg
 RM = rm -rf
 CC = $(CROSS)gcc
 CXX = $(CROSS)g++
@@ -18,24 +18,19 @@ CFLAGS += -I v4l2wrapper/inc
 .DEFAULT_GOAL := all
 
 # raspberry tools using ilclient
-ILCLIENTDIR=/opt/vc/src/hello_pi/libs/ilclient
-ifneq ($(wildcard $(ILCLIENTDIR)),)
-CFLAGS  +=-I /opt/vc/include/ -I /opt/vc/include/interface/vcos/ -I /opt/vc/include/interface/vcos/pthreads/ -I /opt/vc/include/interface/vmcs_host/linux/ -I $(ILCLIENTDIR) 
+
+ifneq ($(HAVE_RPI),)
 CFLAGS  += -DOMX_SKIP64BIT
-LDFLAGS +=-L /opt/vc/lib -L $(ILCLIENTDIR) -lpthread -lopenmaxil -lbcm_host -lvcos -lvchiq_arm
+CFLAGS  += -lpthread -lopenmaxil -lbcm_host -lvcos -lvchostif -lvchiq_arm -lilclient
 
-v4l2compress_omx: src/encode_omx.cpp src/v4l2compress_omx.cpp $(ILCLIENTDIR)/libilclient.a libv4l2wrapper.a 
+v4l2compress_omx: src/encode_omx.cpp src/v4l2compress_omx.cpp libv4l2wrapper.a
 	$(CXX) -o $@ $^ -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi $(CFLAGS) $(LDFLAGS) 
 
-v4l2grab_h264: src/encode_omx.cpp src/v4l2grab_h264.cpp $(ILCLIENTDIR)/libilclient.a libv4l2wrapper.a
+v4l2grab_h264: src/encode_omx.cpp src/v4l2grab_h264.cpp libv4l2wrapper.a
 	$(CXX) -o $@ $^ -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi $(CFLAGS) $(LDFLAGS) 
 
-v4l2display_h264: src/v4l2display_h264.cpp $(ILCLIENTDIR)/libilclient.a libv4l2wrapper.a
+v4l2display_h264: src/v4l2display_h264.cpp libv4l2wrapper.a
 	$(CXX) -o $@ $^ -DHAVE_LIBBCM_HOST -DUSE_EXTERNAL_LIBBCM_HOST -DUSE_VCHIQ_ARM -Wno-psabi $(CFLAGS) $(LDFLAGS) 
-
-
-$(ILCLIENTDIR)/libilclient.a:
-	make -C $(ILCLIENTDIR)
 	
 ALL_PROGS+=v4l2grab_h264
 ALL_PROGS+=v4l2display_h264
@@ -43,48 +38,44 @@ ALL_PROGS+=v4l2compress_omx
 endif
 
 # opencv
-ifneq ($(wildcard /usr/include/opencv),)
-ALL_PROGS+=v4l2detect_yuv
-endif
+#ifneq ($(wildcard /usr/include/opencv),)
+#ALL_PROGS+=v4l2detect_yuv
+#endif
 
 # libx264
-ifneq ($(wildcard /usr/include/x264.h),)
+ifneq ($(HAVE_X264),)
 ALL_PROGS+=v4l2compress_h264
 endif
 
 # libx265
-ifneq ($(wildcard /usr/include/x265.h),)
+ifneq ($(HAVE_X265),)
 ALL_PROGS+=v4l2compress_x265
 endif
 
 # libvpx
-ifneq ($(wildcard /usr/include/vpx),)
+ifneq ($(HAVE_LIBVPX),)
 ALL_PROGS+=v4l2compress_vpx
 endif
 
 # libjpeg
-ifneq ($(wildcard /usr/include/jpeglib.h),)
+ifneq ($(HAVE_LIBJPEG),)
 ALL_PROGS+=v4l2compress_jpeg v4l2uncompress_jpeg
 CFLAGS += -DHAVE_JPEG
-LDFLAGS += -ljpeg
 endif
 
 # libfuse
-ifneq ($(wildcard /usr/include/fuse.h),)
+ifneq ($(HAVE_LIBFUSE),)
 ALL_PROGS+=v4l2fuse
 endif
 
 all: $(ALL_PROGS)
 
 libyuv.a:
-	cd libyuv && cmake . && make 
-	mv libyuv/libyuv.a .
-	make -C libyuv clean
+	cp libyuv/libyuv.a .
 
 libv4l2wrapper.a: 
-	make -C v4l2wrapper
-	mv v4l2wrapper/libv4l2wrapper.a .
-	make -C v4l2wrapper clean
+	make CC="$(CC)" CFLAGS_EXTRA="$(CFLAGS_EXTRA)" LDFLAGS="$(LDFLAGS)" -C v4l2wrapper all
+	cp v4l2wrapper/libv4l2wrapper.a .
 
 # read V4L2 capture -> write V4L2 output
 v4l2copy: src/v4l2copy.cpp  libv4l2wrapper.a
@@ -126,13 +117,13 @@ h264bitstream/Makefile:
 	cd h264bitstream && autoreconf -i -f && ./configure --host $(shell $(CC) -dumpmachine)
 
 h264bitstream/.libs/libh264bitstream.so: h264bitstream/Makefile
-	make -C h264bitstream 
+	make CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" -C h264bitstream
 
 hevcbitstream/Makefile:
 	cd hevcbitstream && autoreconf -i -f && LDFLAGS=-lm ./configure --host $(shell $(CC) -dumpmachine)
 
 hevcbitstream/.libs/libhevcbitstream.so: hevcbitstream/Makefile
-	make -C hevcbitstream 
+	make CC="$(CC)" CFLAGS="$(CFLAGS)" LDFLAGS="$(LDFLAGS)" -C hevcbitstream
 
 v4l2dump: src/v4l2dump.cpp libv4l2wrapper.a h264bitstream/.libs/libh264bitstream.so  hevcbitstream/.libs/libhevcbitstream.so libyuv.a
 	$(CXX) -o $@ $(CFLAGS) $^ $(LDFLAGS) -Ih264bitstream  -Ihevcbitstream -Wl,-rpath=./h264bitstream/.libs,-rpath=./hevcbitstream/.libs -I libyuv/include 
