diff -uNr apispec-0.39.0-orig/setup.py apispec-0.39.0-patched/setup.py
--- apispec-0.39.0-orig/setup.py	2018-06-29 05:50:26.000000000 +0300
+++ apispec-0.39.0-patched/setup.py	2019-01-24 23:36:41.947206745 +0200
@@ -13,7 +13,7 @@
     Raises RuntimeError if not found.
     """
     version = ''
-    with open(fname, 'r') as fp:
+    with open(fname, 'r', encoding='utf-8') as fp:
         reg = re.compile(r'__version__ = [\'"]([^\'"]*)[\'"]')
         for line in fp:
             m = reg.match(line)
